{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-12-05 14:05:07--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 116797085 (111M) [application/x-gzip]\n",
      "Saving to: â€˜genome/genome.fa.gzâ€™\n",
      "\n",
      "genome/genome.fa.gz 100%[===================>] 111.39M   590KB/s    in 3m 14s  \n",
      "\n",
      "2024-12-05 14:08:22 (589 KB/s) - â€˜genome/genome.fa.gzâ€™ saved [116797085/116797085]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# !conda activate final_project_env\n",
    "# !mkdir genome\n",
    "# %cd genome\n",
    "# !wget -nc -O genome/genome.fa.gz https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz \n",
    "# !gzip -d -f genome/genome.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-11-21 16:30:04--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3228084 (3.1M) [application/x-gzip]\n",
      "Saving to: â€˜STDOUTâ€™\n",
      "\n",
      "-                   100%[===================>]   3.08M  3.82MB/s    in 0.8s    \n",
      "\n",
      "2024-11-21 16:30:06 (3.82 MB/s) - written to stdout [3228084/3228084]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# !wget -O - https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz | \\\n",
    "#     gzip -f -d > genome/annotations.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !samtools faidx genome/genome.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory â€˜fastq_trimmedâ€™: File exists\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_trimmed\n",
    "# # !cut  -d ' ' -f 2 /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/M007087_WheelerN_md5sum.txt | tail -n +2 > fastq_files.txt\n",
    "# # !while IFS= read -r line; do \\\n",
    "#  \n",
    "# #     done < fastq_files.txt\n",
    "# #  -q 37 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int_01\n",
      "Int_01\n",
      "Int_02\n",
      "Int_02\n",
      "Int_03\n",
      "Int_03\n",
      "Int_04\n",
      "Int_04\n",
      "Liv_01\n",
      "Liv_01\n",
      "Liv_02\n",
      "Liv_02\n",
      "Liv_03\n",
      "Liv_03\n",
      "Liv_04\n",
      "Liv_04\n"
     ]
    }
   ],
   "source": [
    "for i in [\"Int\",\"Liv\"]:\n",
    "    for a in  [\"01\",\"02\",\"03\",\"04\"]:\n",
    "        print(i + \"_\" + a)\n",
    "        print(i + \"_\" + a)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/*_R1_100.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R1_100.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R2_100.fastq.gz\n",
      "When a paired-end trimming option such as -A/-G/-B/-U, is used, a second output file needs to be specified via -p (--paired-output).\n"
     ]
    }
   ],
   "source": [
    "# !for fastq in /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R1_100.fastq.gz; do \\\n",
    "#     for fastq2 in /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R2_100.fastq.gz; do \\\n",
    "#         base_name=$(basename \"$fastq\" .fastq.gz); \\\n",
    "#     cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/$base_name.fastq.gz $fastq $fastq2; \\\n",
    "# done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_04_1.fastq.gz -p ./fastq_trimmed/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           01:10:18   179,662,246 reads @  23.5 Âµs/read;   2.56 M reads/minute\n",
      "Finished in 4218.643 s (23.481 Âµs/read; 2.56 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:        179,662,246\n",
      "  Read 1 with adapter:              55,990,071 (31.2%)\n",
      "  Read 2 with adapter:              51,084,573 (28.4%)\n",
      "Pairs written (passing filters):   179,662,246 (100.0%)\n",
      "\n",
      "Total basepairs processed: 54,257,998,292 bp\n",
      "  Read 1: 27,128,999,146 bp\n",
      "  Read 2: 27,128,999,146 bp\n",
      "Total written (filtered):  50,300,380,211 bp (92.7%)\n",
      "  Read 1: 24,794,104,131 bp\n",
      "  Read 2: 25,506,276,080 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 55990071 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 13.2%\n",
      "  C: 27.1%\n",
      "  G: 33.1%\n",
      "  T: 17.4%\n",
      "  none/other: 9.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2576086\t2807222.6\t0\t2576086\n",
      "4\t1389355\t701805.6\t0\t1389355\n",
      "5\t1031142\t175451.4\t0\t1031142\n",
      "6\t919429\t43862.9\t0\t919429\n",
      "7\t894925\t10965.7\t0\t894925\n",
      "8\t884633\t2741.4\t0\t884633\n",
      "9\t908306\t685.4\t0\t902794 5512\n",
      "10\t915963\t171.3\t1\t888223 27740\n",
      "11\t931994\t42.8\t1\t909844 22150\n",
      "12\t932941\t10.7\t1\t908932 24009\n",
      "13\t924914\t2.7\t1\t900257 24657\n",
      "14\t923116\t0.7\t1\t895520 27596\n",
      "15\t916552\t0.2\t1\t888199 28353\n",
      "16\t917865\t0.0\t1\t884348 33517\n",
      "17\t911421\t0.0\t1\t876372 35049\n",
      "18\t916165\t0.0\t1\t880487 35252 426\n",
      "19\t915582\t0.0\t1\t877649 36647 1286\n",
      "20\t928280\t0.0\t2\t884175 39446 4659\n",
      "21\t933909\t0.0\t2\t889485 39762 4662\n",
      "22\t928600\t0.0\t2\t882543 41135 4922\n",
      "23\t939665\t0.0\t2\t892693 41943 5029\n",
      "24\t937989\t0.0\t2\t887724 44798 5467\n",
      "25\t916568\t0.0\t2\t865905 45220 5443\n",
      "26\t903784\t0.0\t2\t850156 47699 5929\n",
      "27\t887772\t0.0\t2\t837031 45241 5440 60\n",
      "28\t886102\t0.0\t2\t829355 50927 5743 77\n",
      "29\t889022\t0.0\t2\t830132 52415 6243 232\n",
      "30\t885664\t0.0\t3\t825264 52047 6153 2200\n",
      "31\t890847\t0.0\t3\t829408 53152 6171 2116\n",
      "32\t882015\t0.0\t3\t821329 52260 6214 2212\n",
      "33\t880339\t0.0\t3\t815001 55841 6955 2542\n",
      "34\t863560\t0.0\t3\t805394 50028 5935 2203\n",
      "35\t842328\t0.0\t3\t786258 48095 5843 2132\n",
      "36\t829161\t0.0\t3\t773772 47779 5540 2070\n",
      "37\t795701\t0.0\t3\t743334 45232 5208 1927\n",
      "38\t781836\t0.0\t3\t729445 45209 5313 1869\n",
      "39\t773555\t0.0\t3\t722975 43519 5187 1874\n",
      "40\t756090\t0.0\t3\t707049 42192 4968 1881\n",
      "41\t745585\t0.0\t3\t697809 41124 4889 1763\n",
      "42\t739855\t0.0\t3\t692709 40657 4734 1755\n",
      "43\t721372\t0.0\t3\t674919 40037 4718 1698\n",
      "44\t713788\t0.0\t3\t667005 40487 4626 1670\n",
      "45\t681002\t0.0\t3\t636840 38151 4488 1523\n",
      "46\t653466\t0.0\t3\t610944 36756 4258 1508\n",
      "47\t620783\t0.0\t3\t579078 36291 4001 1413\n",
      "48\t591534\t0.0\t3\t552016 34439 3714 1365\n",
      "49\t567821\t0.0\t3\t531112 31776 3685 1248\n",
      "50\t535000\t0.0\t3\t500457 30017 3330 1196\n",
      "51\t520561\t0.0\t3\t487074 29104 3235 1148\n",
      "52\t503795\t0.0\t3\t471342 28169 3141 1143\n",
      "53\t484865\t0.0\t3\t454081 26720 3024 1040\n",
      "54\t466373\t0.0\t3\t436782 25879 2748 964\n",
      "55\t442340\t0.0\t3\t414172 24529 2692 947\n",
      "56\t416159\t0.0\t3\t389853 23005 2449 852\n",
      "57\t385357\t0.0\t3\t360884 21407 2265 801\n",
      "58\t355911\t0.0\t3\t333609 19497 2040 765\n",
      "59\t328116\t0.0\t3\t307241 18226 1937 712\n",
      "60\t308176\t0.0\t3\t288720 17002 1831 623\n",
      "61\t282184\t0.0\t3\t264758 15260 1593 573\n",
      "62\t266660\t0.0\t3\t249963 14632 1491 574\n",
      "63\t249214\t0.0\t3\t233527 13698 1457 532\n",
      "64\t233598\t0.0\t3\t218518 13280 1310 490\n",
      "65\t219761\t0.0\t3\t205305 12707 1293 456\n",
      "66\t201411\t0.0\t3\t188345 11527 1166 373\n",
      "67\t184193\t0.0\t3\t172216 10580 1063 334\n",
      "68\t167244\t0.0\t3\t156340 9605 971 328\n",
      "69\t151316\t0.0\t3\t141610 8531 875 300\n",
      "70\t137677\t0.0\t3\t128835 7771 781 290\n",
      "71\t124340\t0.0\t3\t116255 7192 680 213\n",
      "72\t113535\t0.0\t3\t105997 6693 663 182\n",
      "73\t104310\t0.0\t3\t97656 5869 593 192\n",
      "74\t96724\t0.0\t3\t90339 5666 532 187\n",
      "75\t89311\t0.0\t3\t83639 4995 510 167\n",
      "76\t82136\t0.0\t3\t76938 4625 424 149\n",
      "77\t74305\t0.0\t3\t69509 4197 434 165\n",
      "78\t67191\t0.0\t3\t63051 3665 358 117\n",
      "79\t60023\t0.0\t3\t56297 3286 321 119\n",
      "80\t56401\t0.0\t3\t52973 3006 313 109\n",
      "81\t48741\t0.0\t3\t45733 2637 281 90\n",
      "82\t43748\t0.0\t3\t40872 2531 236 109\n",
      "83\t40439\t0.0\t3\t37622 2470 250 97\n",
      "84\t36334\t0.0\t3\t33969 2056 203 106\n",
      "85\t33052\t0.0\t3\t30872 1900 203 77\n",
      "86\t30901\t0.0\t3\t28871 1786 184 60\n",
      "87\t28126\t0.0\t3\t26357 1534 174 61\n",
      "88\t25728\t0.0\t3\t24092 1429 150 57\n",
      "89\t23752\t0.0\t3\t22179 1339 154 80\n",
      "90\t23590\t0.0\t3\t22052 1329 149 60\n",
      "91\t21172\t0.0\t3\t19755 1212 141 64\n",
      "92\t17960\t0.0\t3\t16762 1047 103 48\n",
      "93\t16642\t0.0\t3\t15543 944 106 49\n",
      "94\t14200\t0.0\t3\t13244 810 94 52\n",
      "95\t12855\t0.0\t3\t11913 796 105 41\n",
      "96\t12184\t0.0\t3\t11361 682 88 53\n",
      "97\t12228\t0.0\t3\t11332 766 89 41\n",
      "98\t12063\t0.0\t3\t11287 609 116 51\n",
      "99\t12967\t0.0\t3\t12139 695 97 36\n",
      "100\t10547\t0.0\t3\t9798 629 69 51\n",
      "101\t9753\t0.0\t3\t9013 602 98 40\n",
      "102\t7907\t0.0\t3\t7378 438 68 23\n",
      "103\t7201\t0.0\t3\t6694 412 59 36\n",
      "104\t6604\t0.0\t3\t6137 383 53 31\n",
      "105\t6350\t0.0\t3\t5869 397 53 31\n",
      "106\t6122\t0.0\t3\t5662 378 56 26\n",
      "107\t6195\t0.0\t3\t5709 410 52 24\n",
      "108\t5795\t0.0\t3\t5347 370 63 15\n",
      "109\t5266\t0.0\t3\t4889 318 45 14\n",
      "110\t5191\t0.0\t3\t4777 347 46 21\n",
      "111\t4792\t0.0\t3\t4423 299 47 23\n",
      "112\t4576\t0.0\t3\t4237 294 33 12\n",
      "113\t4143\t0.0\t3\t3796 286 40 21\n",
      "114\t3870\t0.0\t3\t3585 242 29 14\n",
      "115\t3869\t0.0\t3\t3546 272 35 16\n",
      "116\t3807\t0.0\t3\t3515 249 35 8\n",
      "117\t3979\t0.0\t3\t3659 263 38 19\n",
      "118\t3886\t0.0\t3\t3610 221 36 19\n",
      "119\t4036\t0.0\t3\t3722 258 38 18\n",
      "120\t3974\t0.0\t3\t3669 272 16 17\n",
      "121\t4386\t0.0\t3\t4028 291 44 23\n",
      "122\t3858\t0.0\t3\t3572 247 26 13\n",
      "123\t4421\t0.0\t3\t4080 282 34 25\n",
      "124\t5966\t0.0\t3\t5554 337 51 24\n",
      "125\t5969\t0.0\t3\t5525 373 49 22\n",
      "126\t5035\t0.0\t3\t4682 302 33 18\n",
      "127\t5453\t0.0\t3\t5030 351 59 13\n",
      "128\t5285\t0.0\t3\t4932 311 30 12\n",
      "129\t5285\t0.0\t3\t4856 384 36 9\n",
      "130\t4539\t0.0\t3\t4143 342 44 10\n",
      "131\t6301\t0.0\t3\t5850 405 38 8\n",
      "132\t5759\t0.0\t3\t5336 378 35 10\n",
      "133\t8754\t0.0\t3\t8086 611 52 5\n",
      "134\t6320\t0.0\t3\t5928 355 29 8\n",
      "135\t8411\t0.0\t3\t7840 520 42 9\n",
      "136\t8800\t0.0\t3\t8230 544 23 3\n",
      "137\t10445\t0.0\t3\t9814 576 44 11\n",
      "138\t11916\t0.0\t3\t11189 686 33 8\n",
      "139\t17180\t0.0\t3\t15676 1399 95 10\n",
      "140\t11366\t0.0\t3\t10622 694 43 7\n",
      "141\t12986\t0.0\t3\t12191 721 63 11\n",
      "142\t27921\t0.0\t3\t26136 1634 136 15\n",
      "143\t11702\t0.0\t3\t10982 650 62 8\n",
      "144\t13503\t0.0\t3\t12663 772 57 11\n",
      "145\t15389\t0.0\t3\t14514 797 71 7\n",
      "146\t19032\t0.0\t3\t17922 1013 88 9\n",
      "147\t25525\t0.0\t3\t24071 1327 112 15\n",
      "148\t27348\t0.0\t3\t25856 1353 120 19\n",
      "149\t9867\t0.0\t3\t9223 574 56 14\n",
      "150\t43797\t0.0\t3\t1810 33439 8131 417\n",
      "151\t5166438\t0.0\t3\t682770 4188686 268020 26962\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 51084573 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.3%\n",
      "  C: 29.4%\n",
      "  G: 41.6%\n",
      "  T: 10.3%\n",
      "  none/other: 1.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2962082\t2807222.6\t0\t2962082\n",
      "4\t1410466\t701805.6\t0\t1410466\n",
      "5\t1060221\t175451.4\t0\t1060221\n",
      "6\t924618\t43862.9\t0\t924618\n",
      "7\t893196\t10965.7\t0\t893196\n",
      "8\t887812\t2741.4\t0\t887812\n",
      "9\t910736\t685.4\t0\t903532 7204\n",
      "10\t912994\t171.3\t1\t889902 23092\n",
      "11\t930287\t42.8\t1\t909655 20632\n",
      "12\t930677\t10.7\t1\t909260 21417\n",
      "13\t921698\t2.7\t1\t899687 22011\n",
      "14\t920333\t0.7\t1\t896948 23385\n",
      "15\t913811\t0.2\t1\t890100 23711\n",
      "16\t915924\t0.0\t1\t889610 26314\n",
      "17\t908785\t0.0\t1\t878851 29934\n",
      "18\t914671\t0.0\t1\t883947 30412 312\n",
      "19\t913186\t0.0\t1\t878061 33648 1477\n",
      "20\t924281\t0.0\t2\t784207 133978 6096\n",
      "21\t927414\t0.0\t2\t627135 266181 34098\n",
      "22\t922846\t0.0\t2\t738525 146470 37851\n",
      "23\t933352\t0.0\t2\t767077 134814 31461\n",
      "24\t931806\t0.0\t2\t772211 137225 22370\n",
      "25\t898699\t0.0\t2\t689984 164740 43975\n",
      "26\t882264\t0.0\t2\t693595 148344 40325\n",
      "27\t868585\t0.0\t2\t672529 152776 43213 67\n",
      "28\t860858\t0.0\t2\t385644 397842 77241 131\n",
      "29\t864741\t0.0\t2\t611587 199305 52369 1480\n",
      "30\t866059\t0.0\t3\t584631 198697 62174 20557\n",
      "31\t861755\t0.0\t3\t578476 195281 60782 27216\n",
      "32\t848879\t0.0\t3\t528567 214505 73945 31862\n",
      "33\t856009\t0.0\t3\t575763 195841 59445 24960\n",
      "34\t838100\t0.0\t3\t558403 192850 61748 25099\n",
      "35\t820729\t0.0\t3\t546893 190856 58935 24045\n",
      "36\t792004\t0.0\t3\t516407 185301 61713 28583\n",
      "37\t757737\t0.0\t3\t523763 165726 47986 20262\n",
      "38\t758799\t0.0\t3\t498400 176836 58848 24715\n",
      "39\t747049\t0.0\t3\t490987 174318 56931 24813\n",
      "40\t734366\t0.0\t3\t487934 170519 53163 22750\n",
      "41\t695070\t0.0\t3\t468627 155753 47518 23172\n",
      "42\t723719\t0.0\t3\t486455 168587 49519 19158\n",
      "43\t697301\t0.0\t3\t461414 161574 51949 22364\n",
      "44\t691469\t0.0\t3\t469888 157521 45630 18430\n",
      "45\t663974\t0.0\t3\t460112 146275 41597 15990\n",
      "46\t639508\t0.0\t3\t431624 145906 44530 17448\n",
      "47\t605972\t0.0\t3\t402773 140571 44305 18323\n",
      "48\t578408\t0.0\t3\t400224 127487 36729 13968\n",
      "49\t555609\t0.0\t3\t374995 127885 38082 14647\n",
      "50\t515958\t0.0\t3\t350007 115628 34829 15494\n",
      "51\t510935\t0.0\t3\t350716 113323 33817 13079\n",
      "52\t490257\t0.0\t3\t342853 105187 29981 12236\n",
      "53\t469431\t0.0\t3\t322690 102482 31166 13093\n",
      "54\t456571\t0.0\t3\t318353 100103 27677 10438\n",
      "55\t429023\t0.0\t3\t282063 100489 32493 13978\n",
      "56\t405243\t0.0\t3\t280266 87486 26217 11274\n",
      "57\t374375\t0.0\t3\t266261 76862 22080 9172\n",
      "58\t346328\t0.0\t3\t238095 76049 22954 9230\n",
      "59\t318393\t0.0\t3\t216611 70968 21934 8880\n",
      "60\t298113\t0.0\t3\t206093 65702 18771 7547\n",
      "61\t275283\t0.0\t3\t193955 57504 16664 7160\n",
      "62\t261271\t0.0\t3\t179133 57881 17282 6975\n",
      "63\t242102\t0.0\t3\t171339 51186 13950 5627\n",
      "64\t229391\t0.0\t3\t165865 46571 12208 4747\n",
      "65\t214128\t0.0\t3\t154274 43378 11859 4617\n",
      "66\t195746\t0.0\t3\t139228 40537 11262 4719\n",
      "67\t180566\t0.0\t3\t127113 38554 10583 4316\n",
      "68\t164296\t0.0\t3\t113183 36444 10606 4063\n",
      "69\t147962\t0.0\t3\t103491 31394 9270 3807\n",
      "70\t135544\t0.0\t3\t96634 28304 7779 2827\n",
      "71\t120161\t0.0\t3\t83827 25554 7642 3138\n",
      "72\t111696\t0.0\t3\t77431 24568 7152 2545\n",
      "73\t101480\t0.0\t3\t69963 22092 6657 2768\n",
      "74\t94143\t0.0\t3\t65537 20482 5730 2394\n",
      "75\t87143\t0.0\t3\t60811 18514 5525 2293\n",
      "76\t80393\t0.0\t3\t56688 16703 4978 2024\n",
      "77\t72380\t0.0\t3\t51487 15001 4129 1763\n",
      "78\t65874\t0.0\t3\t46189 13831 4170 1684\n",
      "79\t58464\t0.0\t3\t41225 12173 3539 1527\n",
      "80\t54883\t0.0\t3\t39613 10974 3038 1258\n",
      "81\t47265\t0.0\t3\t33273 9834 2926 1232\n",
      "82\t42688\t0.0\t3\t29341 9302 2803 1242\n",
      "83\t39495\t0.0\t3\t28179 8113 2236 967\n",
      "84\t35206\t0.0\t3\t25677 6919 1838 772\n",
      "85\t32335\t0.0\t3\t23306 6422 1851 756\n",
      "86\t30411\t0.0\t3\t21960 6050 1755 646\n",
      "87\t27520\t0.0\t3\t19723 5642 1525 630\n",
      "88\t24936\t0.0\t3\t17645 5250 1403 638\n",
      "89\t23184\t0.0\t3\t17027 4389 1251 517\n",
      "90\t23277\t0.0\t3\t16818 4652 1297 510\n",
      "91\t20716\t0.0\t3\t14527 4370 1273 546\n",
      "92\t17662\t0.0\t3\t12604 3517 1057 484\n",
      "93\t16295\t0.0\t3\t11988 3086 857 364\n",
      "94\t13653\t0.0\t3\t9807 2698 787 361\n",
      "95\t12635\t0.0\t3\t9399 2338 610 288\n",
      "96\t11805\t0.0\t3\t8345 2341 740 379\n",
      "97\t11906\t0.0\t3\t8608 2277 722 299\n",
      "98\t11613\t0.0\t3\t8574 2017 715 307\n",
      "99\t12652\t0.0\t3\t9279 2328 721 324\n",
      "100\t10214\t0.0\t3\t7781 1645 505 283\n",
      "101\t9540\t0.0\t3\t7327 1540 449 224\n",
      "102\t7632\t0.0\t3\t5671 1317 434 210\n",
      "103\t7031\t0.0\t3\t5298 1183 366 184\n",
      "104\t6372\t0.0\t3\t4754 1098 358 162\n",
      "105\t6057\t0.0\t3\t4613 1007 286 151\n",
      "106\t5929\t0.0\t3\t4434 1047 316 132\n",
      "107\t6022\t0.0\t3\t4570 1035 270 147\n",
      "108\t5661\t0.0\t3\t4120 1084 299 158\n",
      "109\t5148\t0.0\t3\t3549 1078 340 181\n",
      "110\t5030\t0.0\t3\t3545 1006 317 162\n",
      "111\t4711\t0.0\t3\t3404 906 269 132\n",
      "112\t4490\t0.0\t3\t3223 868 271 128\n",
      "113\t4063\t0.0\t3\t2869 825 249 120\n",
      "114\t3812\t0.0\t3\t2750 741 222 99\n",
      "115\t3779\t0.0\t3\t2551 787 303 138\n",
      "116\t3759\t0.0\t3\t2682 764 223 90\n",
      "117\t3957\t0.0\t3\t2748 813 266 130\n",
      "118\t3825\t0.0\t3\t2748 756 230 91\n",
      "119\t3917\t0.0\t3\t2644 877 273 123\n",
      "120\t3545\t0.0\t3\t2529 694 241 81\n",
      "121\t4076\t0.0\t3\t2775 899 282 120\n",
      "122\t3487\t0.0\t3\t2310 831 242 104\n",
      "123\t4235\t0.0\t3\t2842 917 314 162\n",
      "124\t6196\t0.0\t3\t3991 1405 570 230\n",
      "125\t5616\t0.0\t3\t3685 1262 465 204\n",
      "126\t4793\t0.0\t3\t3309 1040 297 147\n",
      "127\t5163\t0.0\t3\t3180 1230 495 258\n",
      "128\t5152\t0.0\t3\t3469 1154 378 151\n",
      "129\t4982\t0.0\t3\t3084 1223 442 233\n",
      "130\t4305\t0.0\t3\t2887 971 320 127\n",
      "131\t6061\t0.0\t3\t4231 1262 408 160\n",
      "132\t5562\t0.0\t3\t3988 1110 329 135\n",
      "133\t8292\t0.0\t3\t5726 1875 498 193\n",
      "134\t6124\t0.0\t3\t4387 1301 310 126\n",
      "135\t8086\t0.0\t3\t5615 1742 505 224\n",
      "136\t8551\t0.0\t3\t6390 1556 451 154\n",
      "137\t10078\t0.0\t3\t7287 2040 539 212\n",
      "138\t11665\t0.0\t3\t8713 2224 534 194\n",
      "139\t16459\t0.0\t3\t12348 3038 815 258\n",
      "140\t11044\t0.0\t3\t8418 1993 459 174\n",
      "141\t12672\t0.0\t3\t9580 2406 527 159\n",
      "142\t27036\t0.0\t3\t19928 5273 1349 486\n",
      "143\t11481\t0.0\t3\t8780 2055 473 173\n",
      "144\t13349\t0.0\t3\t10386 2288 522 153\n",
      "145\t15037\t0.0\t3\t11858 2290 632 257\n",
      "146\t18571\t0.0\t3\t14736 2957 655 223\n",
      "147\t24798\t0.0\t3\t19101 4192 1079 426\n",
      "148\t26788\t0.0\t3\t20962 4258 1084 484\n",
      "149\t9487\t0.0\t3\t7651 1343 344 149\n",
      "150\t2094\t0.0\t3\t1478 354 143 119\n",
      "151\t727193\t0.0\t3\t580279 107872 27000 12042\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_01_1.fastq.gz -p ./fastq_trimmed/Liv_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "[------=8    ] 00:38:21    98,390,000 reads @  23.6 Âµs/read;   2.54 M reads/minute"
     ]
    }
   ],
   "source": [
    "# !cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_01_1.fastq.gz -p ./fastq_trimmed/Int_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R2_001.fastq.gz\n",
    "# !cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_02_1.fastq.gz -p ./fastq_trimmed/Int_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R2_001.fastq.gz\n",
    "# !cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_03_1.fastq.gz -p ./fastq_trimmed/Int_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_04_1.fastq.gz -p ./fastq_trimmed/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_01_1.fastq.gz -p ./fastq_trimmed/Liv_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_02_1.fastq.gz -p ./fastq_trimmed/Liv_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-02_S39_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-02_S39_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_03_1.fastq.gz -p ./fastq_trimmed/Liv_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-03_S40_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-03_S40_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_04_1.fastq.gz -p ./fastq_trimmed/Liv_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-04_S41_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-04_S41_L005_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.001s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int_01_1.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int_01_2.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int_02_1.fastq.gz\n",
      "Started analysis of Int_02_2.fastq.gz\n",
      "Started analysis of Int_03_1.fastq.gz\n",
      "Started analysis of Int_03_2.fastq.gz\n",
      "Started analysis of Int_04_1.fastq.gz\n",
      "Started analysis of Int_04_2.fastq.gz\n",
      "Approx 5% complete for Int_01_1.fastq.gz\n",
      "Approx 5% complete for Int_01_2.fastq.gz\n",
      "Approx 10% complete for Int_01_1.fastq.gz\n",
      "Approx 10% complete for Int_01_2.fastq.gz\n",
      "Approx 15% complete for Int_01_2.fastq.gz\n",
      "Approx 15% complete for Int_01_1.fastq.gz\n",
      "Approx 20% complete for Int_01_2.fastq.gz\n",
      "Approx 20% complete for Int_01_1.fastq.gz\n",
      "Approx 25% complete for Int_01_2.fastq.gz\n",
      "Approx 25% complete for Int_01_1.fastq.gz\n",
      "Approx 5% complete for Int_03_1.fastq.gz\n",
      "Approx 5% complete for Int_03_2.fastq.gz\n",
      "Approx 5% complete for Int_02_1.fastq.gz\n",
      "Approx 30% complete for Int_01_2.fastq.gz\n",
      "Approx 30% complete for Int_01_1.fastq.gz\n",
      "Approx 5% complete for Int_02_2.fastq.gz\n",
      "Approx 5% complete for Int_04_2.fastq.gz\n",
      "Approx 5% complete for Int_04_1.fastq.gz\n",
      "Approx 35% complete for Int_01_2.fastq.gz\n",
      "Approx 35% complete for Int_01_1.fastq.gz\n",
      "Approx 40% complete for Int_01_2.fastq.gz\n",
      "Approx 40% complete for Int_01_1.fastq.gz\n",
      "Approx 45% complete for Int_01_2.fastq.gz\n",
      "Approx 45% complete for Int_01_1.fastq.gz\n",
      "Approx 50% complete for Int_01_2.fastq.gz\n",
      "Approx 50% complete for Int_01_1.fastq.gz\n",
      "Approx 10% complete for Int_03_1.fastq.gz\n",
      "Approx 55% complete for Int_01_2.fastq.gz\n",
      "Approx 55% complete for Int_01_1.fastq.gz\n",
      "Approx 10% complete for Int_03_2.fastq.gz\n",
      "Approx 10% complete for Int_02_1.fastq.gz\n",
      "Approx 60% complete for Int_01_2.fastq.gz\n",
      "Approx 10% complete for Int_04_2.fastq.gz\n",
      "Approx 60% complete for Int_01_1.fastq.gz\n",
      "Approx 10% complete for Int_02_2.fastq.gz\n",
      "Approx 65% complete for Int_01_2.fastq.gz\n",
      "Approx 65% complete for Int_01_1.fastq.gz\n",
      "Approx 10% complete for Int_04_1.fastq.gz\n",
      "Approx 70% complete for Int_01_2.fastq.gz\n",
      "Approx 70% complete for Int_01_1.fastq.gz\n",
      "Approx 75% complete for Int_01_2.fastq.gz\n",
      "Approx 75% complete for Int_01_1.fastq.gz\n",
      "Approx 80% complete for Int_01_2.fastq.gz\n",
      "Approx 80% complete for Int_01_1.fastq.gz\n",
      "Approx 15% complete for Int_03_1.fastq.gz\n",
      "Approx 15% complete for Int_03_2.fastq.gz\n",
      "Approx 85% complete for Int_01_2.fastq.gz\n",
      "Approx 85% complete for Int_01_1.fastq.gz\n",
      "Approx 15% complete for Int_02_1.fastq.gz\n",
      "Approx 90% complete for Int_01_2.fastq.gz\n",
      "Approx 15% complete for Int_04_2.fastq.gz\n",
      "Approx 15% complete for Int_02_2.fastq.gz\n",
      "Approx 90% complete for Int_01_1.fastq.gz\n",
      "Approx 95% complete for Int_01_2.fastq.gz\n",
      "Approx 95% complete for Int_01_1.fastq.gz\n",
      "Analysis complete for Int_01_2.fastq.gz\n",
      "Approx 15% complete for Int_04_1.fastq.gz\n",
      "Analysis complete for Int_01_1.fastq.gz\n",
      "Approx 20% complete for Int_03_1.fastq.gz\n",
      "Approx 20% complete for Int_03_2.fastq.gz\n",
      "Approx 20% complete for Int_02_1.fastq.gz\n",
      "Approx 20% complete for Int_04_2.fastq.gz\n",
      "Approx 20% complete for Int_02_2.fastq.gz\n",
      "Approx 20% complete for Int_04_1.fastq.gz\n",
      "Approx 25% complete for Int_03_1.fastq.gz\n",
      "Approx 25% complete for Int_03_2.fastq.gz\n",
      "Approx 25% complete for Int_02_1.fastq.gz\n",
      "Approx 25% complete for Int_02_2.fastq.gz\n",
      "Approx 25% complete for Int_04_2.fastq.gz\n",
      "Approx 30% complete for Int_03_1.fastq.gz\n",
      "Approx 30% complete for Int_03_2.fastq.gz\n",
      "Approx 25% complete for Int_04_1.fastq.gz\n",
      "Approx 30% complete for Int_02_1.fastq.gz\n",
      "Approx 30% complete for Int_02_2.fastq.gz\n",
      "Approx 30% complete for Int_04_2.fastq.gz\n",
      "Approx 35% complete for Int_03_1.fastq.gz\n",
      "Approx 35% complete for Int_03_2.fastq.gz\n",
      "Approx 30% complete for Int_04_1.fastq.gz\n",
      "Approx 35% complete for Int_02_1.fastq.gz\n",
      "Approx 35% complete for Int_02_2.fastq.gz\n",
      "Approx 35% complete for Int_04_2.fastq.gz\n",
      "Approx 40% complete for Int_03_1.fastq.gz\n",
      "Approx 40% complete for Int_03_2.fastq.gz\n",
      "Approx 40% complete for Int_02_1.fastq.gz\n",
      "Approx 35% complete for Int_04_1.fastq.gz\n",
      "Approx 40% complete for Int_02_2.fastq.gz\n",
      "Approx 40% complete for Int_04_2.fastq.gz\n",
      "Approx 45% complete for Int_03_1.fastq.gz\n",
      "Approx 45% complete for Int_03_2.fastq.gz\n",
      "Approx 45% complete for Int_02_1.fastq.gz\n",
      "Approx 45% complete for Int_02_2.fastq.gz\n",
      "Approx 40% complete for Int_04_1.fastq.gz\n",
      "Approx 45% complete for Int_04_2.fastq.gz\n",
      "Approx 50% complete for Int_03_1.fastq.gz\n",
      "Approx 50% complete for Int_03_2.fastq.gz\n",
      "Approx 50% complete for Int_02_1.fastq.gz\n",
      "Approx 50% complete for Int_02_2.fastq.gz\n",
      "Approx 55% complete for Int_03_1.fastq.gz\n",
      "Approx 50% complete for Int_04_2.fastq.gz\n",
      "Approx 45% complete for Int_04_1.fastq.gz\n",
      "Approx 55% complete for Int_03_2.fastq.gz\n",
      "Approx 55% complete for Int_02_1.fastq.gz\n",
      "Approx 60% complete for Int_03_1.fastq.gz\n",
      "Approx 55% complete for Int_02_2.fastq.gz\n",
      "Approx 55% complete for Int_04_2.fastq.gz\n",
      "Approx 60% complete for Int_03_2.fastq.gz\n",
      "Approx 50% complete for Int_04_1.fastq.gz\n",
      "Approx 65% complete for Int_03_1.fastq.gz\n",
      "Approx 60% complete for Int_02_1.fastq.gz\n",
      "Approx 60% complete for Int_02_2.fastq.gz\n",
      "Approx 65% complete for Int_03_2.fastq.gz\n",
      "Approx 60% complete for Int_04_2.fastq.gz\n",
      "Approx 55% complete for Int_04_1.fastq.gz\n",
      "Approx 70% complete for Int_03_1.fastq.gz\n",
      "Approx 65% complete for Int_02_1.fastq.gz\n",
      "Approx 65% complete for Int_02_2.fastq.gz\n",
      "Approx 70% complete for Int_03_2.fastq.gz\n",
      "Approx 65% complete for Int_04_2.fastq.gz\n",
      "Approx 75% complete for Int_03_1.fastq.gz\n",
      "Approx 60% complete for Int_04_1.fastq.gz\n",
      "Approx 70% complete for Int_02_1.fastq.gz\n",
      "Approx 70% complete for Int_02_2.fastq.gz\n",
      "Approx 75% complete for Int_03_2.fastq.gz\n",
      "Approx 70% complete for Int_04_2.fastq.gz\n",
      "Approx 80% complete for Int_03_1.fastq.gz\n",
      "Approx 75% complete for Int_02_1.fastq.gz\n",
      "Approx 80% complete for Int_03_2.fastq.gz\n",
      "Approx 75% complete for Int_02_2.fastq.gz\n",
      "Approx 65% complete for Int_04_1.fastq.gz\n",
      "Approx 75% complete for Int_04_2.fastq.gz\n",
      "Approx 85% complete for Int_03_1.fastq.gz\n",
      "Approx 80% complete for Int_02_1.fastq.gz\n",
      "Approx 85% complete for Int_03_2.fastq.gz\n",
      "Approx 80% complete for Int_02_2.fastq.gz\n",
      "Approx 70% complete for Int_04_1.fastq.gz\n",
      "Approx 80% complete for Int_04_2.fastq.gz\n",
      "Approx 90% complete for Int_03_1.fastq.gz\n",
      "Approx 85% complete for Int_02_1.fastq.gz\n",
      "Approx 90% complete for Int_03_2.fastq.gz\n",
      "Approx 85% complete for Int_02_2.fastq.gz\n",
      "Approx 75% complete for Int_04_1.fastq.gz\n",
      "Approx 95% complete for Int_03_1.fastq.gz\n",
      "Approx 85% complete for Int_04_2.fastq.gz\n",
      "Approx 95% complete for Int_03_2.fastq.gz\n",
      "Approx 90% complete for Int_02_1.fastq.gz\n",
      "Approx 90% complete for Int_02_2.fastq.gz\n",
      "Analysis complete for Int_03_1.fastq.gz\n",
      "Approx 90% complete for Int_04_2.fastq.gz\n",
      "Approx 80% complete for Int_04_1.fastq.gz\n",
      "Analysis complete for Int_03_2.fastq.gz\n",
      "Approx 95% complete for Int_02_1.fastq.gz\n",
      "Approx 95% complete for Int_02_2.fastq.gz\n",
      "Approx 95% complete for Int_04_2.fastq.gz\n",
      "Approx 85% complete for Int_04_1.fastq.gz\n",
      "Analysis complete for Int_02_1.fastq.gz\n",
      "Analysis complete for Int_02_2.fastq.gz\n",
      "Failed to process file Int_04_2.fastq.gz\n",
      "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: Ran out of data in the middle of a fastq entry.  Your file is probably truncated\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:187)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:129)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:77)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:833)\n",
      "Approx 90% complete for Int_04_1.fastq.gz\n",
      "Approx 95% complete for Int_04_1.fastq.gz\n",
      "Failed to process file Int_04_1.fastq.gz\n",
      "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: Unexpected end of ZLIB input stream\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:222)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:129)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:77)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:833)\n"
     ]
    }
   ],
   "source": [
    "!fastqc -t 16 fastq_trimmed/*.fastq.gz -o fastq_trimmed_qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.012s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int-01_S42_L005_R1_001.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Approx 5% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-01_S42_L005_R2_001.fastq.gz\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_qc\n",
    "!fastqc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*fastq.gz -o fastq_qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[91m///\u001b[0m \u001b]8;id=304351;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ ðŸ” \u001b[2m| v1.17\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Prepending directory to sample names\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/fastq_qc\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/fastq_trimmed_qc\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m22/22\u001b[0m  stqc.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 11 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report_all_ipynb.html   (overwritten)\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_report_all_ipynb_data   (overwritten)\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_trimmed_qc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/multiqc_report.html\n",
    "# !cp /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/multiqc_report.html multiqc_report.html\n",
    "!multiqc -f -d fastq_qc/ fastq_trimmed_qc/ -n multiqc_report_all_ipynb.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# !mkdir alignment\n",
    "# !mkdir alignment/star\n",
    "# !mkdir alignment/hisat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runThreadN 32 --runMode genomeGenerate --genomeDir alignment/star --genomeFastaFiles genome/genome.fa --sjdbGTFfile genome/annotations.gtf --sjdbOverhang 150 --genomeSAindexNbases 13\n",
      "\tSTAR version: 2.7.10b   compiled: 2022-11-01T09:53:26-04:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Dec 10 14:18:48 ..... started STAR run\n",
      "Dec 10 14:18:48 ... starting to generate Genome files\n",
      "Dec 10 14:18:54 ..... processing annotations GTF\n",
      "Dec 10 14:18:57 ... starting to sort Suffix Array. This may take a long time...\n",
      "Dec 10 14:18:59 ... sorting Suffix Array chunks and saving them to disk...\n",
      "Dec 10 14:19:42 ... loading chunks from disk, packing SA...\n",
      "Dec 10 14:19:52 ... finished generating suffix array\n",
      "Dec 10 14:19:52 ... generating Suffix Array index\n",
      "Dec 10 14:20:37 ... completed Suffix Array index\n",
      "Dec 10 14:20:38 ..... inserting junctions into the genome indices\n",
      "Dec 10 14:21:09 ... writing Genome to disk ...\n",
      "Dec 10 14:21:10 ... writing Suffix Array to disk ...\n",
      "Dec 10 14:21:14 ... writing SAindex to disk\n",
      "Dec 10 14:21:15 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "!STAR \\\n",
    "--runThreadN 32 \\\n",
    "--runMode genomeGenerate \\\n",
    "--genomeDir genome_alignment/star \\\n",
    "--genomeFastaFiles genome/genome.fa \\\n",
    "--sjdbGTFfile genome/annotations.gtf \\\n",
    "--sjdbOverhang 150\\\n",
    "--genomeSAindexNbases 13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!STAR \\\n",
    "--runThreadN 32 \\\n",
    "--runMode alignReads \\\n",
    "--genomeDir genome_alignment/star \\\n",
    "--readFilesManifest manifest.tsv \\\n",
    "--readFilesCommand zcat \\\n",
    "--outSAMtype BAM SortedByCoordinate \\\n",
    "--outSAMunmapped Within \\\n",
    "--outFileNamePrefix alignment/star/\n",
    "!cp alignment/star/Log.final.out alignment/star/firstlog.final.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndentationError",
     "evalue": "unexpected indent (845505616.py, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[9], line 2\u001b[0;36m\u001b[0m\n\u001b[0;31m    --outSAMattributes NH HI AS nM RG \\\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m unexpected indent\n"
     ]
    }
   ],
   "source": [
    "!STAR \\\n",
    "--runThreadN 32 \\\n",
    "--runMode alignReads \\\n",
    "--genomeDir genome_alignment/star \\\n",
    "--readFilesManifest manifest.tsv \\\n",
    "--readFilesCommand zcat \\\n",
    "--outSAMtype BAM SortedByCoordinate \\\n",
    "--outSAMunmapped Within \\\n",
    "--outFileNamePrefix alignment/star/\n",
    "--outSAMattributes NH HI AS nM RG \\\n",
    "--outFileNamePrefix alignment/star/ \\\n",
    "--sjdbFileChrStartEnd alignment/star/SJ.out.tab\n",
    "!cp alignment/star/Log.final.out alignment/star/second.final.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !while read -r line; do \\\n",
    "#     fq=$(echo $line | awk '{print $1}'); \\\n",
    "#     bn=$(basename \"$fq\" .fastq.gz); \\\n",
    "#     rg=$(echo $line | awk '{print $3}' | sed 's/ID://' ); \\\n",
    "#     echo \"Aligning $fq with $rg\"; \\\n",
    "#     hisat2 alignment/hisat/genome -p 16 -U $fq --rg-id $rg --rg SM:$rg --summary-file alignment/hisat/$bn.log --new-summary > alignment/hisat/$bn.sam; \\\n",
    "#     done < manifest.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !for sam in alignment/hisat/*.sam; do \\\n",
    "#     bn=$(basename \"$sam\" .sam); \\\n",
    "#     echo \"Converting $sam to $bn.bam and sorting.\"; \\\n",
    "#     samtools view -@ 32 -b $sam | samtools sort -@ 32 > alignment/hisat/$bn.bam; \\\n",
    "#     done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !samtools merge -@ 32 alignment/hisat/merged.bam  alignment/hisat/*.bam \n",
    "# !samtools index -@ 32 alignment/hisat/merged.bam "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!samtools flagstat alignment/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qualimap bamqc -nt 32 -outdir qualimap/star/bam -bam alignment/star/Aligned.sortedByCoord.out.bam --feature-file genome/annotations.gtf \n",
    "!qualimap rnaseq -outdir qualimap/star/rnaseq -bam alignment/star/Aligned.sortedByCoord.out.bam -gtf genome/annotations.gtf\n",
    "# !qualimap bamqc -nt 32 -outdir qualimap/hisat/bam -bam alignment/hisat/merged.bam --feature-file genome/annotations.gtf -p strand-specific-forward\n",
    "# !qualimap rnaseq -outdir qualimap/hisat/rnaseq -bam alignment/hisat/merged.bam -gtf genome/annotations.gtf -p strand-specific-forward\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!multiqc --force -d fastq_qc/ fastq_trimmed_qc/ alignment/ ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir logs\n",
    "!mkdir dedup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "!picard MarkDuplicates -I alignment/star -M logs/star_duplicates -O dedup/star.bam --VALIDATION_STRINGENCY SILENT\n",
    "# !picard MarkDuplicates -I alignment/hisat/merged.bam -M logs/hisat_duplicates -O dedup/hisat.bam --VALIDATION_STRINGENCY SILENT\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!featureCounts -T 32 \\\n",
    "    dedup/star.bam \\\n",
    "    -T 32 \\\n",
    "    --byReadGroup \\\n",
    "    -s 1 \\\n",
    "    --ignoreDup \\\n",
    "    -M \\\n",
    "    --fraction \\\n",
    "    -a genome/annotations.gtf \\\n",
    "    -o star_counts.tsv \\\n",
    "    --verbose\n",
    "\n",
    "\n",
    "# !featureCounts -T 32 \\\n",
    "#     dedup/hisat.bam \\\n",
    "#     -T 32 \\\n",
    "#     --byReadGroup \\\n",
    "#     -s 1 \\\n",
    "#     --ignoreDup \\\n",
    "#     -M \\\n",
    "#     --fraction \\\n",
    "#     -a genome/annotations.gtf \\\n",
    "#     -o hisat_counts.tsv \\\n",
    "#     --verbose "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!multiqc --force -d fastq_qc/ fastq_trimmed_qc/ alignment/ alignment_qc."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "in single end seq we get a single end primer that does the seq by synthesis and seq all accross the fragment and reads the fluoresence\n",
    "for paired end we add an entire second round that adds to the other end and does the reverse end of the original \n",
    "star deals with reads that overlap differently based on their number? maybe"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "final_project_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
