{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-11-21 16:29:56--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 116797085 (111M) [application/x-gzip]\n",
      "Saving to: ‘genome/genome.fa.gz’\n",
      "\n",
      "genome/genome.fa.gz 100%[===================>] 111.39M  28.0MB/s    in 4.6s    \n",
      "\n",
      "2024-11-21 16:30:01 (24.1 MB/s) - ‘genome/genome.fa.gz’ saved [116797085/116797085]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# !conda activate final_project_env\n",
    "# !mkdir genome\n",
    "# %cd genome\n",
    "!wget -nc -O genome/genome.fa.gz https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz \n",
    "!gzip -d -f genome/genome.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-11-21 16:30:04--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3228084 (3.1M) [application/x-gzip]\n",
      "Saving to: ‘STDOUT’\n",
      "\n",
      "-                   100%[===================>]   3.08M  3.82MB/s    in 0.8s    \n",
      "\n",
      "2024-11-21 16:30:06 (3.82 MB/s) - written to stdout [3228084/3228084]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget -O - https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz | \\\n",
    "    gzip -f -d > genome/annotations.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "!samtools faidx genome/genome.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘fastq_trimmed’: File exists\n"
     ]
    }
   ],
   "source": [
    "!mkdir fastq_trimmed\n",
    "# !cut  -d ' ' -f 2 /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/M007087_WheelerN_md5sum.txt | tail -n +2 > fastq_files.txt\n",
    "# !while IFS= read -r line; do \\\n",
    "\n",
    "#     done < fastq_files.txt\n",
    "#  -q 37 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int_01\n",
      "Int_01\n",
      "Int_02\n",
      "Int_02\n",
      "Int_03\n",
      "Int_03\n",
      "Int_04\n",
      "Int_04\n",
      "Liv_01\n",
      "Liv_01\n",
      "Liv_02\n",
      "Liv_02\n",
      "Liv_03\n",
      "Liv_03\n",
      "Liv_04\n",
      "Liv_04\n"
     ]
    }
   ],
   "source": [
    "for i in [\"Int\",\"Liv\"]:\n",
    "    for a in  [\"01\",\"02\",\"03\",\"04\"]:\n",
    "        print(i + \"_\" + a)\n",
    "        print(i + \"_\" + a)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/*_R1_100.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R1_100.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R2_100.fastq.gz\n",
      "When a paired-end trimming option such as -A/-G/-B/-U, is used, a second output file needs to be specified via -p (--paired-output).\n"
     ]
    }
   ],
   "source": [
    "# !for fastq in /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R1_100.fastq.gz; do \\\n",
    "#     for fastq2 in /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R2_100.fastq.gz; do \\\n",
    "#         base_name=$(basename \"$fastq\" .fastq.gz); \\\n",
    "#     cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/$base_name.fastq.gz $fastq $fastq2; \\\n",
    "# done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_01_1.fastq.gz -p ./fastq_trimmed/Int_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "[    8=------] 00:09:24    26,870,000 reads @  20.8 µs/read;   2.88 M reads/minute"
     ]
    }
   ],
   "source": [
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_01_1.fastq.gz -p ./fastq_trimmed/Int_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_02_1.fastq.gz -p ./fastq_trimmed/Int_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_03_1.fastq.gz -p ./fastq_trimmed/Int_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_04_1.fastq.gz -p ./fastq_trimmed/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_01_1.fastq.gz -p ./fastq_trimmed/Liv_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_02_1.fastq.gz -p ./fastq_trimmed/Liv_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-02_S39_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-02_S39_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_03_1.fastq.gz -p ./fastq_trimmed/Liv_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-03_S40_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-03_S40_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_04_1.fastq.gz -p ./fastq_trimmed/Liv_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-04_S41_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-04_S41_L005_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.001s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "application/gzip\n",
      "application/gzip\n",
      "Failed to process fastq_trimmed/Int-01_S42_L005_R2_001.fastq.gz\n",
      "java.io.EOFException\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:258)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:91)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:37)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.<init>(FastQFile.java:84)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:106)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:62)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.processFile(OfflineRunner.java:163)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.<init>(OfflineRunner.java:125)\n",
      "\tat uk.ac.babraham.FastQC.FastQCApplication.main(FastQCApplication.java:316)\n",
      "Started analysis of Int-01_S42_L005_R1_001.fastq.gz\n",
      "application/gzip\n",
      "Approx 5% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Failed to process file Int-01_S42_L005_R1_001.fastq.gz\n",
      "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: Corrupt GZIP trailer\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:222)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:129)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:77)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:833)\n",
      "Approx 5% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R2_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!fastqc -t 16 fastq_trimmed/*.fastq.gz -o fastq_trimmed_qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.001s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int-01_S42_L005_R1_001.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Approx 5% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_qc\n",
    "!fastqc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*fastq.gz -o fastq_qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[91m///\u001b[0m \u001b]8;id=389488;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m66/66\u001b[0m  66\u001b[0m  \n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 1 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Existing reports found, adding suffix to filenames. Use '--force' to overwrite.\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report_2.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_data_2\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_trimmed_qc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/multiqc_report.html\n",
    "# !cp /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/multiqc_report.html multiqc_report.html\n",
    "!multiqc -f -d fastq_qc/ fastq_trimmed_qc/ -n multiqc_report_all_ipynb.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.015s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "text/html\n",
      "Failed to process multiqc_report.html\n",
      "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: ID line didn't start with '@' at line 1\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:163)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.<init>(FastQFile.java:93)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:106)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:62)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.processFile(OfflineRunner.java:163)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.<init>(OfflineRunner.java:125)\n",
      "\tat uk.ac.babraham.FastQC.FastQCApplication.main(FastQCApplication.java:316)\n"
     ]
    }
   ],
   "source": [
    "# !fastqc multiqc_report.html\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                \n",
      " \u001b[91m///\u001b[0m \u001b]8;id=494690;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m                                                         \n",
      "                                                                                \n",
      " \u001b[33mUsage:\u001b[0m \u001b[1mmultiqc\u001b[0m [\u001b[1;36mOPTIONS\u001b[0m] [\u001b[1;36mANALYSIS\u001b[0m \u001b[1;36mDIRECTORY\u001b[0m]                                  \n",
      "                                                                                \n",
      " MultiQC aggregates results from bioinformatics analyses across many samples    \n",
      " into a single report.                                                          \n",
      " \u001b[2mIt searches a given directory for analysis logs and compiles a HTML report. \u001b[0m   \n",
      " \u001b[2mIt's a general use tool, perfect for summarising the output from numerous \u001b[0m     \n",
      " \u001b[2mbioinformatics tools.\u001b[0m                                                          \n",
      " \u001b[2mTo run, supply with one or more directory to scan for analysis results. For \u001b[0m   \n",
      " \u001b[2mexample, to run in the current working directory, use '\u001b[0m\u001b[1;2;34mmultiqc .\u001b[0m\u001b[2m'\u001b[0m              \n",
      "                                                                                \n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Main options \u001b[0m\u001b[2m──────────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--force\u001b[0m            \u001b[1;32m-f\u001b[0m  Overwrite any existing reports                        \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--config\u001b[0m           \u001b[1;32m-c\u001b[0m  Specific config file to load, after those in MultiQC  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        dir / home dir / working dir.                         \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(PATH)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--cl-config\u001b[0m            Specify MultiQC config YAML on the command line       \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                         \u001b[0m       \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--filename\u001b[0m         \u001b[1;32m-n\u001b[0m  Report filename. Use '\u001b[3;33mstdout\u001b[0m' to print to standard    \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        out.                                                  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--outdir\u001b[0m           \u001b[1;32m-o\u001b[0m  Create report in the specified output directory.      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                          \u001b[0m      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--ignore\u001b[0m           \u001b[1;32m-x\u001b[0m  Ignore analysis files \u001b[2;33m(GLOB EXPRESSION)\u001b[0m               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--ignore-samples\u001b[0m       Ignore sample names \u001b[2;33m(GLOB EXPRESSION)\u001b[0m                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--ignore-symlinks\u001b[0m      Ignore symlinked directories and files                \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--file-list\u001b[0m        \u001b[1;32m-l\u001b[0m  Supply a file containing a list of file paths to be   \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        searched, one per row                                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Choosing modules to run \u001b[0m\u001b[2m───────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--module\u001b[0m     \u001b[1;32m-m\u001b[0m  Use only this module. Can specify multiple times.           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                  \u001b[2;33m(MODULE NAME)                                    \u001b[0m           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--exclude\u001b[0m    \u001b[1;32m-e\u001b[0m  Do not use this module. Can specify multiple times.         \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                  \u001b[2;33m(MODULE NAME)                                      \u001b[0m         \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--tag\u001b[0m            Use only modules which tagged with this keyword \u001b[2;33m(TEXT)\u001b[0m      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--view-tags\u001b[0m      View the available tags and which modules they load         \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Sample handling \u001b[0m\u001b[2m───────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--dirs\u001b[0m           \u001b[1;32m-d\u001b[0m   Prepend directory to sample names                      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--dirs-depth\u001b[0m     \u001b[1;32m-dd\u001b[0m  Prepend \u001b[3;33mn\u001b[0m directories to sample names. Negative number \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       to take from start of path.                            \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       \u001b[2;33m(INTEGER)                                             \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--fullnames\u001b[0m      \u001b[1;32m-s\u001b[0m   Do not clean the sample names \u001b[3m(leave as full file \u001b[0m     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       \u001b[3mname)\u001b[0m                                                  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--fn_as_s_name\u001b[0m        Use the log filename as the sample name                \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--replace-names\u001b[0m       TSV file to rename sample names during report          \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       generation                                             \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       \u001b[2;33m(PATH)                                                \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Report customisation \u001b[0m\u001b[2m──────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--title\u001b[0m            \u001b[1;32m-i\u001b[0m  Report title. Printed as page header, used for        \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        filename if not otherwise specified.                  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--comment\u001b[0m          \u001b[1;32m-b\u001b[0m  Custom comment, will be printed at the top of the     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        report.                                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--template\u001b[0m         \u001b[1;32m-t\u001b[0m  Report template to use.                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(default|default_dev|gathered|geo|sections|simple)\u001b[0m    \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--sample-names\u001b[0m         TSV file containing alternative sample names for      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        renaming buttons in the report                        \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(PATH)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--sample-filters\u001b[0m       TSV file containing show/hide patterns for the report \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(PATH)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--custom-css-file\u001b[0m      Custom CSS file to add to the final report \u001b[2;33m(PATH)\u001b[0m     \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Output files \u001b[0m\u001b[2m──────────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--flat\u001b[0m          \u001b[1;32m-fp\u001b[0m  Use only flat plots \u001b[3m(static images)\u001b[0m                     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--interactive\u001b[0m   \u001b[1;32m-ip\u001b[0m  Use only interactive plots \u001b[3m(in-browser Javascript)\u001b[0m      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--export\u001b[0m        \u001b[1;32m-p\u001b[0m   Export plots as static images in addition to the report \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--data-dir\u001b[0m           Force the parsed data directory to be created.          \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-data-dir\u001b[0m        Prevent the parsed data directory from being created.   \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--data-format\u001b[0m   \u001b[1;32m-k\u001b[0m   Output parsed data in a different format.               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                      \u001b[2;33m(tsv|json|yaml)                          \u001b[0m               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--zip-data-dir\u001b[0m  \u001b[1;32m-z\u001b[0m   Compress the data directory.                            \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-report\u001b[0m          Do not generate a report, only export data and plots    \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--pdf\u001b[0m                Creates PDF report with the \u001b[3m'simple'\u001b[0m template. Requires \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                      \u001b]8;id=183873;https://pandoc.org/\u001b\\Pandoc\u001b]8;;\u001b\\ to be installed.                                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m MultiQC behaviour \u001b[0m\u001b[2m─────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--verbose\u001b[0m           \u001b[1;32m-v\u001b[0m  Increase output verbosity. \u001b[2;33m(INTEGER RANGE)\u001b[0m           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--quiet\u001b[0m             \u001b[1;32m-q\u001b[0m  Only show log warnings                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--strict\u001b[0m                Don't catch exceptions, run additional code checks   \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         to help development.                                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--require-logs\u001b[0m          Require all explicitly requested modules to have log \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         files. If not, MultiQC will exit with an error.      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--profile-runtime\u001b[0m       Add analysis of how long MultiQC takes to run to the \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         report                                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-megaqc-upload\u001b[0m      Don't upload generated report to MegaQC, even if     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         MegaQC options are found                             \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-ansi\u001b[0m               Disable coloured log output                          \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--version\u001b[0m               Show the version and exit.                           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--help\u001b[0m              \u001b[1;32m-h\u001b[0m  Show this message and exit.                          \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "                                                                                \n",
      " See \u001b]8;id=176947;http://multiqc.info\u001b\\http://multiqc.info\u001b]8;;\u001b\\ for more details.                                      \n",
      "\n"
     ]
    }
   ],
   "source": [
    "!multiqc --help\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Step 1: Review Metadata and QC Data\n",
    "1.1 Review QC data from the qc/ folder:\n",
    "Check the quality of the RNA extraction, library prep, and sequencing process. These details will inform decisions about trimming and filtering reads.\n",
    "The QC reports may include metrics such as sequencing depth, base composition, GC content, and read length distributions, which can guide trimming and filtering steps.\n",
    "1.2 Read the reference paper:\n",
    "Understanding the study in the paper \"Winners vs. losers: Schistosoma mansoni intestinal and liver eggs exhibit striking differences in gene expression and immunogenicity\" will give you insights into the expected differences between liver and intestine miracidia, helping you interpret the results later.\n",
    "##Step 2: Read Preprocessing (Trimming and Quality Control)\n",
    "2.1 Assess the raw reads:\n",
    "Use tools like FastQC to assess the raw sequencing data. Create a MultiQC report to summarize the results for all samples. This will help identify issues like adapter contamination, GC bias, or low-quality regions.\n",
    "2.2 Trim low-quality bases and adapters:\n",
    "Use Trimmomatic or Cutadapt to remove adapters and trim low-quality bases (e.g., Q-score below 20). Focus on trimming the 3' and 5' ends of the reads if necessary.\n",
    "After trimming, run FastQC again and generate a new MultiQC report to ensure quality has improved.\n",
    "2.3 Verify trimming results:\n",
    "Check that the trimmed reads have good quality, proper length distributions, and no obvious contamination. If necessary, adjust the trimming parameters.\n",
    "##Step 3: Read Alignment to the Reference Genome\n",
    "3.1 Align trimmed reads:\n",
    "Use STAR or HISAT2 for aligning the reads to the Schistosoma mansoni genome. The choice of aligner will depend on the quality of your reference genome.\n",
    "Align the reads in a strand-specific manner since the NEBNext Ultra II Directional mRNA kit was used. This will ensure that the strand information is preserved for downstream analyses.\n",
    "3.2 Assess alignment quality:\n",
    "Use SAMtools to check the alignment quality. This will include assessing mapping rate, proper pairs, and other metrics.\n",
    "You can use Picard tools to generate alignment statistics and visualize any potential issues.\n",
    "##Step 4: Counting Reads\n",
    "4.1 Generate gene counts:\n",
    "Use featureCounts (from the Subread package) or HTSeq-count to generate a gene count matrix based on the aligned BAM files. Ensure you use a GTF file for Schistosoma mansoni gene annotations.\n",
    "The output should be a count matrix, with genes in rows and samples in columns.\n",
    "4.2 Inspect counts:\n",
    "Visualize the count distribution across samples using boxplots or histograms to ensure that the counts look reasonable.\n",
    "##Step 5: PCA Analysis\n",
    "5.1 Normalize the counts:\n",
    "Normalize the gene count data to account for sequencing depth and gene length. Use DESeq2 or edgeR for this purpose.\n",
    "Normalize to obtain variance-stabilized counts or log-transformed counts for downstream analyses.\n",
    "5.2 Perform PCA:\n",
    "Use DESeq2 or prcomp (in R or Python) to perform Principal Component Analysis (PCA) on the normalized count data.\n",
    "Visualize the PCA results with a ggplot2 scatter plot, coloring by sample origin (liver vs. intestine). Look for clustering patterns between the two groups.\n",
    "##Step 6: Differential Expression Analysis\n",
    "6.1 Perform differential expression analysis:\n",
    "Use DESeq2 to identify differentially expressed genes (DEGs) between the liver and intestine samples.\n",
    "Set up a contrast in DESeq2 to compare liver versus intestine groups, and run the differential expression pipeline.\n",
    "6.2 Visualize DEGs:\n",
    "Use a volcano plot to visualize the DEGs. Plot log-fold changes (logFC) on the x-axis and the significance (–log10 p-value) on the y-axis. Highlight the most significant DEGs.\n",
    "Consider additional visualizations like MA plots or heatmaps of DEGs, especially genes with high or low expression.\n",
    "##Step 7: Additional Requirements\n",
    "7.1 Create a Conda Environment:\n",
    "Provide an environment.yaml file that includes the required packages for the analysis (e.g., DESeq2, FastQC, Trimmomatic, STAR, featureCounts, MultiQC, etc.).\n",
    "Ensure the environment is reproducible, and list all dependencies.\n",
    "7.2 Version Control with Git:\n",
    "Initialize a Git repository for version control.\n",
    "Make commits at regular intervals (at least one per week). Write informative commit messages and organize the project structure logically.\n",
    "7.3 Create a Snakemake Pipeline:\n",
    "Write a Snakefile to automate the analysis from raw reads to differential expression analysis. The pipeline should handle tasks like trimming, alignment, counting, and DESeq2 analysis in a reproducible manner.\n",
    "Specify the necessary Conda environments using --use-conda to ensure reproducibility.\n",
    "##Step 8: Report and Documentation\n",
    "8.1 Write an IPython Notebook:\n",
    "Create a well-documented and organized IPython notebook to explain the steps of the analysis.\n",
    "Include sections for each step of the process, with markdown cells explaining the rationale behind each step and what you are looking for in the output logs and plots.\n",
    "Include the necessary plots (PCA and volcano plot) and interpret the results.\n",
    "8.2 Include a MultiQC Report:\n",
    "Generate a MultiQC report that aggregates the quality control results from all stages of the analysis (raw reads, trimmed reads, alignment, counting, and PCA). This will help assess the overall quality of the data at each stage.\n",
    "Final Deliverables:\n",
    "IPython Notebook with full analysis and interpretations.\n",
    "MultiQC Report with QC information.\n",
    "environment.yaml file for Conda environment setup.\n",
    "Gene count matrix.\n",
    "Snakefile for reproducible analysis pipeline (if aiming for an A grade).\n",
    "Git repository with version-controlled code (for B or higher).\n",
    "Proper documentation and clear presentation of results.\n",
    "By following this structure, you will meet the grade specifications and ensure a reproducible and comprehensive RNA-seq analysis pipeline.\n",
    "\n",
    "\n",
    "\n",
    "You said:\n",
    "what does it mean by include YAML that uses --use-conda\n",
    "ChatGPT said:\n",
    "ChatGPT\n",
    "In this context, the requirement to \"include YAML that uses --use-conda\" refers to specifying the dependencies for your analysis pipeline in a YAML file (specifically for Snakemake), which ensures that the necessary software and tools can be installed and used within a conda environment.\n",
    "\n",
    "When you're using Snakemake to automate your analysis, you can use the --use-conda option to let Snakemake automatically create and manage conda environments for each of the steps in the workflow. The YAML file, known as the environment file, defines the conda environment's dependencies (i.e., the software packages needed for each step).\n",
    "\n",
    "Here’s a breakdown:\n",
    "\n",
    "1. YAML File (Environment Specification)\n",
    "This YAML file lists all the conda dependencies required for the analysis. You can create an environment YAML file for each tool or step (e.g., FastQC, DESeq2, featureCounts, STAR), and Snakemake will automatically set up the appropriate environments.\n",
    "\n",
    "Example of a simple environment.yaml file:\n",
    "\n",
    "yaml\n",
    "Copy code\n",
    "name: rnaseq_analysis\n",
    "channels:\n",
    "  - bioconda\n",
    "  - conda-forge\n",
    "  - defaults\n",
    "dependencies:\n",
    "  - python=3.8\n",
    "  - fastqc\n",
    "  - star\n",
    "  - samtools\n",
    "  - deeq2\n",
    "  - multiqc\n",
    "  - r-base\n",
    "  - r-ggplot2\n",
    "2. Using --use-conda in Snakemake\n",
    "When running Snakemake, you specify --use-conda to tell Snakemake to automatically use conda environments for each rule. Snakemake will look for an environment.yaml file in the same directory or specified for each individual step of your workflow and create the environment from the dependencies listed in that file.\n",
    "\n",
    "Example Snakemake command:\n",
    "\n",
    "bash\n",
    "Copy code\n",
    "snakemake --use-conda\n",
    "This command will make sure that Snakemake uses conda environments where necessary to run the steps in your pipeline.\n",
    "\n",
    "3. Snakemake Rules with Environment YAMLs\n",
    "For each step of your pipeline, you can link to a specific environment YAML file by using the conda directive in your Snakemake rule. For example:\n",
    "\n",
    "python\n",
    "Copy code\n",
    "rule fastqc:\n",
    "    input:\n",
    "        \"raw_data/{sample}.fastq\"\n",
    "    output:\n",
    "        \"qc/{sample}_fastqc.html\"\n",
    "    conda:\n",
    "        \"envs/fastqc_env.yaml\"  # this is your conda environment file for FastQC\n",
    "    shell:\n",
    "        \"fastqc {input} -o {output}\"\n",
    "Why This is Useful:\n",
    "Reproducibility: By specifying the environment and the exact versions of dependencies, it ensures that the analysis can be run again with the same software versions, even on a different machine.\n",
    "Dependency Management: Automatically manages the installation of all required software packages for each step in the workflow without worrying about potential version conflicts.\n",
    "In summary, to meet the requirement of \"include YAML that uses --use-conda,\" you would:\n",
    "\n",
    "Create an environment.yaml file listing the necessary software dependencies.\n",
    "Use Snakemake with the --use-conda option to automatically create conda environments and ensure reproducibility."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "in single end seq we get a single end primer that does the seq by synthesis and seq all accross the fragment and reads the fluoresence\n",
    "for paired end we add an entire second round that adds to the other end and does the reverse end of the original \n",
    "star deals with reads that overlap differently based on their number? maybe"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "final_project_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
