{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-12-05 14:05:07--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 116797085 (111M) [application/x-gzip]\n",
      "Saving to: ‘genome/genome.fa.gz’\n",
      "\n",
      "genome/genome.fa.gz 100%[===================>] 111.39M   590KB/s    in 3m 14s  \n",
      "\n",
      "2024-12-05 14:08:22 (589 KB/s) - ‘genome/genome.fa.gz’ saved [116797085/116797085]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# !conda activate final_project_env\n",
    "# !mkdir genome\n",
    "# %cd genome\n",
    "# !wget -nc -O genome/genome.fa.gz https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz \n",
    "# !gzip -d -f genome/genome.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-11-21 16:30:04--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3228084 (3.1M) [application/x-gzip]\n",
      "Saving to: ‘STDOUT’\n",
      "\n",
      "-                   100%[===================>]   3.08M  3.82MB/s    in 0.8s    \n",
      "\n",
      "2024-11-21 16:30:06 (3.82 MB/s) - written to stdout [3228084/3228084]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# !wget -O - https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz | \\\n",
    "#     gzip -f -d > genome/annotations.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !samtools faidx genome/genome.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘fastq_trimmed’: File exists\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_trimmed\n",
    "# # !cut  -d ' ' -f 2 /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/M007087_WheelerN_md5sum.txt | tail -n +2 > fastq_files.txt\n",
    "# # !while IFS= read -r line; do \\\n",
    "#  \n",
    "# #     done < fastq_files.txt\n",
    "# #  -q 37 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int_01\n",
      "Int_01\n",
      "Int_02\n",
      "Int_02\n",
      "Int_03\n",
      "Int_03\n",
      "Int_04\n",
      "Int_04\n",
      "Liv_01\n",
      "Liv_01\n",
      "Liv_02\n",
      "Liv_02\n",
      "Liv_03\n",
      "Liv_03\n",
      "Liv_04\n",
      "Liv_04\n"
     ]
    }
   ],
   "source": [
    "for i in [\"Int\",\"Liv\"]:\n",
    "    for a in  [\"01\",\"02\",\"03\",\"04\"]:\n",
    "        print(i + \"_\" + a)\n",
    "        print(i + \"_\" + a)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/*_R1_100.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R1_100.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R2_100.fastq.gz\n",
      "When a paired-end trimming option such as -A/-G/-B/-U, is used, a second output file needs to be specified via -p (--paired-output).\n"
     ]
    }
   ],
   "source": [
    "# !for fastq in /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R1_100.fastq.gz; do \\\n",
    "#     for fastq2 in /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*_R2_100.fastq.gz; do \\\n",
    "#         base_name=$(basename \"$fastq\" .fastq.gz); \\\n",
    "#     cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/$base_name.fastq.gz $fastq $fastq2; \\\n",
    "# done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_02_1.fastq.gz -p ./fastq_trimmed/Int_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:57:03   157,425,190 reads @  21.7 µs/read;   2.76 M reads/minute\n",
      "Finished in 3423.187 s (21.745 µs/read; 2.76 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:        157,425,190\n",
      "  Read 1 with adapter:              35,438,851 (22.5%)\n",
      "  Read 2 with adapter:              29,708,115 (18.9%)\n",
      "Pairs written (passing filters):   157,425,190 (100.0%)\n",
      "\n",
      "Total basepairs processed: 47,542,407,380 bp\n",
      "  Read 1: 23,771,203,690 bp\n",
      "  Read 2: 23,771,203,690 bp\n",
      "Total written (filtered):  45,007,312,258 bp (94.7%)\n",
      "  Read 1: 22,073,471,703 bp\n",
      "  Read 2: 22,933,840,555 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 35438851 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 11.5%\n",
      "  C: 26.1%\n",
      "  G: 29.3%\n",
      "  T: 15.5%\n",
      "  none/other: 17.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2346858\t2459768.6\t0\t2346858\n",
      "4\t1102264\t614942.1\t0\t1102264\n",
      "5\t759888\t153735.5\t0\t759888\n",
      "6\t653518\t38433.9\t0\t653518\n",
      "7\t630163\t9608.5\t0\t630163\n",
      "8\t614556\t2402.1\t0\t614556\n",
      "9\t623915\t600.5\t0\t619633 4282\n",
      "10\t626085\t150.1\t1\t605786 20299\n",
      "11\t630366\t37.5\t1\t614517 15849\n",
      "12\t629408\t9.4\t1\t612312 17096\n",
      "13\t612925\t2.3\t1\t595566 17359\n",
      "14\t616061\t0.6\t1\t597647 18414\n",
      "15\t604629\t0.1\t1\t585165 19464\n",
      "16\t601231\t0.0\t1\t579706 21525\n",
      "17\t594567\t0.0\t1\t570416 24151\n",
      "18\t586650\t0.0\t1\t563135 23295 220\n",
      "19\t580140\t0.0\t1\t555173 24062 905\n",
      "20\t589627\t0.0\t2\t560759 25854 3014\n",
      "21\t581947\t0.0\t2\t552842 26022 3083\n",
      "22\t578941\t0.0\t2\t549071 26745 3125\n",
      "23\t572032\t0.0\t2\t541954 27013 3065\n",
      "24\t566139\t0.0\t2\t534979 27752 3408\n",
      "25\t545725\t0.0\t2\t515404 27086 3235\n",
      "26\t532909\t0.0\t2\t499986 29443 3480\n",
      "27\t525113\t0.0\t2\t494057 27653 3373 30\n",
      "28\t515123\t0.0\t2\t481978 29666 3416 63\n",
      "29\t523957\t0.0\t2\t488735 31304 3769 149\n",
      "30\t504268\t0.0\t3\t469084 30347 3600 1237\n",
      "31\t500499\t0.0\t3\t465281 30456 3474 1288\n",
      "32\t488326\t0.0\t3\t453724 29749 3646 1207\n",
      "33\t482819\t0.0\t3\t446485 31080 3846 1408\n",
      "34\t470526\t0.0\t3\t437105 28696 3453 1272\n",
      "35\t453291\t0.0\t3\t421519 27271 3294 1207\n",
      "36\t431730\t0.0\t3\t401191 26175 3170 1194\n",
      "37\t405544\t0.0\t3\t377528 24109 2867 1040\n",
      "38\t397602\t0.0\t3\t369469 24096 2940 1097\n",
      "39\t391034\t0.0\t3\t364349 22851 2835 999\n",
      "40\t381606\t0.0\t3\t355584 22279 2730 1013\n",
      "41\t368591\t0.0\t3\t343616 21289 2705 981\n",
      "42\t358597\t0.0\t3\t334475 20705 2446 971\n",
      "43\t339111\t0.0\t3\t316155 19917 2222 817\n",
      "44\t327434\t0.0\t3\t304648 19668 2268 850\n",
      "45\t309886\t0.0\t3\t288657 18328 2082 819\n",
      "46\t294964\t0.0\t3\t274391 17589 2118 866\n",
      "47\t276034\t0.0\t3\t256651 16708 1942 733\n",
      "48\t254909\t0.0\t3\t237239 15225 1797 648\n",
      "49\t241067\t0.0\t3\t224416 14356 1655 640\n",
      "50\t225696\t0.0\t3\t210471 13235 1459 531\n",
      "51\t217411\t0.0\t3\t202687 12766 1446 512\n",
      "52\t207812\t0.0\t3\t193863 12020 1433 496\n",
      "53\t197201\t0.0\t3\t184059 11428 1245 469\n",
      "54\t185784\t0.0\t3\t173434 10747 1164 439\n",
      "55\t173664\t0.0\t3\t162083 10010 1186 385\n",
      "56\t158802\t0.0\t3\t148300 9126 1004 372\n",
      "57\t146830\t0.0\t3\t137190 8442 890 308\n",
      "58\t133811\t0.0\t3\t124941 7708 829 333\n",
      "59\t120969\t0.0\t3\t112992 6969 723 285\n",
      "60\t111739\t0.0\t3\t104344 6468 669 258\n",
      "61\t99816\t0.0\t3\t93211 5778 622 205\n",
      "62\t91590\t0.0\t3\t85497 5270 621 202\n",
      "63\t85162\t0.0\t3\t79649 4801 508 204\n",
      "64\t79099\t0.0\t3\t73643 4732 540 184\n",
      "65\t73772\t0.0\t3\t68811 4341 454 166\n",
      "66\t66533\t0.0\t3\t62014 3952 433 134\n",
      "67\t58815\t0.0\t3\t54843 3485 367 120\n",
      "68\t53489\t0.0\t3\t49855 3162 338 134\n",
      "69\t46676\t0.0\t3\t43493 2788 288 107\n",
      "70\t42688\t0.0\t3\t39841 2491 255 101\n",
      "71\t38808\t0.0\t3\t36132 2332 261 83\n",
      "72\t35391\t0.0\t3\t32831 2225 244 91\n",
      "73\t31623\t0.0\t3\t29538 1805 188 92\n",
      "74\t29151\t0.0\t3\t27083 1772 219 77\n",
      "75\t27878\t0.0\t3\t25915 1741 157 65\n",
      "76\t24584\t0.0\t3\t22842 1512 172 58\n",
      "77\t21768\t0.0\t3\t20294 1277 137 60\n",
      "78\t20218\t0.0\t3\t18794 1216 142 66\n",
      "79\t17983\t0.0\t3\t16743 1063 123 54\n",
      "80\t17978\t0.0\t3\t16741 1103 95 39\n",
      "81\t14594\t0.0\t3\t13609 852 93 40\n",
      "82\t12994\t0.0\t3\t12075 797 89 33\n",
      "83\t11522\t0.0\t3\t10676 726 76 44\n",
      "84\t10725\t0.0\t3\t9969 628 86 42\n",
      "85\t9137\t0.0\t3\t8457 568 83 29\n",
      "86\t9199\t0.0\t3\t8527 557 71 44\n",
      "87\t8547\t0.0\t3\t7895 541 71 40\n",
      "88\t7475\t0.0\t3\t6956 442 62 15\n",
      "89\t7552\t0.0\t3\t6987 480 58 27\n",
      "90\t8349\t0.0\t3\t7711 539 64 35\n",
      "91\t7858\t0.0\t3\t7254 507 68 29\n",
      "92\t5896\t0.0\t3\t5438 371 60 27\n",
      "93\t5647\t0.0\t3\t5174 385 59 29\n",
      "94\t4730\t0.0\t3\t4317 340 51 22\n",
      "95\t4319\t0.0\t3\t3987 262 42 28\n",
      "96\t4039\t0.0\t3\t3718 242 54 25\n",
      "97\t4337\t0.0\t3\t4000 258 52 27\n",
      "98\t4632\t0.0\t3\t4244 297 54 37\n",
      "99\t5387\t0.0\t3\t4978 315 62 32\n",
      "100\t4057\t0.0\t3\t3741 248 45 23\n",
      "101\t3568\t0.0\t3\t3273 230 38 27\n",
      "102\t3006\t0.0\t3\t2698 235 41 32\n",
      "103\t2658\t0.0\t3\t2382 214 34 28\n",
      "104\t2467\t0.0\t3\t2219 173 41 34\n",
      "105\t2458\t0.0\t3\t2234 163 48 13\n",
      "106\t2471\t0.0\t3\t2219 189 33 30\n",
      "107\t2612\t0.0\t3\t2375 197 25 15\n",
      "108\t2360\t0.0\t3\t2116 181 40 23\n",
      "109\t2126\t0.0\t3\t1929 167 20 10\n",
      "110\t2000\t0.0\t3\t1775 168 29 28\n",
      "111\t1774\t0.0\t3\t1605 137 21 11\n",
      "112\t1911\t0.0\t3\t1737 139 24 11\n",
      "113\t1826\t0.0\t3\t1627 155 34 10\n",
      "114\t1705\t0.0\t3\t1554 121 17 13\n",
      "115\t1697\t0.0\t3\t1538 119 22 18\n",
      "116\t1575\t0.0\t3\t1397 135 35 8\n",
      "117\t1619\t0.0\t3\t1476 112 22 9\n",
      "118\t1689\t0.0\t3\t1520 132 24 13\n",
      "119\t2418\t0.0\t3\t2163 208 27 20\n",
      "120\t1932\t0.0\t3\t1756 145 24 7\n",
      "121\t1720\t0.0\t3\t1593 107 13 7\n",
      "122\t1985\t0.0\t3\t1809 149 16 11\n",
      "123\t1657\t0.0\t3\t1491 134 21 11\n",
      "124\t2495\t0.0\t3\t2251 196 34 14\n",
      "125\t2248\t0.0\t3\t2027 189 19 13\n",
      "126\t1846\t0.0\t3\t1675 141 21 9\n",
      "127\t1631\t0.0\t3\t1479 125 26 1\n",
      "128\t1662\t0.0\t3\t1512 124 20 6\n",
      "129\t1154\t0.0\t3\t1024 108 10 12\n",
      "130\t1234\t0.0\t3\t1105 114 9 6\n",
      "131\t1585\t0.0\t3\t1416 144 19 6\n",
      "132\t1550\t0.0\t3\t1424 104 18 4\n",
      "133\t2418\t0.0\t3\t2219 175 19 5\n",
      "134\t1475\t0.0\t3\t1363 102 8 2\n",
      "135\t2188\t0.0\t3\t1999 164 20 5\n",
      "136\t2162\t0.0\t3\t1973 175 9 5\n",
      "137\t2490\t0.0\t3\t2282 177 22 9\n",
      "138\t3186\t0.0\t3\t2948 212 21 5\n",
      "139\t7816\t0.0\t3\t6822 916 65 13\n",
      "140\t3561\t0.0\t3\t3256 284 16 5\n",
      "141\t3217\t0.0\t3\t2973 230 12 2\n",
      "142\t14387\t0.0\t3\t13286 999 88 14\n",
      "143\t3551\t0.0\t3\t3307 223 18 3\n",
      "144\t3908\t0.0\t3\t3629 254 21 4\n",
      "145\t4376\t0.0\t3\t4102 246 22 6\n",
      "146\t4185\t0.0\t3\t3917 237 23 8\n",
      "147\t6853\t0.0\t3\t6352 463 27 11\n",
      "148\t5773\t0.0\t3\t5434 305 29 5\n",
      "149\t2532\t0.0\t3\t2245 244 32 11\n",
      "150\t50892\t0.0\t3\t969 39719 9673 531\n",
      "151\t6232930\t0.0\t3\t651011 5202172 346579 33168\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 29708115 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.4%\n",
      "  C: 29.5%\n",
      "  G: 41.5%\n",
      "  T: 10.3%\n",
      "  none/other: 2.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2598300\t2459768.6\t0\t2598300\n",
      "4\t1116369\t614942.1\t0\t1116369\n",
      "5\t779827\t153735.5\t0\t779827\n",
      "6\t659827\t38433.9\t0\t659827\n",
      "7\t628238\t9608.5\t0\t628238\n",
      "8\t617438\t2402.1\t0\t617438\n",
      "9\t627680\t600.5\t0\t620045 7635\n",
      "10\t623811\t150.1\t1\t607627 16184\n",
      "11\t629461\t37.5\t1\t614887 14574\n",
      "12\t627636\t9.4\t1\t612829 14807\n",
      "13\t610158\t2.3\t1\t593086 17072\n",
      "14\t613444\t0.6\t1\t597432 16012\n",
      "15\t602367\t0.1\t1\t585801 16566\n",
      "16\t599170\t0.0\t1\t581094 18076\n",
      "17\t592383\t0.0\t1\t572293 20090\n",
      "18\t585204\t0.0\t1\t564596 20405 203\n",
      "19\t577997\t0.0\t1\t554982 22087 928\n",
      "20\t586595\t0.0\t2\t523618 59022 3955\n",
      "21\t577429\t0.0\t2\t392889 168793 15747\n",
      "22\t574827\t0.0\t2\t456731 98323 19773\n",
      "23\t567529\t0.0\t2\t461442 89443 16644\n",
      "24\t561843\t0.0\t2\t461177 88763 11903\n",
      "25\t535738\t0.0\t2\t408178 102612 24948\n",
      "26\t520969\t0.0\t2\t407547 90670 22752\n",
      "27\t514557\t0.0\t2\t396840 93784 23898 35\n",
      "28\t501525\t0.0\t2\t234147 220334 46926 118\n",
      "29\t510305\t0.0\t2\t354743 122617 31567 1378\n",
      "30\t493502\t0.0\t3\t330649 116349 35171 11333\n",
      "31\t484247\t0.0\t3\t314382 115522 37528 16815\n",
      "32\t470080\t0.0\t3\t295074 117956 39667 17383\n",
      "33\t468573\t0.0\t3\t311987 109326 32818 14442\n",
      "34\t456462\t0.0\t3\t302320 106504 33531 14107\n",
      "35\t440921\t0.0\t3\t292125 103888 31849 13059\n",
      "36\t411883\t0.0\t3\t266291 98208 32324 15060\n",
      "37\t385817\t0.0\t3\t263727 86303 24866 10921\n",
      "38\t384892\t0.0\t3\t249155 92699 30253 12785\n",
      "39\t376938\t0.0\t3\t244904 90198 28823 13013\n",
      "40\t370116\t0.0\t3\t243326 87831 27115 11844\n",
      "41\t343467\t0.0\t3\t229788 78163 23613 11903\n",
      "42\t350299\t0.0\t3\t233765 83295 23937 9302\n",
      "43\t327684\t0.0\t3\t215430 76965 24296 10993\n",
      "44\t316830\t0.0\t3\t214022 73124 20940 8744\n",
      "45\t301865\t0.0\t3\t207516 68002 18885 7462\n",
      "46\t288390\t0.0\t3\t193297 66951 20184 7958\n",
      "47\t269276\t0.0\t3\t177522 63818 19760 8176\n",
      "48\t249012\t0.0\t3\t170791 56122 15968 6131\n",
      "49\t235619\t0.0\t3\t157379 55850 16126 6264\n",
      "50\t217438\t0.0\t3\t146693 49467 14607 6671\n",
      "51\t213030\t0.0\t3\t145033 48336 14134 5527\n",
      "52\t202160\t0.0\t3\t140212 44447 12343 5158\n",
      "53\t190859\t0.0\t3\t130160 42776 12532 5391\n",
      "54\t181811\t0.0\t3\t126353 40459 10800 4199\n",
      "55\t168221\t0.0\t3\t110487 40215 12171 5348\n",
      "56\t154568\t0.0\t3\t106709 33838 9781 4240\n",
      "57\t142460\t0.0\t3\t100737 29874 8357 3492\n",
      "58\t130175\t0.0\t3\t88902 29465 8356 3452\n",
      "59\t117539\t0.0\t3\t79449 26770 7980 3340\n",
      "60\t108147\t0.0\t3\t74052 24517 6769 2809\n",
      "61\t97305\t0.0\t3\t67862 21142 5809 2492\n",
      "62\t89611\t0.0\t3\t61172 20297 5742 2400\n",
      "63\t82857\t0.0\t3\t58161 18015 4698 1983\n",
      "64\t77666\t0.0\t3\t55854 16067 4198 1547\n",
      "65\t71910\t0.0\t3\t51510 14839 3922 1639\n",
      "66\t64621\t0.0\t3\t45712 13770 3624 1515\n",
      "67\t57579\t0.0\t3\t40252 12671 3281 1375\n",
      "68\t52507\t0.0\t3\t35969 11936 3300 1302\n",
      "69\t45613\t0.0\t3\t31694 9911 2759 1249\n",
      "70\t42037\t0.0\t3\t29563 9107 2460 907\n",
      "71\t37503\t0.0\t3\t25857 8259 2365 1022\n",
      "72\t34797\t0.0\t3\t23865 7994 2104 834\n",
      "73\t30796\t0.0\t3\t20969 7025 1965 837\n",
      "74\t28394\t0.0\t3\t19624 6317 1740 713\n",
      "75\t27211\t0.0\t3\t18914 6030 1565 702\n",
      "76\t24099\t0.0\t3\t17008 4981 1497 613\n",
      "77\t21201\t0.0\t3\t14890 4566 1234 511\n",
      "78\t19790\t0.0\t3\t13659 4345 1252 534\n",
      "79\t17525\t0.0\t3\t12222 3765 1082 456\n",
      "80\t17531\t0.0\t3\t12458 3647 988 438\n",
      "81\t14170\t0.0\t3\t9786 3075 892 417\n",
      "82\t12714\t0.0\t3\t8511 2927 857 419\n",
      "83\t11280\t0.0\t3\t7891 2407 703 279\n",
      "84\t10427\t0.0\t3\t7418 2184 569 256\n",
      "85\t8957\t0.0\t3\t6249 1922 551 235\n",
      "86\t9056\t0.0\t3\t6446 1932 470 208\n",
      "87\t8365\t0.0\t3\t5903 1800 447 215\n",
      "88\t7262\t0.0\t3\t5081 1552 440 189\n",
      "89\t7369\t0.0\t3\t5364 1441 388 176\n",
      "90\t8206\t0.0\t3\t5812 1712 470 212\n",
      "91\t7727\t0.0\t3\t5284 1717 481 245\n",
      "92\t5769\t0.0\t3\t4059 1219 353 138\n",
      "93\t5532\t0.0\t3\t3968 1097 318 149\n",
      "94\t4561\t0.0\t3\t3204 934 284 139\n",
      "95\t4254\t0.0\t3\t3111 796 241 106\n",
      "96\t3872\t0.0\t3\t2711 770 265 126\n",
      "97\t4177\t0.0\t3\t2961 841 252 123\n",
      "98\t4382\t0.0\t3\t3177 833 246 126\n",
      "99\t5235\t0.0\t3\t3768 1031 280 156\n",
      "100\t3889\t0.0\t3\t2910 670 205 104\n",
      "101\t3459\t0.0\t3\t2568 611 185 95\n",
      "102\t2895\t0.0\t3\t2068 551 178 98\n",
      "103\t2554\t0.0\t3\t1847 483 131 93\n",
      "104\t2299\t0.0\t3\t1633 455 130 81\n",
      "105\t2294\t0.0\t3\t1659 421 146 68\n",
      "106\t2357\t0.0\t3\t1693 441 154 69\n",
      "107\t2540\t0.0\t3\t1830 467 156 87\n",
      "108\t2226\t0.0\t3\t1563 454 144 65\n",
      "109\t2033\t0.0\t3\t1408 421 143 61\n",
      "110\t1927\t0.0\t3\t1302 404 124 97\n",
      "111\t1753\t0.0\t3\t1249 326 124 54\n",
      "112\t1879\t0.0\t3\t1322 398 99 60\n",
      "113\t1774\t0.0\t3\t1209 376 125 64\n",
      "114\t1673\t0.0\t3\t1153 342 126 52\n",
      "115\t1675\t0.0\t3\t1115 349 124 87\n",
      "116\t1558\t0.0\t3\t1065 316 106 71\n",
      "117\t1599\t0.0\t3\t1088 359 106 46\n",
      "118\t1610\t0.0\t3\t1119 347 96 48\n",
      "119\t1699\t0.0\t3\t1110 385 128 76\n",
      "120\t1455\t0.0\t3\t1038 303 78 36\n",
      "121\t1536\t0.0\t3\t1060 310 110 56\n",
      "122\t1492\t0.0\t3\t832 346 149 165\n",
      "123\t1561\t0.0\t3\t1005 343 128 85\n",
      "124\t2631\t0.0\t3\t1597 670 243 121\n",
      "125\t2094\t0.0\t3\t1353 487 172 82\n",
      "126\t1726\t0.0\t3\t1148 393 109 76\n",
      "127\t1509\t0.0\t3\t887 353 176 93\n",
      "128\t1602\t0.0\t3\t1058 373 110 61\n",
      "129\t1048\t0.0\t3\t622 243 98 85\n",
      "130\t1119\t0.0\t3\t753 230 85 51\n",
      "131\t1488\t0.0\t3\t1043 307 93 45\n",
      "132\t1461\t0.0\t3\t1015 301 107 38\n",
      "133\t2284\t0.0\t3\t1504 551 159 70\n",
      "134\t1411\t0.0\t3\t977 309 88 37\n",
      "135\t2078\t0.0\t3\t1394 475 146 63\n",
      "136\t2076\t0.0\t3\t1474 459 99 44\n",
      "137\t2358\t0.0\t3\t1624 555 126 53\n",
      "138\t3105\t0.0\t3\t2239 627 167 72\n",
      "139\t7225\t0.0\t3\t5445 1329 323 128\n",
      "140\t3435\t0.0\t3\t2550 666 160 59\n",
      "141\t3101\t0.0\t3\t2316 601 133 51\n",
      "142\t13918\t0.0\t3\t10237 2737 686 258\n",
      "143\t3471\t0.0\t3\t2612 655 147 57\n",
      "144\t3833\t0.0\t3\t2888 753 144 48\n",
      "145\t4308\t0.0\t3\t3311 779 151 67\n",
      "146\t4072\t0.0\t3\t3126 732 152 62\n",
      "147\t6653\t0.0\t3\t4928 1311 298 116\n",
      "148\t5643\t0.0\t3\t4253 1011 250 129\n",
      "149\t2342\t0.0\t3\t1819 378 108 37\n",
      "150\t1175\t0.0\t3\t779 251 84 61\n",
      "151\t694436\t0.0\t3\t530881 123364 28689 11502\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_03_1.fastq.gz -p ./fastq_trimmed/Int_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "Done           00:50:12   135,259,783 reads @  22.3 µs/read;   2.69 M reads/minute\n",
      "Finished in 3012.855 s (22.275 µs/read; 2.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:        135,259,783\n",
      "  Read 1 with adapter:              30,399,506 (22.5%)\n",
      "  Read 2 with adapter:              27,995,603 (20.7%)\n",
      "Pairs written (passing filters):   135,259,783 (100.0%)\n",
      "\n",
      "Total basepairs processed: 40,848,454,466 bp\n",
      "  Read 1: 20,424,227,233 bp\n",
      "  Read 2: 20,424,227,233 bp\n",
      "Total written (filtered):  38,985,441,144 bp (95.4%)\n",
      "  Read 1: 19,319,910,191 bp\n",
      "  Read 2: 19,665,530,953 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 30399506 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 13.1%\n",
      "  C: 28.4%\n",
      "  G: 33.2%\n",
      "  T: 17.5%\n",
      "  none/other: 7.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2033275\t2113434.1\t0\t2033275\n",
      "4\t1005480\t528358.5\t0\t1005480\n",
      "5\t696243\t132089.6\t0\t696243\n",
      "6\t604084\t33022.4\t0\t604084\n",
      "7\t584858\t8255.6\t0\t584858\n",
      "8\t569466\t2063.9\t0\t569466\n",
      "9\t576366\t516.0\t0\t572128 4238\n",
      "10\t583099\t129.0\t1\t564791 18308\n",
      "11\t587808\t32.2\t1\t573489 14319\n",
      "12\t584698\t8.1\t1\t569361 15337\n",
      "13\t575738\t2.0\t1\t559942 15796\n",
      "14\t571578\t0.5\t1\t554327 17251\n",
      "15\t563802\t0.1\t1\t546198 17604\n",
      "16\t561584\t0.0\t1\t541046 20538\n",
      "17\t553056\t0.0\t1\t531502 21554\n",
      "18\t553369\t0.0\t1\t531253 21777 339\n",
      "19\t546692\t0.0\t1\t523340 22502 850\n",
      "20\t555964\t0.0\t2\t528912 24106 2946\n",
      "21\t554357\t0.0\t2\t526597 24715 3045\n",
      "22\t546698\t0.0\t2\t518518 25029 3151\n",
      "23\t548611\t0.0\t2\t519930 25571 3110\n",
      "24\t543738\t0.0\t2\t513391 26928 3419\n",
      "25\t523310\t0.0\t2\t493452 26492 3366\n",
      "26\t512171\t0.0\t2\t480427 28070 3674\n",
      "27\t502387\t0.0\t2\t472675 26220 3449 43\n",
      "28\t496514\t0.0\t2\t463748 29168 3526 72\n",
      "29\t499674\t0.0\t2\t465676 30122 3706 170\n",
      "30\t488796\t0.0\t3\t453684 29849 3926 1337\n",
      "31\t487391\t0.0\t3\t452199 30062 3818 1312\n",
      "32\t478520\t0.0\t3\t443506 30025 3698 1291\n",
      "33\t467903\t0.0\t3\t431396 30940 4117 1450\n",
      "34\t463485\t0.0\t3\t430477 28114 3565 1329\n",
      "35\t443106\t0.0\t3\t411596 26906 3359 1245\n",
      "36\t430033\t0.0\t3\t399490 26079 3328 1136\n",
      "37\t410671\t0.0\t3\t381803 24812 3021 1035\n",
      "38\t398607\t0.0\t3\t370300 24118 3026 1163\n",
      "39\t391930\t0.0\t3\t364734 23278 2854 1064\n",
      "40\t380791\t0.0\t3\t354408 22545 2876 962\n",
      "41\t369946\t0.0\t3\t344551 21666 2754 975\n",
      "42\t362175\t0.0\t3\t337571 21046 2643 915\n",
      "43\t349542\t0.0\t3\t325715 20422 2446 959\n",
      "44\t338834\t0.0\t3\t315386 20148 2434 866\n",
      "45\t317984\t0.0\t3\t295909 18997 2276 802\n",
      "46\t303981\t0.0\t3\t282839 18069 2243 830\n",
      "47\t285920\t0.0\t3\t265542 17592 2063 723\n",
      "48\t267737\t0.0\t3\t248806 16417 1861 653\n",
      "49\t255017\t0.0\t3\t237727 14856 1798 636\n",
      "50\t238622\t0.0\t3\t222198 14175 1711 538\n",
      "51\t228368\t0.0\t3\t212703 13552 1564 549\n",
      "52\t217795\t0.0\t3\t202804 12907 1535 549\n",
      "53\t207744\t0.0\t3\t193806 12018 1384 536\n",
      "54\t195474\t0.0\t3\t182234 11534 1302 404\n",
      "55\t183859\t0.0\t3\t171529 10680 1210 440\n",
      "56\t170435\t0.0\t3\t159004 9846 1155 430\n",
      "57\t156175\t0.0\t3\t145539 9214 1074 348\n",
      "58\t141821\t0.0\t3\t132353 8139 952 377\n",
      "59\t129895\t0.0\t3\t121094 7627 857 317\n",
      "60\t120420\t0.0\t3\t112283 7005 800 332\n",
      "61\t108269\t0.0\t3\t100956 6340 713 260\n",
      "62\t99026\t0.0\t3\t92336 5829 606 255\n",
      "63\t93256\t0.0\t3\t87081 5330 607 238\n",
      "64\t85641\t0.0\t3\t79772 5138 549 182\n",
      "65\t79375\t0.0\t3\t73978 4673 538 186\n",
      "66\t71952\t0.0\t3\t67015 4279 508 150\n",
      "67\t64501\t0.0\t3\t59996 3913 448 144\n",
      "68\t58015\t0.0\t3\t54025 3436 399 155\n",
      "69\t51323\t0.0\t3\t47669 3161 356 137\n",
      "70\t45919\t0.0\t3\t42750 2740 303 126\n",
      "71\t41525\t0.0\t3\t38650 2481 298 96\n",
      "72\t37919\t0.0\t3\t35271 2293 274 81\n",
      "73\t33842\t0.0\t3\t31522 2013 225 82\n",
      "74\t30462\t0.0\t3\t28357 1824 191 90\n",
      "75\t28119\t0.0\t3\t26166 1716 171 66\n",
      "76\t25635\t0.0\t3\t23831 1569 168 67\n",
      "77\t22986\t0.0\t3\t21377 1408 145 56\n",
      "78\t20314\t0.0\t3\t18938 1185 136 55\n",
      "79\t18190\t0.0\t3\t16948 1077 118 47\n",
      "80\t17136\t0.0\t3\t16042 960 93 41\n",
      "81\t14657\t0.0\t3\t13601 907 105 44\n",
      "82\t13045\t0.0\t3\t12095 820 100 30\n",
      "83\t11588\t0.0\t3\t10741 712 93 42\n",
      "84\t10479\t0.0\t3\t9713 643 93 30\n",
      "85\t9294\t0.0\t3\t8611 586 70 27\n",
      "86\t8568\t0.0\t3\t7953 507 75 33\n",
      "87\t7852\t0.0\t3\t7277 502 48 25\n",
      "88\t6941\t0.0\t3\t6413 444 68 16\n",
      "89\t6655\t0.0\t3\t6108 455 60 32\n",
      "90\t6898\t0.0\t3\t6409 410 49 30\n",
      "91\t6150\t0.0\t3\t5665 395 61 29\n",
      "92\t5031\t0.0\t3\t4638 311 53 29\n",
      "93\t4840\t0.0\t3\t4433 329 50 28\n",
      "94\t3877\t0.0\t3\t3569 251 40 17\n",
      "95\t3676\t0.0\t3\t3382 239 40 15\n",
      "96\t3226\t0.0\t3\t2952 210 39 25\n",
      "97\t3412\t0.0\t3\t3128 228 36 20\n",
      "98\t3542\t0.0\t3\t3227 250 39 26\n",
      "99\t3971\t0.0\t3\t3659 254 36 22\n",
      "100\t2923\t0.0\t3\t2676 212 20 15\n",
      "101\t2718\t0.0\t3\t2482 178 37 21\n",
      "102\t2245\t0.0\t3\t2027 162 34 22\n",
      "103\t1925\t0.0\t3\t1741 148 23 13\n",
      "104\t1814\t0.0\t3\t1634 145 19 16\n",
      "105\t1682\t0.0\t3\t1513 129 23 17\n",
      "106\t1763\t0.0\t3\t1576 144 27 16\n",
      "107\t1680\t0.0\t3\t1514 126 26 14\n",
      "108\t1580\t0.0\t3\t1415 128 26 11\n",
      "109\t1403\t0.0\t3\t1263 102 20 18\n",
      "110\t1335\t0.0\t3\t1184 121 16 14\n",
      "111\t1336\t0.0\t3\t1213 84 30 9\n",
      "112\t1265\t0.0\t3\t1155 86 15 9\n",
      "113\t1206\t0.0\t3\t1072 105 16 13\n",
      "114\t999\t0.0\t3\t905 75 9 10\n",
      "115\t1178\t0.0\t3\t1060 90 12 16\n",
      "116\t967\t0.0\t3\t847 106 11 3\n",
      "117\t1022\t0.0\t3\t933 65 18 6\n",
      "118\t1007\t0.0\t3\t919 69 12 7\n",
      "119\t1133\t0.0\t3\t1011 87 20 15\n",
      "120\t1113\t0.0\t3\t1025 69 12 7\n",
      "121\t1166\t0.0\t3\t1063 88 10 5\n",
      "122\t1079\t0.0\t3\t964 92 15 8\n",
      "123\t1170\t0.0\t3\t1065 83 14 8\n",
      "124\t1623\t0.0\t3\t1501 107 7 8\n",
      "125\t1703\t0.0\t3\t1552 123 23 5\n",
      "126\t1199\t0.0\t3\t1085 93 14 7\n",
      "127\t1526\t0.0\t3\t1397 104 16 9\n",
      "128\t1399\t0.0\t3\t1273 102 19 5\n",
      "129\t1297\t0.0\t3\t1174 105 12 6\n",
      "130\t1057\t0.0\t3\t952 89 12 4\n",
      "131\t1533\t0.0\t3\t1376 133 15 9\n",
      "132\t1418\t0.0\t3\t1278 128 9 3\n",
      "133\t2210\t0.0\t3\t2033 164 9 4\n",
      "134\t1314\t0.0\t3\t1208 96 9 1\n",
      "135\t2043\t0.0\t3\t1897 138 7 1\n",
      "136\t1936\t0.0\t3\t1783 136 15 2\n",
      "137\t2304\t0.0\t3\t2129 157 11 7\n",
      "138\t2795\t0.0\t3\t2605 169 16 5\n",
      "139\t4695\t0.0\t3\t4209 453 27 6\n",
      "140\t2899\t0.0\t3\t2682 200 12 5\n",
      "141\t2911\t0.0\t3\t2704 182 19 6\n",
      "142\t7665\t0.0\t3\t7085 524 49 7\n",
      "143\t2844\t0.0\t3\t2658 172 13 1\n",
      "144\t3225\t0.0\t3\t3010 196 19\n",
      "145\t3618\t0.0\t3\t3393 199 21 5\n",
      "146\t4378\t0.0\t3\t4108 249 19 2\n",
      "147\t5900\t0.0\t3\t5496 362 37 5\n",
      "148\t5977\t0.0\t3\t5599 343 30 5\n",
      "149\t2242\t0.0\t3\t2045 169 19 9\n",
      "150\t18978\t0.0\t3\t556 14557 3670 195\n",
      "151\t2388404\t0.0\t3\t242094 1996737 136016 13557\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 27995603 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.5%\n",
      "  C: 29.5%\n",
      "  G: 41.8%\n",
      "  T: 10.2%\n",
      "  none/other: 0.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2290726\t2113434.1\t0\t2290726\n",
      "4\t1010341\t528358.5\t0\t1010341\n",
      "5\t718248\t132089.6\t0\t718248\n",
      "6\t608829\t33022.4\t0\t608829\n",
      "7\t582817\t8255.6\t0\t582817\n",
      "8\t571473\t2063.9\t0\t571473\n",
      "9\t578021\t516.0\t0\t571532 6489\n",
      "10\t581798\t129.0\t1\t565763 16035\n",
      "11\t587055\t32.2\t1\t572002 15053\n",
      "12\t583051\t8.1\t1\t568467 14584\n",
      "13\t573132\t2.0\t1\t555354 17778\n",
      "14\t569694\t0.5\t1\t553348 16346\n",
      "15\t561926\t0.1\t1\t545089 16837\n",
      "16\t560003\t0.0\t1\t541890 18113\n",
      "17\t551083\t0.0\t1\t530972 20111\n",
      "18\t551936\t0.0\t1\t531038 20719 179\n",
      "19\t545061\t0.0\t1\t521784 22284 993\n",
      "20\t553122\t0.0\t2\t434817 113467 4838\n",
      "21\t549695\t0.0\t2\t359942 161766 27987\n",
      "22\t542382\t0.0\t2\t422184 93572 26626\n",
      "23\t544316\t0.0\t2\t435335 86988 21993\n",
      "24\t539531\t0.0\t2\t435466 87354 16711\n",
      "25\t510963\t0.0\t2\t379445 102579 28939\n",
      "26\t497475\t0.0\t2\t379734 91338 26403\n",
      "27\t488777\t0.0\t2\t368270 93282 27193 32\n",
      "28\t479158\t0.0\t2\t233669 199877 45532 80\n",
      "29\t483770\t0.0\t2\t337901 112353 32294 1222\n",
      "30\t477088\t0.0\t3\t320155 108940 35372 12621\n",
      "31\t468401\t0.0\t3\t278892 122537 45742 21230\n",
      "32\t456962\t0.0\t3\t287410 110902 39650 19000\n",
      "33\t449150\t0.0\t3\t294939 102996 34186 17029\n",
      "34\t445060\t0.0\t3\t290226 102458 35659 16717\n",
      "35\t427372\t0.0\t3\t279137 99424 33133 15678\n",
      "36\t405604\t0.0\t3\t259037 95197 34003 17367\n",
      "37\t387530\t0.0\t3\t262480 85390 27091 12569\n",
      "38\t382395\t0.0\t3\t245482 89132 32071 15710\n",
      "39\t373428\t0.0\t3\t240597 87133 30644 15054\n",
      "40\t365470\t0.0\t3\t236792 84991 29227 14460\n",
      "41\t340730\t0.0\t3\t225475 77018 25282 12955\n",
      "42\t350873\t0.0\t3\t229405 82595 26833 12040\n",
      "43\t333931\t0.0\t3\t217210 77421 26342 12958\n",
      "44\t325039\t0.0\t3\t216329 74063 23734 10913\n",
      "45\t307314\t0.0\t3\t209320 68040 20577 9377\n",
      "46\t294741\t0.0\t3\t194915 67364 22211 10251\n",
      "47\t276248\t0.0\t3\t179813 63839 22055 10541\n",
      "48\t259673\t0.0\t3\t175925 57218 18259 8271\n",
      "49\t247316\t0.0\t3\t163282 56901 18691 8442\n",
      "50\t227545\t0.0\t3\t151009 51150 17000 8386\n",
      "51\t221911\t0.0\t3\t148800 49331 16221 7559\n",
      "52\t209752\t0.0\t3\t143487 45403 14345 6517\n",
      "53\t199193\t0.0\t3\t133995 43881 14347 6970\n",
      "54\t189880\t0.0\t3\t130199 41416 12685 5580\n",
      "55\t176180\t0.0\t3\t113565 41028 14572 7015\n",
      "56\t164325\t0.0\t3\t111585 35225 11898 5617\n",
      "57\t150349\t0.0\t3\t105068 30988 9711 4582\n",
      "58\t136734\t0.0\t3\t92304 29904 9840 4686\n",
      "59\t124715\t0.0\t3\t83119 27874 9314 4408\n",
      "60\t115447\t0.0\t3\t77936 25432 8242 3837\n",
      "61\t104459\t0.0\t3\t71363 22263 7413 3420\n",
      "62\t95832\t0.0\t3\t64099 21464 6937 3332\n",
      "63\t89801\t0.0\t3\t62169 19031 5882 2719\n",
      "64\t83520\t0.0\t3\t58989 17052 5128 2351\n",
      "65\t76661\t0.0\t3\t54265 15422 4746 2228\n",
      "66\t69427\t0.0\t3\t48458 14352 4417 2200\n",
      "67\t62655\t0.0\t3\t42930 13480 4243 2002\n",
      "68\t56492\t0.0\t3\t38076 12435 4115 1866\n",
      "69\t49613\t0.0\t3\t33857 10504 3565 1687\n",
      "70\t44794\t0.0\t3\t31290 9328 2835 1341\n",
      "71\t39706\t0.0\t3\t27253 8349 2766 1338\n",
      "72\t36963\t0.0\t3\t24968 8090 2657 1248\n",
      "73\t32510\t0.0\t3\t21741 7150 2415 1204\n",
      "74\t29346\t0.0\t3\t19897 6349 2055 1045\n",
      "75\t27222\t0.0\t3\t18549 5811 1918 944\n",
      "76\t24854\t0.0\t3\t17082 5077 1820 875\n",
      "77\t22180\t0.0\t3\t15468 4535 1471 706\n",
      "78\t19790\t0.0\t3\t13468 4190 1413 719\n",
      "79\t17533\t0.0\t3\t12106 3588 1236 603\n",
      "80\t16531\t0.0\t3\t11539 3472 986 534\n",
      "81\t14109\t0.0\t3\t9703 2949 972 485\n",
      "82\t12554\t0.0\t3\t8434 2710 909 501\n",
      "83\t11198\t0.0\t3\t7674 2404 732 388\n",
      "84\t10058\t0.0\t3\t6980 2114 649 315\n",
      "85\t8998\t0.0\t3\t6294 1789 596 319\n",
      "86\t8334\t0.0\t3\t5854 1676 520 284\n",
      "87\t7630\t0.0\t3\t5342 1558 504 226\n",
      "88\t6641\t0.0\t3\t4589 1381 452 219\n",
      "89\t6456\t0.0\t3\t4663 1241 371 181\n",
      "90\t6747\t0.0\t3\t4702 1382 436 227\n",
      "91\t5968\t0.0\t3\t4050 1248 448 222\n",
      "92\t4899\t0.0\t3\t3361 1010 346 182\n",
      "93\t4723\t0.0\t3\t3294 952 305 172\n",
      "94\t3696\t0.0\t3\t2548 739 272 137\n",
      "95\t3601\t0.0\t3\t2560 708 211 122\n",
      "96\t3065\t0.0\t3\t2117 596 236 116\n",
      "97\t3262\t0.0\t3\t2294 605 228 135\n",
      "98\t3339\t0.0\t3\t2331 661 225 122\n",
      "99\t3827\t0.0\t3\t2654 766 272 135\n",
      "100\t2814\t0.0\t3\t2027 483 210 94\n",
      "101\t2633\t0.0\t3\t1913 485 151 84\n",
      "102\t2152\t0.0\t3\t1487 403 170 92\n",
      "103\t1832\t0.0\t3\t1285 369 113 65\n",
      "104\t1706\t0.0\t3\t1205 296 131 74\n",
      "105\t1575\t0.0\t3\t1062 321 125 67\n",
      "106\t1669\t0.0\t3\t1196 302 118 53\n",
      "107\t1615\t0.0\t3\t1146 288 122 59\n",
      "108\t1526\t0.0\t3\t1054 301 104 67\n",
      "109\t1349\t0.0\t3\t889 275 116 69\n",
      "110\t1277\t0.0\t3\t863 274 91 49\n",
      "111\t1283\t0.0\t3\t864 259 106 54\n",
      "112\t1220\t0.0\t3\t822 239 90 69\n",
      "113\t1156\t0.0\t3\t766 244 88 58\n",
      "114\t964\t0.0\t3\t657 195 67 45\n",
      "115\t1104\t0.0\t3\t707 235 106 56\n",
      "116\t938\t0.0\t3\t652 182 63 41\n",
      "117\t996\t0.0\t3\t615 226 117 38\n",
      "118\t979\t0.0\t3\t658 210 57 54\n",
      "119\t1073\t0.0\t3\t672 238 109 54\n",
      "120\t984\t0.0\t3\t695 188 64 37\n",
      "121\t1059\t0.0\t3\t681 248 84 46\n",
      "122\t906\t0.0\t3\t574 202 76 54\n",
      "123\t1092\t0.0\t3\t715 222 85 70\n",
      "124\t1673\t0.0\t3\t1061 379 151 82\n",
      "125\t1603\t0.0\t3\t1039 328 135 101\n",
      "126\t1128\t0.0\t3\t725 237 108 58\n",
      "127\t1417\t0.0\t3\t834 341 143 99\n",
      "128\t1330\t0.0\t3\t874 307 102 47\n",
      "129\t1183\t0.0\t3\t700 281 131 71\n",
      "130\t970\t0.0\t3\t614 227 82 47\n",
      "131\t1450\t0.0\t3\t998 289 109 54\n",
      "132\t1339\t0.0\t3\t953 257 93 36\n",
      "133\t2081\t0.0\t3\t1383 484 133 81\n",
      "134\t1263\t0.0\t3\t879 273 73 38\n",
      "135\t1931\t0.0\t3\t1313 424 122 72\n",
      "136\t1863\t0.0\t3\t1390 325 109 39\n",
      "137\t2198\t0.0\t3\t1532 471 138 57\n",
      "138\t2711\t0.0\t3\t1985 535 139 52\n",
      "139\t4419\t0.0\t3\t3225 849 245 100\n",
      "140\t2784\t0.0\t3\t2071 539 116 58\n",
      "141\t2817\t0.0\t3\t2062 562 140 53\n",
      "142\t7369\t0.0\t3\t5363 1413 432 161\n",
      "143\t2776\t0.0\t3\t2090 514 121 51\n",
      "144\t3173\t0.0\t3\t2430 571 121 51\n",
      "145\t3525\t0.0\t3\t2717 592 151 65\n",
      "146\t4260\t0.0\t3\t3355 653 169 83\n",
      "147\t5693\t0.0\t3\t4249 1028 294 122\n",
      "148\t5832\t0.0\t3\t4422 1022 249 139\n",
      "149\t2098\t0.0\t3\t1614 369 90 25\n",
      "150\t628\t0.0\t3\t444 107 45 32\n",
      "151\t258454\t0.0\t3\t200144 42439 10966 4905\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_04_1.fastq.gz -p ./fastq_trimmed/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "[---=8       ] 01:04:11   164,480,000 reads @  24.0 µs/read;   2.50 M reads/minute"
     ]
    }
   ],
   "source": [
    "# !cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_01_1.fastq.gz -p ./fastq_trimmed/Int_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-01_S42_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_02_1.fastq.gz -p ./fastq_trimmed/Int_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-02_S43_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_03_1.fastq.gz -p ./fastq_trimmed/Int_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-03_S44_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Int_04_1.fastq.gz -p ./fastq_trimmed/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Int-04_S45_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_01_1.fastq.gz -p ./fastq_trimmed/Liv_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-01_S38_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_02_1.fastq.gz -p ./fastq_trimmed/Liv_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-02_S39_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-02_S39_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_03_1.fastq.gz -p ./fastq_trimmed/Liv_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-03_S40_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-03_S40_L005_R2_001.fastq.gz\n",
    "!cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed/Liv_04_1.fastq.gz -p ./fastq_trimmed/Liv_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-04_S41_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/Liv-04_S41_L005_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.001s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "application/gzip\n",
      "application/gzip\n",
      "Failed to process fastq_trimmed/Int-01_S42_L005_R2_001.fastq.gz\n",
      "java.io.EOFException\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:258)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:91)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:37)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.<init>(FastQFile.java:84)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:106)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:62)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.processFile(OfflineRunner.java:163)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.<init>(OfflineRunner.java:125)\n",
      "\tat uk.ac.babraham.FastQC.FastQCApplication.main(FastQCApplication.java:316)\n",
      "Started analysis of Int-01_S42_L005_R1_001.fastq.gz\n",
      "application/gzip\n",
      "Approx 5% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Failed to process file Int-01_S42_L005_R1_001.fastq.gz\n",
      "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: Corrupt GZIP trailer\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:222)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:129)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:77)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:833)\n",
      "Approx 5% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R2_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!fastqc -t 16 fastq_trimmed/*.fastq.gz -o fastq_trimmed_qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.001s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int-01_S42_L005_R1_001.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Approx 5% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_qc\n",
    "!fastqc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/*fastq.gz -o fastq_qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[91m///\u001b[0m \u001b]8;id=389488;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m66/66\u001b[0m  66\u001b[0m  \n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 1 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Existing reports found, adding suffix to filenames. Use '--force' to overwrite.\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report_2.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_data_2\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "# !mkdir fastq_trimmed_qc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/multiqc_report.html\n",
    "# !cp /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/multiqc_report.html multiqc_report.html\n",
    "!multiqc -f -d fastq_qc/ fastq_trimmed_qc/ -n multiqc_report_all_ipynb.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.015s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /dev/cpuset.\n",
      "text/html\n",
      "Failed to process multiqc_report.html\n",
      "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: ID line didn't start with '@' at line 1\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:163)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.<init>(FastQFile.java:93)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:106)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:62)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.processFile(OfflineRunner.java:163)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.<init>(OfflineRunner.java:125)\n",
      "\tat uk.ac.babraham.FastQC.FastQCApplication.main(FastQCApplication.java:316)\n"
     ]
    }
   ],
   "source": [
    "# !fastqc multiqc_report.html\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                \n",
      " \u001b[91m///\u001b[0m \u001b]8;id=494690;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m                                                         \n",
      "                                                                                \n",
      " \u001b[33mUsage:\u001b[0m \u001b[1mmultiqc\u001b[0m [\u001b[1;36mOPTIONS\u001b[0m] [\u001b[1;36mANALYSIS\u001b[0m \u001b[1;36mDIRECTORY\u001b[0m]                                  \n",
      "                                                                                \n",
      " MultiQC aggregates results from bioinformatics analyses across many samples    \n",
      " into a single report.                                                          \n",
      " \u001b[2mIt searches a given directory for analysis logs and compiles a HTML report. \u001b[0m   \n",
      " \u001b[2mIt's a general use tool, perfect for summarising the output from numerous \u001b[0m     \n",
      " \u001b[2mbioinformatics tools.\u001b[0m                                                          \n",
      " \u001b[2mTo run, supply with one or more directory to scan for analysis results. For \u001b[0m   \n",
      " \u001b[2mexample, to run in the current working directory, use '\u001b[0m\u001b[1;2;34mmultiqc .\u001b[0m\u001b[2m'\u001b[0m              \n",
      "                                                                                \n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Main options \u001b[0m\u001b[2m──────────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--force\u001b[0m            \u001b[1;32m-f\u001b[0m  Overwrite any existing reports                        \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--config\u001b[0m           \u001b[1;32m-c\u001b[0m  Specific config file to load, after those in MultiQC  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        dir / home dir / working dir.                         \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(PATH)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--cl-config\u001b[0m            Specify MultiQC config YAML on the command line       \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                         \u001b[0m       \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--filename\u001b[0m         \u001b[1;32m-n\u001b[0m  Report filename. Use '\u001b[3;33mstdout\u001b[0m' to print to standard    \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        out.                                                  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--outdir\u001b[0m           \u001b[1;32m-o\u001b[0m  Create report in the specified output directory.      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                          \u001b[0m      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--ignore\u001b[0m           \u001b[1;32m-x\u001b[0m  Ignore analysis files \u001b[2;33m(GLOB EXPRESSION)\u001b[0m               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--ignore-samples\u001b[0m       Ignore sample names \u001b[2;33m(GLOB EXPRESSION)\u001b[0m                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--ignore-symlinks\u001b[0m      Ignore symlinked directories and files                \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--file-list\u001b[0m        \u001b[1;32m-l\u001b[0m  Supply a file containing a list of file paths to be   \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        searched, one per row                                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Choosing modules to run \u001b[0m\u001b[2m───────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--module\u001b[0m     \u001b[1;32m-m\u001b[0m  Use only this module. Can specify multiple times.           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                  \u001b[2;33m(MODULE NAME)                                    \u001b[0m           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--exclude\u001b[0m    \u001b[1;32m-e\u001b[0m  Do not use this module. Can specify multiple times.         \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                  \u001b[2;33m(MODULE NAME)                                      \u001b[0m         \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--tag\u001b[0m            Use only modules which tagged with this keyword \u001b[2;33m(TEXT)\u001b[0m      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--view-tags\u001b[0m      View the available tags and which modules they load         \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Sample handling \u001b[0m\u001b[2m───────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--dirs\u001b[0m           \u001b[1;32m-d\u001b[0m   Prepend directory to sample names                      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--dirs-depth\u001b[0m     \u001b[1;32m-dd\u001b[0m  Prepend \u001b[3;33mn\u001b[0m directories to sample names. Negative number \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       to take from start of path.                            \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       \u001b[2;33m(INTEGER)                                             \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--fullnames\u001b[0m      \u001b[1;32m-s\u001b[0m   Do not clean the sample names \u001b[3m(leave as full file \u001b[0m     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       \u001b[3mname)\u001b[0m                                                  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--fn_as_s_name\u001b[0m        Use the log filename as the sample name                \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--replace-names\u001b[0m       TSV file to rename sample names during report          \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       generation                                             \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                       \u001b[2;33m(PATH)                                                \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Report customisation \u001b[0m\u001b[2m──────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--title\u001b[0m            \u001b[1;32m-i\u001b[0m  Report title. Printed as page header, used for        \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        filename if not otherwise specified.                  \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--comment\u001b[0m          \u001b[1;32m-b\u001b[0m  Custom comment, will be printed at the top of the     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        report.                                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(TEXT)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--template\u001b[0m         \u001b[1;32m-t\u001b[0m  Report template to use.                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(default|default_dev|gathered|geo|sections|simple)\u001b[0m    \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--sample-names\u001b[0m         TSV file containing alternative sample names for      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        renaming buttons in the report                        \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(PATH)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--sample-filters\u001b[0m       TSV file containing show/hide patterns for the report \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                        \u001b[2;33m(PATH)                                               \u001b[0m \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--custom-css-file\u001b[0m      Custom CSS file to add to the final report \u001b[2;33m(PATH)\u001b[0m     \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m Output files \u001b[0m\u001b[2m──────────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--flat\u001b[0m          \u001b[1;32m-fp\u001b[0m  Use only flat plots \u001b[3m(static images)\u001b[0m                     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--interactive\u001b[0m   \u001b[1;32m-ip\u001b[0m  Use only interactive plots \u001b[3m(in-browser Javascript)\u001b[0m      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--export\u001b[0m        \u001b[1;32m-p\u001b[0m   Export plots as static images in addition to the report \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--data-dir\u001b[0m           Force the parsed data directory to be created.          \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-data-dir\u001b[0m        Prevent the parsed data directory from being created.   \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--data-format\u001b[0m   \u001b[1;32m-k\u001b[0m   Output parsed data in a different format.               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                      \u001b[2;33m(tsv|json|yaml)                          \u001b[0m               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--zip-data-dir\u001b[0m  \u001b[1;32m-z\u001b[0m   Compress the data directory.                            \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-report\u001b[0m          Do not generate a report, only export data and plots    \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--pdf\u001b[0m                Creates PDF report with the \u001b[3m'simple'\u001b[0m template. Requires \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                      \u001b]8;id=183873;https://pandoc.org/\u001b\\Pandoc\u001b]8;;\u001b\\ to be installed.                                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "\u001b[2m╭─\u001b[0m\u001b[2m MultiQC behaviour \u001b[0m\u001b[2m─────────────────────────────────────────────────────────\u001b[0m\u001b[2m─╮\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--verbose\u001b[0m           \u001b[1;32m-v\u001b[0m  Increase output verbosity. \u001b[2;33m(INTEGER RANGE)\u001b[0m           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--quiet\u001b[0m             \u001b[1;32m-q\u001b[0m  Only show log warnings                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--strict\u001b[0m                Don't catch exceptions, run additional code checks   \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         to help development.                                 \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--require-logs\u001b[0m          Require all explicitly requested modules to have log \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         files. If not, MultiQC will exit with an error.      \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--profile-runtime\u001b[0m       Add analysis of how long MultiQC takes to run to the \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         report                                               \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-megaqc-upload\u001b[0m      Don't upload generated report to MegaQC, even if     \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m                         MegaQC options are found                             \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--no-ansi\u001b[0m               Disable coloured log output                          \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--version\u001b[0m               Show the version and exit.                           \u001b[2m│\u001b[0m\n",
      "\u001b[2m│\u001b[0m \u001b[1;36m--help\u001b[0m              \u001b[1;32m-h\u001b[0m  Show this message and exit.                          \u001b[2m│\u001b[0m\n",
      "\u001b[2m╰──────────────────────────────────────────────────────────────────────────────╯\u001b[0m\n",
      "                                                                                \n",
      " See \u001b]8;id=176947;http://multiqc.info\u001b\\http://multiqc.info\u001b]8;;\u001b\\ for more details.                                      \n",
      "\n"
     ]
    }
   ],
   "source": [
    "!multiqc --help\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Step 1: Review Metadata and QC Data\n",
    "1.1 Review QC data from the qc/ folder:\n",
    "Check the quality of the RNA extraction, library prep, and sequencing process. These details will inform decisions about trimming and filtering reads.\n",
    "The QC reports may include metrics such as sequencing depth, base composition, GC content, and read length distributions, which can guide trimming and filtering steps.\n",
    "1.2 Read the reference paper:\n",
    "Understanding the study in the paper \"Winners vs. losers: Schistosoma mansoni intestinal and liver eggs exhibit striking differences in gene expression and immunogenicity\" will give you insights into the expected differences between liver and intestine miracidia, helping you interpret the results later.\n",
    "##Step 2: Read Preprocessing (Trimming and Quality Control)\n",
    "2.1 Assess the raw reads:\n",
    "Use tools like FastQC to assess the raw sequencing data. Create a MultiQC report to summarize the results for all samples. This will help identify issues like adapter contamination, GC bias, or low-quality regions.\n",
    "2.2 Trim low-quality bases and adapters:\n",
    "Use Trimmomatic or Cutadapt to remove adapters and trim low-quality bases (e.g., Q-score below 20). Focus on trimming the 3' and 5' ends of the reads if necessary.\n",
    "After trimming, run FastQC again and generate a new MultiQC report to ensure quality has improved.\n",
    "2.3 Verify trimming results:\n",
    "Check that the trimmed reads have good quality, proper length distributions, and no obvious contamination. If necessary, adjust the trimming parameters.\n",
    "##Step 3: Read Alignment to the Reference Genome\n",
    "3.1 Align trimmed reads:\n",
    "Use STAR or HISAT2 for aligning the reads to the Schistosoma mansoni genome. The choice of aligner will depend on the quality of your reference genome.\n",
    "Align the reads in a strand-specific manner since the NEBNext Ultra II Directional mRNA kit was used. This will ensure that the strand information is preserved for downstream analyses.\n",
    "3.2 Assess alignment quality:\n",
    "Use SAMtools to check the alignment quality. This will include assessing mapping rate, proper pairs, and other metrics.\n",
    "You can use Picard tools to generate alignment statistics and visualize any potential issues.\n",
    "##Step 4: Counting Reads\n",
    "4.1 Generate gene counts:\n",
    "Use featureCounts (from the Subread package) or HTSeq-count to generate a gene count matrix based on the aligned BAM files. Ensure you use a GTF file for Schistosoma mansoni gene annotations.\n",
    "The output should be a count matrix, with genes in rows and samples in columns.\n",
    "4.2 Inspect counts:\n",
    "Visualize the count distribution across samples using boxplots or histograms to ensure that the counts look reasonable.\n",
    "##Step 5: PCA Analysis\n",
    "5.1 Normalize the counts:\n",
    "Normalize the gene count data to account for sequencing depth and gene length. Use DESeq2 or edgeR for this purpose.\n",
    "Normalize to obtain variance-stabilized counts or log-transformed counts for downstream analyses.\n",
    "5.2 Perform PCA:\n",
    "Use DESeq2 or prcomp (in R or Python) to perform Principal Component Analysis (PCA) on the normalized count data.\n",
    "Visualize the PCA results with a ggplot2 scatter plot, coloring by sample origin (liver vs. intestine). Look for clustering patterns between the two groups.\n",
    "##Step 6: Differential Expression Analysis\n",
    "6.1 Perform differential expression analysis:\n",
    "Use DESeq2 to identify differentially expressed genes (DEGs) between the liver and intestine samples.\n",
    "Set up a contrast in DESeq2 to compare liver versus intestine groups, and run the differential expression pipeline.\n",
    "6.2 Visualize DEGs:\n",
    "Use a volcano plot to visualize the DEGs. Plot log-fold changes (logFC) on the x-axis and the significance (–log10 p-value) on the y-axis. Highlight the most significant DEGs.\n",
    "Consider additional visualizations like MA plots or heatmaps of DEGs, especially genes with high or low expression.\n",
    "##Step 7: Additional Requirements\n",
    "7.1 Create a Conda Environment:\n",
    "Provide an environment.yaml file that includes the required packages for the analysis (e.g., DESeq2, FastQC, Trimmomatic, STAR, featureCounts, MultiQC, etc.).\n",
    "Ensure the environment is reproducible, and list all dependencies.\n",
    "7.2 Version Control with Git:\n",
    "Initialize a Git repository for version control.\n",
    "Make commits at regular intervals (at least one per week). Write informative commit messages and organize the project structure logically.\n",
    "7.3 Create a Snakemake Pipeline:\n",
    "Write a Snakefile to automate the analysis from raw reads to differential expression analysis. The pipeline should handle tasks like trimming, alignment, counting, and DESeq2 analysis in a reproducible manner.\n",
    "Specify the necessary Conda environments using --use-conda to ensure reproducibility.\n",
    "##Step 8: Report and Documentation\n",
    "8.1 Write an IPython Notebook:\n",
    "Create a well-documented and organized IPython notebook to explain the steps of the analysis.\n",
    "Include sections for each step of the process, with markdown cells explaining the rationale behind each step and what you are looking for in the output logs and plots.\n",
    "Include the necessary plots (PCA and volcano plot) and interpret the results.\n",
    "8.2 Include a MultiQC Report:\n",
    "Generate a MultiQC report that aggregates the quality control results from all stages of the analysis (raw reads, trimmed reads, alignment, counting, and PCA). This will help assess the overall quality of the data at each stage.\n",
    "Final Deliverables:\n",
    "IPython Notebook with full analysis and interpretations.\n",
    "MultiQC Report with QC information.\n",
    "environment.yaml file for Conda environment setup.\n",
    "Gene count matrix.\n",
    "Snakefile for reproducible analysis pipeline (if aiming for an A grade).\n",
    "Git repository with version-controlled code (for B or higher).\n",
    "Proper documentation and clear presentation of results.\n",
    "By following this structure, you will meet the grade specifications and ensure a reproducible and comprehensive RNA-seq analysis pipeline.\n",
    "\n",
    "\n",
    "\n",
    "You said:\n",
    "what does it mean by include YAML that uses --use-conda\n",
    "ChatGPT said:\n",
    "ChatGPT\n",
    "In this context, the requirement to \"include YAML that uses --use-conda\" refers to specifying the dependencies for your analysis pipeline in a YAML file (specifically for Snakemake), which ensures that the necessary software and tools can be installed and used within a conda environment.\n",
    "\n",
    "When you're using Snakemake to automate your analysis, you can use the --use-conda option to let Snakemake automatically create and manage conda environments for each of the steps in the workflow. The YAML file, known as the environment file, defines the conda environment's dependencies (i.e., the software packages needed for each step).\n",
    "\n",
    "Here’s a breakdown:\n",
    "\n",
    "1. YAML File (Environment Specification)\n",
    "This YAML file lists all the conda dependencies required for the analysis. You can create an environment YAML file for each tool or step (e.g., FastQC, DESeq2, featureCounts, STAR), and Snakemake will automatically set up the appropriate environments.\n",
    "\n",
    "Example of a simple environment.yaml file:\n",
    "\n",
    "yaml\n",
    "Copy code\n",
    "name: rnaseq_analysis\n",
    "channels:\n",
    "  - bioconda\n",
    "  - conda-forge\n",
    "  - defaults\n",
    "dependencies:\n",
    "  - python=3.8\n",
    "  - fastqc\n",
    "  - star\n",
    "  - samtools\n",
    "  - deeq2\n",
    "  - multiqc\n",
    "  - r-base\n",
    "  - r-ggplot2\n",
    "2. Using --use-conda in Snakemake\n",
    "When running Snakemake, you specify --use-conda to tell Snakemake to automatically use conda environments for each rule. Snakemake will look for an environment.yaml file in the same directory or specified for each individual step of your workflow and create the environment from the dependencies listed in that file.\n",
    "\n",
    "Example Snakemake command:\n",
    "\n",
    "bash\n",
    "Copy code\n",
    "snakemake --use-conda\n",
    "This command will make sure that Snakemake uses conda environments where necessary to run the steps in your pipeline.\n",
    "\n",
    "3. Snakemake Rules with Environment YAMLs\n",
    "For each step of your pipeline, you can link to a specific environment YAML file by using the conda directive in your Snakemake rule. For example:\n",
    "\n",
    "python\n",
    "Copy code\n",
    "rule fastqc:\n",
    "    input:\n",
    "        \"raw_data/{sample}.fastq\"\n",
    "    output:\n",
    "        \"qc/{sample}_fastqc.html\"\n",
    "    conda:\n",
    "        \"envs/fastqc_env.yaml\"  # this is your conda environment file for FastQC\n",
    "    shell:\n",
    "        \"fastqc {input} -o {output}\"\n",
    "Why This is Useful:\n",
    "Reproducibility: By specifying the environment and the exact versions of dependencies, it ensures that the analysis can be run again with the same software versions, even on a different machine.\n",
    "Dependency Management: Automatically manages the installation of all required software packages for each step in the workflow without worrying about potential version conflicts.\n",
    "In summary, to meet the requirement of \"include YAML that uses --use-conda,\" you would:\n",
    "\n",
    "Create an environment.yaml file listing the necessary software dependencies.\n",
    "Use Snakemake with the --use-conda option to automatically create conda environments and ensure reproducibility."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "in single end seq we get a single end primer that does the seq by synthesis and seq all accross the fragment and reads the fluoresence\n",
    "for paired end we add an entire second round that adds to the other end and does the reverse end of the original \n",
    "star deals with reads that overlap differently based on their number? maybe"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "final_project_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
