{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NEXT_seq Worm genome Alignment\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this notebook, \n",
    "-   import reference genome and its annotations\n",
    "-   trim fasta files from the Worm Study\n",
    "-   analyze the trimming to confirm the quality is met and improved\n",
    "-   "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘genome’: File exists\n",
      "--2024-12-19 13:49:59--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 116797085 (111M) [application/x-gzip]\n",
      "Saving to: ‘genome/genome.fa.gz’\n",
      "\n",
      "genome/genome.fa.gz 100%[===================>] 111.39M   588KB/s    in 3m 14s  \n",
      "\n",
      "2024-12-19 13:53:13 (587 KB/s) - ‘genome/genome.fa.gz’ saved [116797085/116797085]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!mkdir genome \n",
    "    #creates the directory that will hold the reference genome and it's annotated gene files\n",
    "!wget -nc -O genome/genome.fa.gz https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz \n",
    "    #downloads the file from the given url, in this case it is downloading the genome fasta file the genetic sequence for schistosoma_mansoni\n",
    "    #-O allows the specification of the output directory\n",
    "    #-nc prevents overwritting a file that already exists\n",
    "!gzip -d -f genome/genome.fa.gz\n",
    "    # the gzip takes the sequence data from the wget and compresses it so that the file can be transferred easier and saves storage space"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import the annotations for known genes in worm genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-11-21 16:30:04--  https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz\n",
      "Resolving ftp.ebi.ac.uk... 193.62.193.165\n",
      "Connecting to ftp.ebi.ac.uk|193.62.193.165|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3228084 (3.1M) [application/x-gzip]\n",
      "Saving to: ‘STDOUT’\n",
      "\n",
      "-                   100%[===================>]   3.08M  3.82MB/s    in 0.8s    \n",
      "\n",
      "2024-11-21 16:30:06 (3.82 MB/s) - written to stdout [3228084/3228084]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget -O - https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz | \\\n",
    "        #grabs the annotation file from the given url and outputs it to the specified directory\n",
    "    gzip -f -d > genome/annotations.gtf\n",
    "        #-f is to force it to overwrite any existing annotation file and -d decompress the file so it's not as large"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Index the genome for efficient gene referencing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "!samtools faidx genome/genome.fa\n",
    "    #creates an index file that allows for efficient random access to the sequences in the fasta file"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CutAdapt Trimming the Genome\n",
    "    In this step we prepare the reads to have a higher quality to give the read a higher probility of success in aligning"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each of the following cutadapt calls takes the samples and trims them based on the specified instructions\n",
    "- -j sets cutadapt to use, 0 automatically detects the number available\n",
    "- -m sets the minimum length of a read to 20, that way any reads that are too small to process won't cause problems in the alignment steps\n",
    "- --nextseq-trim accounts for the instruments used for sequencing that encode G, the normal quality trimmer is unable to handle this problem so this is used rather than q=10\n",
    "- -u and -U removes the first 10 base pairs from the start of both of-  the paired end reads, this was chosed because qc output indicated that these bp were causing data to be less optimal\n",
    "- -a and -A with adapter sequence, removes the adapter used by next seq \n",
    "        removing the adapter improves alignment since the adapter is not supposed to be in the gene of a worm so it will most likely not be able to find alignments if we don't remove the adapter that isn't part of the genes in a worm\n",
    "- after specifying how we want to cut the reads, the location of the trimmed reads is given as -o for the forward end and -p for the reverse read\n",
    "- then the location of the files that we are trimming is specified namining the forward read and then the reverse read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_01_1.fastq.gz -p ./fastq_trimmed_sub/Int_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-01_S42_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-01_S42_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:20    12,421,987 reads @   1.6 µs/read;  36.58 M reads/minute\n",
      "Finished in 20.452 s (1.646 µs/read; 36.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         12,421,987\n",
      "  Read 1 with adapter:               3,346,061 (26.9%)\n",
      "  Read 2 with adapter:               3,263,630 (26.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             253,185 (2.0%)\n",
      "Pairs written (passing filters):    12,168,802 (98.0%)\n",
      "\n",
      "Total basepairs processed: 3,751,440,074 bp\n",
      "  Read 1: 1,875,720,037 bp\n",
      "  Read 2: 1,875,720,037 bp\n",
      "Quality-trimmed:              64,228,854 bp (1.7%)\n",
      "  Read 1:    22,070,914 bp\n",
      "  Read 2:    42,157,940 bp\n",
      "Total written (filtered):  3,247,008,955 bp (86.6%)\n",
      "  Read 1: 1,622,608,797 bp\n",
      "  Read 2: 1,624,400,158 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 3346061 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.9%\n",
      "  C: 33.8%\n",
      "  G: 34.1%\n",
      "  T: 17.1%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t276633\t194093.5\t0\t276633\n",
      "4\t106193\t48523.4\t0\t106193\n",
      "5\t206926\t12130.8\t0\t206926\n",
      "8\t63916\t189.5\t0\t63916\n",
      "9\t127708\t47.4\t0\t127708\n",
      "10\t828\t11.8\t1\t0 828\n",
      "11\t133723\t3.0\t1\t132218 1505\n",
      "12\t628\t0.7\t1\t0 628\n",
      "13\t65758\t0.2\t1\t64518 1240\n",
      "14\t65866\t0.0\t1\t64470 1396\n",
      "15\t64602\t0.0\t1\t63079 1523\n",
      "16\t65947\t0.0\t1\t63750 2197\n",
      "17\t126603\t0.0\t1\t124255 2348\n",
      "18\t875\t0.0\t1\t0 860 15\n",
      "19\t63916\t0.0\t1\t62124 1718 74\n",
      "20\t65034\t0.0\t2\t62144 2676 214\n",
      "21\t130609\t0.0\t2\t127847 2408 354\n",
      "22\t1159\t0.0\t2\t0 1067 92\n",
      "23\t66637\t0.0\t2\t64351 2059 227\n",
      "24\t66977\t0.0\t2\t64399 2316 262\n",
      "25\t62732\t0.0\t2\t60347 2160 225\n",
      "26\t60880\t0.0\t2\t58307 2318 255\n",
      "27\t58665\t0.0\t2\t56322 2099 244\n",
      "28\t59875\t0.0\t2\t56224 3382 269\n",
      "29\t117868\t0.0\t2\t114498 2935 427 8\n",
      "30\t1413\t0.0\t3\t0 1273 105 35\n",
      "31\t59751\t0.0\t3\t56958 2453 251 89\n",
      "32\t57198\t0.0\t3\t54465 2348 295 90\n",
      "33\t58415\t0.0\t3\t54388 3629 285 113\n",
      "34\t111190\t0.0\t3\t107780 2847 407 156\n",
      "35\t1510\t0.0\t3\t136 1254 90 30\n",
      "36\t52178\t0.0\t3\t49799 2077 225 77\n",
      "37\t49601\t0.0\t3\t47267 2056 207 71\n",
      "38\t48514\t0.0\t3\t46231 1975 226 82\n",
      "39\t47670\t0.0\t3\t45390 1985 221 74\n",
      "40\t47812\t0.0\t3\t45647 1883 197 85\n",
      "41\t46299\t0.0\t3\t44224 1848 158 69\n",
      "42\t45533\t0.0\t3\t43377 1879 205 72\n",
      "43\t42342\t0.0\t3\t40420 1694 165 63\n",
      "44\t41134\t0.0\t3\t39052 1774 227 81\n",
      "45\t40678\t0.0\t3\t37753 2626 213 86\n",
      "46\t89212\t0.0\t3\t85864 2867 332 149\n",
      "47\t39418\t0.0\t3\t38027 1180 151 60\n",
      "48\t1115\t0.0\t3\t183 889 37 6\n",
      "49\t42406\t0.0\t3\t41281 973 111 41\n",
      "50\t9720\t0.0\t3\t8513 1117 65 25\n",
      "51\t46658\t0.0\t3\t45591 940 89 38\n",
      "52\t1756\t0.0\t3\t1162 514 62 18\n",
      "53\t18610\t0.0\t3\t17371 1149 62 28\n",
      "54\t42094\t0.0\t3\t41032 932 88 42\n",
      "55\t1635\t0.0\t3\t1059 502 55 19\n",
      "56\t19062\t0.0\t3\t18273 703 65 21\n",
      "57\t18174\t0.0\t3\t17361 738 60 15\n",
      "58\t17628\t0.0\t3\t16830 701 64 33\n",
      "59\t16268\t0.0\t3\t15501 671 70 26\n",
      "60\t14562\t0.0\t3\t13811 684 47 20\n",
      "61\t17396\t0.0\t3\t16768 542 63 23\n",
      "62\t7442\t0.0\t3\t7019 374 31 18\n",
      "63\t10434\t0.0\t3\t9934 437 41 22\n",
      "64\t10794\t0.0\t3\t10157 578 42 17\n",
      "65\t16187\t0.0\t3\t15469 646 51 21\n",
      "66\t11394\t0.0\t3\t11026 313 35 20\n",
      "67\t2620\t0.0\t3\t2289 264 38 29\n",
      "68\t7591\t0.0\t3\t7075 343 71 102\n",
      "69\t8885\t0.0\t3\t8050 539 152 144\n",
      "70\t14929\t0.0\t3\t13602 1017 275 35\n",
      "71\t22968\t0.0\t3\t21538 1323 82 25\n",
      "72\t21719\t0.0\t3\t20934 718 45 22\n",
      "73\t2748\t0.0\t3\t2501 210 22 15\n",
      "74\t469\t0.0\t3\t383 56 17 13\n",
      "75\t155\t0.0\t3\t118 28 6 3\n",
      "76\t63\t0.0\t3\t33 16 5 9\n",
      "77\t26\t0.0\t3\t13 5 5 3\n",
      "78\t17\t0.0\t3\t6 4 3 4\n",
      "79\t22\t0.0\t3\t8 4 4 6\n",
      "80\t9\t0.0\t3\t4 2 3\n",
      "81\t9\t0.0\t3\t2 1 4 2\n",
      "82\t5\t0.0\t3\t2 2 1\n",
      "83\t8\t0.0\t3\t4 0 4\n",
      "84\t3\t0.0\t3\t2 1\n",
      "86\t3\t0.0\t3\t0 0 1 2\n",
      "87\t1\t0.0\t3\t0 0 0 1\n",
      "88\t4\t0.0\t3\t1 1 1 1\n",
      "92\t1\t0.0\t3\t0 1\n",
      "94\t1\t0.0\t3\t1\n",
      "96\t1\t0.0\t3\t1\n",
      "98\t1\t0.0\t3\t0 0 0 1\n",
      "101\t1\t0.0\t3\t1\n",
      "103\t1\t0.0\t3\t0 1\n",
      "104\t1\t0.0\t3\t1\n",
      "105\t4\t0.0\t3\t1 1 2\n",
      "107\t1\t0.0\t3\t1\n",
      "109\t1\t0.0\t3\t0 1\n",
      "110\t1\t0.0\t3\t0 0 1\n",
      "111\t1\t0.0\t3\t0 0 0 1\n",
      "112\t3\t0.0\t3\t1 1 0 1\n",
      "113\t1\t0.0\t3\t0 0 0 1\n",
      "115\t2\t0.0\t3\t1 0 1\n",
      "116\t1\t0.0\t3\t1\n",
      "117\t1\t0.0\t3\t0 1\n",
      "120\t2\t0.0\t3\t1 1\n",
      "121\t2\t0.0\t3\t0 1 0 1\n",
      "124\t3\t0.0\t3\t2 1\n",
      "125\t1\t0.0\t3\t0 0 0 1\n",
      "126\t1\t0.0\t3\t1\n",
      "129\t1\t0.0\t3\t0 1\n",
      "131\t1\t0.0\t3\t1\n",
      "132\t5\t0.0\t3\t2 1 2\n",
      "137\t2\t0.0\t3\t2\n",
      "138\t1\t0.0\t3\t1\n",
      "139\t1\t0.0\t3\t1\n",
      "140\t3\t0.0\t3\t0 1 1 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 3263630 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.4%\n",
      "  C: 27.4%\n",
      "  G: 47.1%\n",
      "  T: 10.1%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t258276\t194093.5\t0\t258276\n",
      "4\t104227\t48523.4\t0\t104227\n",
      "5\t205286\t12130.8\t0\t205286\n",
      "8\t63846\t189.5\t0\t63846\n",
      "9\t127322\t47.4\t0\t127322\n",
      "10\t769\t11.8\t1\t0 769\n",
      "11\t133457\t3.0\t1\t132018 1439\n",
      "12\t788\t0.7\t1\t0 788\n",
      "13\t129818\t0.2\t1\t128177 1641\n",
      "14\t564\t0.0\t1\t0 564\n",
      "15\t64744\t0.0\t1\t62977 1767\n",
      "16\t128183\t0.0\t1\t126297 1886\n",
      "17\t1418\t0.0\t1\t0 1418\n",
      "18\t143288\t0.0\t1\t140885 2388 15\n",
      "19\t1199\t0.0\t1\t0 1071 128\n",
      "20\t78176\t0.0\t2\t73621 4268 287\n",
      "21\t232768\t0.0\t2\t208376 22547 1845\n",
      "22\t2650\t0.0\t2\t0 1906 744\n",
      "23\t223\t0.0\t2\t0 92 131\n",
      "24\t1242\t0.0\t2\t0 1138 104\n",
      "25\t57098\t0.0\t2\t47596 8163 1339\n",
      "26\t60077\t0.0\t2\t47032 10356 2689\n",
      "27\t86792\t0.0\t2\t70636 12565 3591\n",
      "28\t28223\t0.0\t2\t0 23745 4475 3\n",
      "29\t115784\t0.0\t2\t83390 25034 7338 22\n",
      "30\t7440\t0.0\t3\t0 4018 2358 1064\n",
      "31\t101247\t0.0\t3\t72616 21088 5885 1658\n",
      "32\t3085\t0.0\t3\t0 1850 767 468\n",
      "33\t55743\t0.0\t3\t37228 12753 4016 1746\n",
      "34\t54646\t0.0\t3\t36715 12321 3945 1665\n",
      "35\t53814\t0.0\t3\t35912 12281 3935 1686\n",
      "36\t50404\t0.0\t3\t32868 11439 4207 1890\n",
      "37\t46920\t0.0\t3\t32351 10260 3081 1228\n",
      "38\t47551\t0.0\t3\t30920 11323 3646 1662\n",
      "39\t67342\t0.0\t3\t46811 13878 4613 2040\n",
      "40\t24615\t0.0\t3\t13526 7328 2529 1232\n",
      "41\t82297\t0.0\t3\t57838 16935 5230 2294\n",
      "42\t4797\t0.0\t3\t1033 2449 814 501\n",
      "43\t68637\t0.0\t3\t48223 13919 4540 1955\n",
      "44\t10428\t0.0\t3\t5491 3242 1177 518\n",
      "45\t39042\t0.0\t3\t26541 8847 2571 1083\n",
      "46\t65863\t0.0\t3\t46249 13479 4380 1755\n",
      "47\t4562\t0.0\t3\t1672 1760 730 400\n",
      "48\t31895\t0.0\t3\t22217 6791 2041 846\n",
      "49\t29738\t0.0\t3\t20056 6714 2077 891\n",
      "50\t27473\t0.0\t3\t18696 6057 1859 861\n",
      "51\t27721\t0.0\t3\t18394 6700 1872 755\n",
      "52\t62325\t0.0\t3\t46450 11678 3047 1150\n",
      "53\t9635\t0.0\t3\t4747 2751 1414 723\n",
      "54\t1608\t0.0\t3\t124 1175 203 106\n",
      "55\t55782\t0.0\t3\t40451 10630 3374 1327\n",
      "56\t3659\t0.0\t3\t1606 1082 616 355\n",
      "57\t610\t0.0\t3\t61 374 118 57\n",
      "58\t17364\t0.0\t3\t11924 3835 1123 482\n",
      "59\t26566\t0.0\t3\t19080 5250 1594 642\n",
      "60\t2968\t0.0\t3\t1353 1024 384 207\n",
      "61\t12490\t0.0\t3\t8705 2750 724 311\n",
      "62\t19912\t0.0\t3\t14520 3845 1096 451\n",
      "63\t1974\t0.0\t3\t892 662 278 142\n",
      "64\t10862\t0.0\t3\t7918 2130 568 246\n",
      "65\t9484\t0.0\t3\t6935 1850 507 192\n",
      "66\t8718\t0.0\t3\t6211 1834 480 193\n",
      "67\t8004\t0.0\t3\t5630 1708 467 199\n",
      "68\t6986\t0.0\t3\t4831 1528 430 197\n",
      "69\t6166\t0.0\t3\t4338 1274 359 195\n",
      "70\t5694\t0.0\t3\t4052 1159 352 131\n",
      "71\t5075\t0.0\t3\t3529 1052 338 156\n",
      "72\t4652\t0.0\t3\t3169 980 320 183\n",
      "73\t4402\t0.0\t3\t2871 888 358 285\n",
      "74\t4738\t0.0\t3\t2706 1318 526 188\n",
      "75\t13298\t0.0\t3\t9291 2932 769 306\n",
      "76\t24982\t0.0\t3\t18468 4516 1398 600\n",
      "77\t3439\t0.0\t3\t1761 878 501 299\n",
      "78\t432\t0.0\t3\t130 123 93 86\n",
      "79\t110\t0.0\t3\t22 30 32 26\n",
      "80\t39\t0.0\t3\t8 12 8 11\n",
      "81\t20\t0.0\t3\t1 1 10 8\n",
      "82\t11\t0.0\t3\t3 2 1 5\n",
      "83\t12\t0.0\t3\t1 2 3 6\n",
      "84\t6\t0.0\t3\t0 0 5 1\n",
      "85\t9\t0.0\t3\t3 2 3 1\n",
      "86\t4\t0.0\t3\t0 1 2 1\n",
      "87\t3\t0.0\t3\t0 1 1 1\n",
      "88\t3\t0.0\t3\t1 1 0 1\n",
      "89\t6\t0.0\t3\t2 0 3 1\n",
      "90\t1\t0.0\t3\t0 0 1\n",
      "91\t5\t0.0\t3\t2 1 0 2\n",
      "92\t6\t0.0\t3\t1 1 2 2\n",
      "93\t3\t0.0\t3\t1 0 1 1\n",
      "95\t1\t0.0\t3\t0 0 0 1\n",
      "96\t1\t0.0\t3\t1\n",
      "97\t1\t0.0\t3\t1\n",
      "98\t1\t0.0\t3\t1\n",
      "99\t2\t0.0\t3\t1 1\n",
      "100\t1\t0.0\t3\t0 0 0 1\n",
      "102\t5\t0.0\t3\t1 1 2 1\n",
      "104\t2\t0.0\t3\t0 0 1 1\n",
      "105\t5\t0.0\t3\t0 0 4 1\n",
      "106\t3\t0.0\t3\t0 1 0 2\n",
      "107\t3\t0.0\t3\t0 1 2\n",
      "108\t6\t0.0\t3\t2 3 1\n",
      "109\t13\t0.0\t3\t5 5 2 1\n",
      "110\t6\t0.0\t3\t3 1 2\n",
      "111\t1\t0.0\t3\t0 0 1\n",
      "112\t1\t0.0\t3\t0 0 1\n",
      "114\t2\t0.0\t3\t0 0 0 2\n",
      "115\t1\t0.0\t3\t0 1\n",
      "117\t1\t0.0\t3\t0 1\n",
      "118\t1\t0.0\t3\t0 0 0 1\n",
      "119\t2\t0.0\t3\t0 0 0 2\n",
      "121\t1\t0.0\t3\t0 0 1\n",
      "122\t1\t0.0\t3\t0 0 1\n",
      "123\t1\t0.0\t3\t0 0 1\n",
      "124\t2\t0.0\t3\t2\n",
      "125\t1\t0.0\t3\t0 0 0 1\n",
      "127\t1\t0.0\t3\t1\n",
      "130\t3\t0.0\t3\t1 1 1\n",
      "131\t2\t0.0\t3\t0 0 2\n",
      "136\t1\t0.0\t3\t1\n",
      "137\t1\t0.0\t3\t1\n",
      "138\t5\t0.0\t3\t2 1 0 2\n",
      "139\t2\t0.0\t3\t1 0 1\n",
      "140\t7\t0.0\t3\t2 3 1 1\n",
      "141\t2\t0.0\t3\t0 0 2\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_02_1.fastq.gz -p ./fastq_trimmed_sub/Int_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-02_S43_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-02_S43_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:34    15,738,415 reads @   2.2 µs/read;  27.68 M reads/minute\n",
      "Finished in 34.205 s (2.173 µs/read; 27.61 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         15,738,415\n",
      "  Read 1 with adapter:               3,040,184 (19.3%)\n",
      "  Read 2 with adapter:               2,981,250 (18.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             679,775 (4.3%)\n",
      "Pairs written (passing filters):    15,058,640 (95.7%)\n",
      "\n",
      "Total basepairs processed: 4,753,001,330 bp\n",
      "  Read 1: 2,376,500,665 bp\n",
      "  Read 2: 2,376,500,665 bp\n",
      "Quality-trimmed:             164,494,985 bp (3.5%)\n",
      "  Read 1:    57,747,262 bp\n",
      "  Read 2:   106,747,723 bp\n",
      "Total written (filtered):  4,080,075,957 bp (85.8%)\n",
      "  Read 1: 2,038,968,771 bp\n",
      "  Read 2: 2,041,107,186 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 3040184 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.0%\n",
      "  C: 31.1%\n",
      "  G: 35.3%\n",
      "  T: 18.5%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t345576\t245912.7\t0\t345576\n",
      "4\t126642\t61478.2\t0\t126642\n",
      "5\t204614\t15369.5\t0\t204614\n",
      "8\t61659\t240.1\t0\t61659\n",
      "9\t122416\t60.0\t0\t122416\n",
      "10\t1119\t15.0\t1\t0 1119\n",
      "11\t124951\t3.8\t1\t122862 2089\n",
      "12\t658\t0.9\t1\t0 658\n",
      "13\t61441\t0.2\t1\t59831 1610\n",
      "14\t61333\t0.1\t1\t59604 1729\n",
      "15\t60393\t0.0\t1\t58576 1817\n",
      "16\t61004\t0.0\t1\t58423 2581\n",
      "17\t116080\t0.0\t1\t113225 2855\n",
      "18\t1018\t0.0\t1\t0 1001 17\n",
      "19\t58325\t0.0\t1\t55977 2238 110\n",
      "20\t60124\t0.0\t2\t56498 3315 311\n",
      "21\t113769\t0.0\t2\t110288 2999 482\n",
      "22\t1321\t0.0\t2\t0 1194 127\n",
      "23\t57400\t0.0\t2\t54561 2557 282\n",
      "24\t56359\t0.0\t2\t53341 2674 344\n",
      "25\t54733\t0.0\t2\t51756 2671 306\n",
      "26\t53535\t0.0\t2\t50265 2945 325\n",
      "27\t52419\t0.0\t2\t49235 2788 396\n",
      "28\t52425\t0.0\t2\t48068 4030 319 8\n",
      "29\t100062\t0.0\t2\t96017 3463 572 10\n",
      "30\t1743\t0.0\t3\t0 1548 147 48\n",
      "31\t49946\t0.0\t3\t46613 2833 360 140\n",
      "32\t48552\t0.0\t3\t45252 2865 336 99\n",
      "33\t48241\t0.0\t3\t44797 2959 360 125\n",
      "34\t47101\t0.0\t3\t43762 2873 340 126\n",
      "35\t45234\t0.0\t3\t42087 2708 316 123\n",
      "36\t43127\t0.0\t3\t40077 2561 337 152\n",
      "37\t40656\t0.0\t3\t37881 2379 265 131\n",
      "38\t39783\t0.0\t3\t36997 2400 296 90\n",
      "39\t39035\t0.0\t3\t36462 2189 279 105\n",
      "40\t38350\t0.0\t3\t35701 2268 268 113\n",
      "41\t37025\t0.0\t3\t34514 2152 274 85\n",
      "42\t36286\t0.0\t3\t33832 2093 270 91\n",
      "43\t33985\t0.0\t3\t31704 1974 216 91\n",
      "44\t33602\t0.0\t3\t31121 2066 321 94\n",
      "45\t34534\t0.0\t3\t30091 4037 304 102\n",
      "46\t112102\t0.0\t3\t106555 4754 537 256\n",
      "47\t38667\t0.0\t3\t36786 1589 226 66\n",
      "48\t1792\t0.0\t3\t314 1384 77 17\n",
      "49\t54563\t0.0\t3\t52783 1601 127 52\n",
      "50\t1892\t0.0\t3\t1396 408 65 23\n",
      "51\t6903\t0.0\t3\t6456 344 73 30\n",
      "52\t1102\t0.0\t3\t808 212 63 19\n",
      "53\t4452\t0.0\t3\t3835 535 48 34\n",
      "54\t14393\t0.0\t3\t13636 621 102 34\n",
      "55\t2026\t0.0\t3\t1596 358 46 26\n",
      "56\t8393\t0.0\t3\t7768 526 74 25\n",
      "57\t10037\t0.0\t3\t9318 634 69 16\n",
      "58\t12367\t0.0\t3\t11546 729 62 30\n",
      "59\t12110\t0.0\t3\t11349 668 71 22\n",
      "60\t11353\t0.0\t3\t10531 732 59 31\n",
      "61\t11789\t0.0\t3\t11090 605 66 28\n",
      "62\t8431\t0.0\t3\t7877 483 47 24\n",
      "63\t7356\t0.0\t3\t6800 438 61 57\n",
      "64\t7934\t0.0\t3\t7223 560 113 38\n",
      "65\t10836\t0.0\t3\t9684 1040 87 25\n",
      "66\t21486\t0.0\t3\t20588 816 52 30\n",
      "67\t4738\t0.0\t3\t4361 292 35 50\n",
      "68\t2979\t0.0\t3\t2454 322 78 125\n",
      "69\t6467\t0.0\t3\t5565 578 165 159\n",
      "70\t9893\t0.0\t3\t8650 983 203 57\n",
      "71\t14008\t0.0\t3\t12646 1226 102 34\n",
      "72\t11434\t0.0\t3\t10700 637 72 25\n",
      "73\t2740\t0.0\t3\t2452 248 32 8\n",
      "74\t659\t0.0\t3\t525 92 30 12\n",
      "75\t245\t0.0\t3\t176 44 15 10\n",
      "76\t140\t0.0\t3\t91 34 12 3\n",
      "77\t76\t0.0\t3\t37 22 9 8\n",
      "78\t45\t0.0\t3\t20 16 5 4\n",
      "79\t26\t0.0\t3\t10 8 6 2\n",
      "80\t23\t0.0\t3\t9 11 1 2\n",
      "81\t13\t0.0\t3\t2 6 2 3\n",
      "82\t4\t0.0\t3\t3 0 0 1\n",
      "83\t7\t0.0\t3\t4 1 2\n",
      "84\t9\t0.0\t3\t3 1 2 3\n",
      "85\t6\t0.0\t3\t2 0 1 3\n",
      "86\t6\t0.0\t3\t2 4\n",
      "87\t2\t0.0\t3\t2\n",
      "88\t7\t0.0\t3\t0 5 1 1\n",
      "89\t8\t0.0\t3\t2 1 1 4\n",
      "90\t3\t0.0\t3\t2 0 1\n",
      "91\t7\t0.0\t3\t0 2 3 2\n",
      "92\t2\t0.0\t3\t1 0 1\n",
      "93\t2\t0.0\t3\t0 0 0 2\n",
      "94\t3\t0.0\t3\t2 0 0 1\n",
      "95\t3\t0.0\t3\t2 0 1\n",
      "98\t1\t0.0\t3\t1\n",
      "99\t2\t0.0\t3\t0 0 1 1\n",
      "100\t2\t0.0\t3\t1 1\n",
      "101\t1\t0.0\t3\t1\n",
      "102\t2\t0.0\t3\t0 1 0 1\n",
      "103\t5\t0.0\t3\t1 1 3\n",
      "105\t2\t0.0\t3\t0 2\n",
      "106\t2\t0.0\t3\t0 0 2\n",
      "107\t2\t0.0\t3\t2\n",
      "108\t2\t0.0\t3\t1 1\n",
      "109\t3\t0.0\t3\t1 2\n",
      "110\t4\t0.0\t3\t1 1 0 2\n",
      "112\t3\t0.0\t3\t1 0 1 1\n",
      "113\t1\t0.0\t3\t0 0 0 1\n",
      "114\t1\t0.0\t3\t0 0 1\n",
      "115\t2\t0.0\t3\t1 0 0 1\n",
      "116\t2\t0.0\t3\t0 1 1\n",
      "118\t3\t0.0\t3\t2 0 1\n",
      "119\t2\t0.0\t3\t0 0 0 2\n",
      "120\t1\t0.0\t3\t0 0 0 1\n",
      "121\t1\t0.0\t3\t0 0 1\n",
      "122\t2\t0.0\t3\t1 0 1\n",
      "124\t2\t0.0\t3\t0 0 2\n",
      "125\t1\t0.0\t3\t1\n",
      "126\t1\t0.0\t3\t0 0 1\n",
      "130\t1\t0.0\t3\t0 0 0 1\n",
      "131\t1\t0.0\t3\t1\n",
      "132\t2\t0.0\t3\t1 0 1\n",
      "133\t1\t0.0\t3\t0 0 0 1\n",
      "135\t1\t0.0\t3\t1\n",
      "136\t2\t0.0\t3\t1 1\n",
      "137\t1\t0.0\t3\t0 1\n",
      "138\t3\t0.0\t3\t2 0 1\n",
      "139\t3\t0.0\t3\t2 0 1\n",
      "140\t3\t0.0\t3\t1 1 1\n",
      "141\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 2981250 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.2%\n",
      "  C: 30.0%\n",
      "  G: 42.1%\n",
      "  T: 10.7%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t325108\t245912.7\t0\t325108\n",
      "4\t129058\t61478.2\t0\t129058\n",
      "5\t206667\t15369.5\t0\t206667\n",
      "8\t61900\t240.1\t0\t61900\n",
      "9\t122280\t60.0\t0\t122280\n",
      "10\t981\t15.0\t1\t0 981\n",
      "11\t124932\t3.8\t1\t123121 1811\n",
      "12\t1010\t0.9\t1\t0 1010\n",
      "13\t121066\t0.2\t1\t118967 2099\n",
      "14\t642\t0.1\t1\t0 642\n",
      "15\t60727\t0.0\t1\t58675 2052\n",
      "16\t117569\t0.0\t1\t115155 2414\n",
      "17\t1610\t0.0\t1\t0 1610\n",
      "18\t122634\t0.0\t1\t119785 2823 26\n",
      "19\t1303\t0.0\t1\t0 1136 167\n",
      "20\t74015\t0.0\t2\t68668 4944 403\n",
      "21\t204977\t0.0\t2\t179248 23907 1822\n",
      "22\t3106\t0.0\t2\t0 2451 655\n",
      "23\t361\t0.0\t2\t0 154 207\n",
      "24\t1270\t0.0\t2\t0 1138 132\n",
      "25\t50066\t0.0\t2\t40980 7983 1103\n",
      "26\t53330\t0.0\t2\t40939 10109 2282\n",
      "27\t79304\t0.0\t2\t63224 12684 3396\n",
      "28\t23226\t0.0\t2\t0 19150 4065 11\n",
      "29\t98108\t0.0\t2\t70343 21732 6022 11\n",
      "30\t5296\t0.0\t3\t0 3053 1583 660\n",
      "31\t85928\t0.0\t3\t61018 18486 5003 1421\n",
      "32\t3603\t0.0\t3\t0 2043 979 581\n",
      "33\t46761\t0.0\t3\t31033 11062 3286 1380\n",
      "34\t45556\t0.0\t3\t30263 10554 3326 1413\n",
      "35\t44011\t0.0\t3\t29137 10446 3153 1275\n",
      "36\t41249\t0.0\t3\t26665 9831 3260 1493\n",
      "37\t39378\t0.0\t3\t26516 9249 2545 1068\n",
      "38\t56419\t0.0\t3\t38725 12184 3914 1596\n",
      "39\t21058\t0.0\t3\t10858 6839 2225 1136\n",
      "40\t100224\t0.0\t3\t69791 21577 6276 2580\n",
      "41\t3825\t0.0\t3\t1203 1126 874 622\n",
      "42\t1837\t0.0\t3\t107 1413 202 115\n",
      "43\t54413\t0.0\t3\t38075 11505 3346 1487\n",
      "44\t9688\t0.0\t3\t4992 3025 1158 513\n",
      "45\t32438\t0.0\t3\t21540 7792 2113 993\n",
      "46\t51546\t0.0\t3\t35836 11100 3321 1289\n",
      "47\t2576\t0.0\t3\t820 1121 435 200\n",
      "48\t24441\t0.0\t3\t16988 5416 1491 546\n",
      "49\t23608\t0.0\t3\t15666 5693 1622 627\n",
      "50\t21883\t0.0\t3\t14694 4920 1590 679\n",
      "51\t22220\t0.0\t3\t14659 5594 1420 547\n",
      "52\t48672\t0.0\t3\t35962 9514 2324 872\n",
      "53\t7111\t0.0\t3\t3463 2005 1105 538\n",
      "54\t1531\t0.0\t3\t150 1102 165 114\n",
      "55\t42606\t0.0\t3\t30711 8577 2340 978\n",
      "56\t2598\t0.0\t3\t1014 804 483 297\n",
      "57\t634\t0.0\t3\t69 392 120 53\n",
      "58\t13139\t0.0\t3\t8840 3160 791 348\n",
      "59\t20640\t0.0\t3\t14697 4245 1213 485\n",
      "60\t1851\t0.0\t3\t716 680 275 180\n",
      "61\t9829\t0.0\t3\t6809 2211 546 263\n",
      "62\t15296\t0.0\t3\t11169 2976 816 335\n",
      "63\t1737\t0.0\t3\t785 595 247 110\n",
      "64\t7939\t0.0\t3\t5712 1644 439 144\n",
      "65\t6892\t0.0\t3\t4947 1393 395 157\n",
      "66\t6504\t0.0\t3\t4641 1408 322 133\n",
      "67\t5757\t0.0\t3\t4023 1267 333 134\n",
      "68\t5260\t0.0\t3\t3601 1206 314 139\n",
      "69\t4571\t0.0\t3\t3199 1003 251 118\n",
      "70\t4172\t0.0\t3\t2918 916 247 91\n",
      "71\t3747\t0.0\t3\t2572 802 262 111\n",
      "72\t3460\t0.0\t3\t2347 788 209 116\n",
      "73\t3488\t0.0\t3\t2153 718 258 359\n",
      "74\t3826\t0.0\t3\t2065 1115 480 166\n",
      "75\t10233\t0.0\t3\t6537 2769 675 252\n",
      "76\t21835\t0.0\t3\t16204 4033 1110 488\n",
      "77\t3794\t0.0\t3\t2116 979 421 278\n",
      "78\t486\t0.0\t3\t162 145 92 87\n",
      "79\t153\t0.0\t3\t29 47 40 37\n",
      "80\t64\t0.0\t3\t5 15 21 23\n",
      "81\t28\t0.0\t3\t3 3 12 10\n",
      "82\t13\t0.0\t3\t0 3 6 4\n",
      "83\t11\t0.0\t3\t0 3 4 4\n",
      "84\t6\t0.0\t3\t2 1 1 2\n",
      "85\t11\t0.0\t3\t1 5 1 4\n",
      "86\t4\t0.0\t3\t0 0 3 1\n",
      "87\t9\t0.0\t3\t1 0 5 3\n",
      "88\t3\t0.0\t3\t0 1 1 1\n",
      "89\t10\t0.0\t3\t3 2 2 3\n",
      "91\t3\t0.0\t3\t0 1 0 2\n",
      "92\t3\t0.0\t3\t2 0 0 1\n",
      "93\t3\t0.0\t3\t0 0 3\n",
      "94\t5\t0.0\t3\t1 0 1 3\n",
      "95\t3\t0.0\t3\t0 0 1 2\n",
      "96\t2\t0.0\t3\t0 0 2\n",
      "97\t4\t0.0\t3\t0 0 3 1\n",
      "98\t3\t0.0\t3\t0 0 1 2\n",
      "99\t3\t0.0\t3\t0 3\n",
      "100\t2\t0.0\t3\t0 1 0 1\n",
      "101\t1\t0.0\t3\t0 0 1\n",
      "102\t1\t0.0\t3\t0 1\n",
      "103\t6\t0.0\t3\t2 1 2 1\n",
      "104\t1\t0.0\t3\t0 1\n",
      "105\t3\t0.0\t3\t1 1 1\n",
      "106\t3\t0.0\t3\t0 0 1 2\n",
      "107\t4\t0.0\t3\t1 1 1 1\n",
      "108\t6\t0.0\t3\t2 4\n",
      "109\t9\t0.0\t3\t6 0 3\n",
      "110\t7\t0.0\t3\t2 2 2 1\n",
      "111\t2\t0.0\t3\t1 1\n",
      "112\t1\t0.0\t3\t0 1\n",
      "113\t4\t0.0\t3\t0 0 1 3\n",
      "117\t1\t0.0\t3\t0 1\n",
      "119\t1\t0.0\t3\t0 0 0 1\n",
      "120\t1\t0.0\t3\t0 0 0 1\n",
      "121\t1\t0.0\t3\t1\n",
      "122\t1\t0.0\t3\t0 0 0 1\n",
      "123\t2\t0.0\t3\t0 0 0 2\n",
      "124\t1\t0.0\t3\t0 0 1\n",
      "125\t2\t0.0\t3\t0 0 1 1\n",
      "126\t3\t0.0\t3\t2 0 1\n",
      "127\t4\t0.0\t3\t1 2 0 1\n",
      "128\t1\t0.0\t3\t0 0 0 1\n",
      "130\t1\t0.0\t3\t0 0 0 1\n",
      "132\t1\t0.0\t3\t0 0 1\n",
      "133\t2\t0.0\t3\t0 0 2\n",
      "134\t1\t0.0\t3\t0 0 0 1\n",
      "135\t1\t0.0\t3\t0 0 0 1\n",
      "136\t1\t0.0\t3\t0 0 0 1\n",
      "137\t2\t0.0\t3\t0 2\n",
      "138\t5\t0.0\t3\t1 1 0 3\n",
      "139\t3\t0.0\t3\t2 0 1\n",
      "140\t3\t0.0\t3\t0 1 1 1\n",
      "141\t5\t0.0\t3\t0 2 1 2\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_03_1.fastq.gz -p ./fastq_trimmed_sub/Int_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-03_S44_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-03_S44_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:30    13,521,806 reads @   2.2 µs/read;  26.78 M reads/minute\n",
      "Finished in 30.363 s (2.246 µs/read; 26.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         13,521,806\n",
      "  Read 1 with adapter:               2,871,566 (21.2%)\n",
      "  Read 2 with adapter:               2,842,705 (21.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             318,271 (2.4%)\n",
      "Pairs written (passing filters):    13,203,535 (97.6%)\n",
      "\n",
      "Total basepairs processed: 4,083,585,412 bp\n",
      "  Read 1: 2,041,792,706 bp\n",
      "  Read 2: 2,041,792,706 bp\n",
      "Quality-trimmed:              78,937,636 bp (1.9%)\n",
      "  Read 1:    27,030,073 bp\n",
      "  Read 2:    51,907,563 bp\n",
      "Total written (filtered):  3,567,297,192 bp (87.4%)\n",
      "  Read 1: 1,782,572,061 bp\n",
      "  Read 2: 1,784,725,131 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 2871566 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.5%\n",
      "  C: 30.7%\n",
      "  G: 35.9%\n",
      "  T: 18.8%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t265627\t211278.2\t0\t265627\n",
      "4\t114508\t52819.6\t0\t114508\n",
      "5\t189021\t13204.9\t0\t189021\n",
      "8\t57250\t206.3\t0\t57250\n",
      "9\t114150\t51.6\t0\t114150\n",
      "10\t1081\t12.9\t1\t0 1081\n",
      "11\t116335\t3.2\t1\t114655 1680\n",
      "12\t694\t0.8\t1\t0 694\n",
      "13\t57481\t0.2\t1\t56048 1433\n",
      "14\t57229\t0.1\t1\t55611 1618\n",
      "15\t56488\t0.0\t1\t54822 1666\n",
      "16\t56909\t0.0\t1\t54291 2618\n",
      "17\t109463\t0.0\t1\t107027 2436\n",
      "18\t1062\t0.0\t1\t0 1026 36\n",
      "19\t54384\t0.0\t1\t52242 2041 101\n",
      "20\t56382\t0.0\t2\t52783 3325 274\n",
      "21\t107765\t0.0\t2\t104820 2510 435\n",
      "22\t1372\t0.0\t2\t0 1236 136\n",
      "23\t54415\t0.0\t2\t51772 2324 319\n",
      "24\t54443\t0.0\t2\t51535 2590 318\n",
      "25\t52418\t0.0\t2\t49510 2585 323\n",
      "26\t51133\t0.0\t2\t48069 2719 345\n",
      "27\t50403\t0.0\t2\t47407 2580 416\n",
      "28\t51137\t0.0\t2\t46677 4160 295 5\n",
      "29\t94893\t0.0\t2\t91490 2889 503 11\n",
      "30\t1888\t0.0\t3\t0 1634 188 66\n",
      "31\t48459\t0.0\t3\t45190 2729 411 129\n",
      "32\t48082\t0.0\t3\t44717 2877 374 114\n",
      "33\t46589\t0.0\t3\t43050 2963 427 149\n",
      "34\t46357\t0.0\t3\t43016 2828 387 126\n",
      "35\t45313\t0.0\t3\t41074 3774 338 127\n",
      "36\t80687\t0.0\t3\t77412 2636 456 183\n",
      "37\t1600\t0.0\t3\t100 1333 133 34\n",
      "38\t39628\t0.0\t3\t36933 2248 326 121\n",
      "39\t39279\t0.0\t3\t36561 2342 263 113\n",
      "40\t37807\t0.0\t3\t35186 2259 279 83\n",
      "41\t36959\t0.0\t3\t34363 2199 305 92\n",
      "42\t36339\t0.0\t3\t33930 2041 278 90\n",
      "43\t35539\t0.0\t3\t32936 2201 254 148\n",
      "44\t34856\t0.0\t3\t31936 2187 619 114\n",
      "45\t40132\t0.0\t3\t31555 7968 465 144\n",
      "46\t224259\t0.0\t3\t215839 7338 760 322\n",
      "47\t51460\t0.0\t3\t49030 2114 212 104\n",
      "48\t713\t0.0\t3\t389 268 41 15\n",
      "49\t1296\t0.0\t3\t1115 112 45 24\n",
      "50\t570\t0.0\t3\t408 100 36 26\n",
      "51\t1077\t0.0\t3\t897 124 39 17\n",
      "52\t479\t0.0\t3\t339 88 35 17\n",
      "53\t1280\t0.0\t3\t1012 216 30 22\n",
      "54\t2694\t0.0\t3\t2360 242 63 29\n",
      "55\t1070\t0.0\t3\t808 188 46 28\n",
      "56\t2115\t0.0\t3\t1824 219 59 13\n",
      "57\t3224\t0.0\t3\t2756 386 66 16\n",
      "58\t7981\t0.0\t3\t7294 591 67 29\n",
      "59\t12207\t0.0\t3\t11421 674 89 23\n",
      "60\t11100\t0.0\t3\t10124 846 97 33\n",
      "61\t15691\t0.0\t3\t14910 683 66 32\n",
      "62\t7282\t0.0\t3\t6778 391 65 48\n",
      "63\t5276\t0.0\t3\t4669 399 73 135\n",
      "64\t7510\t0.0\t3\t6503 754 203 50\n",
      "65\t17007\t0.0\t3\t15287 1576 118 26\n",
      "66\t27782\t0.0\t3\t26589 1091 66 36\n",
      "67\t6142\t0.0\t3\t5727 328 57 30\n",
      "68\t1561\t0.0\t3\t1226 232 42 61\n",
      "69\t3417\t0.0\t3\t3017 287 79 34\n",
      "70\t3992\t0.0\t3\t3536 369 64 23\n",
      "71\t4160\t0.0\t3\t3798 297 53 12\n",
      "72\t2903\t0.0\t3\t2650 207 32 14\n",
      "73\t916\t0.0\t3\t782 101 22 11\n",
      "74\t332\t0.0\t3\t262 45 12 13\n",
      "75\t162\t0.0\t3\t102 37 19 4\n",
      "76\t88\t0.0\t3\t58 18 8 4\n",
      "77\t55\t0.0\t3\t32 16 5 2\n",
      "78\t35\t0.0\t3\t16 8 8 3\n",
      "79\t25\t0.0\t3\t12 10 1 2\n",
      "80\t19\t0.0\t3\t6 10 0 3\n",
      "81\t6\t0.0\t3\t1 0 2 3\n",
      "82\t10\t0.0\t3\t4 5 0 1\n",
      "83\t8\t0.0\t3\t3 1 3 1\n",
      "84\t6\t0.0\t3\t2 1 1 2\n",
      "85\t3\t0.0\t3\t0 1 1 1\n",
      "86\t6\t0.0\t3\t1 2 2 1\n",
      "87\t6\t0.0\t3\t2 1 0 3\n",
      "88\t6\t0.0\t3\t0 3 2 1\n",
      "89\t1\t0.0\t3\t1\n",
      "90\t6\t0.0\t3\t2 0 2 2\n",
      "91\t2\t0.0\t3\t0 0 0 2\n",
      "92\t2\t0.0\t3\t1 1\n",
      "94\t3\t0.0\t3\t1 1 1\n",
      "95\t4\t0.0\t3\t2 0 1 1\n",
      "96\t1\t0.0\t3\t0 0 1\n",
      "97\t3\t0.0\t3\t0 0 2 1\n",
      "98\t2\t0.0\t3\t0 0 0 2\n",
      "99\t6\t0.0\t3\t1 2 3\n",
      "100\t2\t0.0\t3\t0 0 0 2\n",
      "102\t1\t0.0\t3\t0 0 1\n",
      "103\t2\t0.0\t3\t0 1 1\n",
      "104\t1\t0.0\t3\t0 1\n",
      "105\t2\t0.0\t3\t1 0 1\n",
      "106\t3\t0.0\t3\t0 2 1\n",
      "107\t3\t0.0\t3\t0 1 0 2\n",
      "108\t1\t0.0\t3\t0 0 1\n",
      "109\t5\t0.0\t3\t0 1 2 2\n",
      "110\t1\t0.0\t3\t0 0 1\n",
      "111\t2\t0.0\t3\t0 0 2\n",
      "113\t1\t0.0\t3\t0 0 0 1\n",
      "114\t2\t0.0\t3\t0 0 1 1\n",
      "116\t1\t0.0\t3\t0 0 0 1\n",
      "121\t2\t0.0\t3\t1 0 0 1\n",
      "126\t1\t0.0\t3\t0 1\n",
      "128\t2\t0.0\t3\t1 1\n",
      "129\t2\t0.0\t3\t1 1\n",
      "130\t2\t0.0\t3\t0 1 1\n",
      "131\t1\t0.0\t3\t0 1\n",
      "132\t1\t0.0\t3\t0 1\n",
      "133\t2\t0.0\t3\t1 0 0 1\n",
      "134\t3\t0.0\t3\t0 3\n",
      "136\t2\t0.0\t3\t1 0 1\n",
      "138\t1\t0.0\t3\t0 1\n",
      "141\t2\t0.0\t3\t0 1 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 2842705 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.1%\n",
      "  C: 29.6%\n",
      "  G: 41.8%\n",
      "  T: 10.4%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t286116\t211278.2\t0\t286116\n",
      "4\t116236\t52819.6\t0\t116236\n",
      "5\t190916\t13204.9\t0\t190916\n",
      "8\t57422\t206.3\t0\t57422\n",
      "9\t114097\t51.6\t0\t114097\n",
      "10\t1015\t12.9\t1\t0 1015\n",
      "11\t116252\t3.2\t1\t114377 1875\n",
      "12\t986\t0.8\t1\t0 986\n",
      "13\t113014\t0.2\t1\t110793 2221\n",
      "14\t618\t0.1\t1\t0 618\n",
      "15\t56958\t0.0\t1\t54692 2266\n",
      "16\t110033\t0.0\t1\t107632 2401\n",
      "17\t1634\t0.0\t1\t0 1634\n",
      "18\t124338\t0.0\t1\t121168 3147 23\n",
      "19\t1286\t0.0\t1\t0 1128 158\n",
      "20\t66961\t0.0\t2\t61646 4975 340\n",
      "21\t190351\t0.0\t2\t167463 21125 1763\n",
      "22\t2338\t0.0\t2\t0 1672 666\n",
      "23\t349\t0.0\t2\t0 125 224\n",
      "24\t1336\t0.0\t2\t0 1210 126\n",
      "25\t47094\t0.0\t2\t38083 7707 1304\n",
      "26\t50677\t0.0\t2\t37897 10165 2615\n",
      "27\t77455\t0.0\t2\t60557 12970 3928\n",
      "28\t20620\t0.0\t2\t0 17008 3603 9\n",
      "29\t99663\t0.0\t2\t70336 21934 7372 21\n",
      "30\t6113\t0.0\t3\t0 3435 1829 849\n",
      "31\t78774\t0.0\t3\t56955 16176 4321 1322\n",
      "32\t2344\t0.0\t3\t0 1446 598 300\n",
      "33\t47178\t0.0\t3\t29949 11636 3749 1844\n",
      "34\t82172\t0.0\t3\t55892 17421 6145 2714\n",
      "35\t4854\t0.0\t3\t452 3404 614 384\n",
      "36\t138367\t0.0\t3\t94604 29021 9990 4752\n",
      "37\t4906\t0.0\t3\t1770 1474 1008 654\n",
      "38\t824\t0.0\t3\t87 524 133 80\n",
      "39\t7512\t0.0\t3\t3874 2455 787 396\n",
      "40\t36009\t0.0\t3\t23302 8624 2723 1360\n",
      "41\t38569\t0.0\t3\t25060 8843 3137 1529\n",
      "42\t62316\t0.0\t3\t41004 14145 4828 2339\n",
      "43\t28868\t0.0\t3\t19877 6088 1902 1001\n",
      "44\t6842\t0.0\t3\t3630 2230 682 300\n",
      "45\t43575\t0.0\t3\t29400 9752 3058 1365\n",
      "46\t38352\t0.0\t3\t26253 8224 2685 1190\n",
      "47\t5897\t0.0\t3\t2813 1920 780 384\n",
      "48\t27578\t0.0\t3\t18986 5931 1873 788\n",
      "49\t20774\t0.0\t3\t13840 4842 1444 648\n",
      "50\t22593\t0.0\t3\t15056 5041 1701 795\n",
      "51\t23036\t0.0\t3\t15163 5545 1590 738\n",
      "52\t25570\t0.0\t3\t17364 5592 1811 803\n",
      "53\t31344\t0.0\t3\t22084 6091 2174 995\n",
      "54\t4717\t0.0\t3\t2037 1826 540 314\n",
      "55\t32299\t0.0\t3\t22520 6642 2138 999\n",
      "56\t14054\t0.0\t3\t9863 2731 997 463\n",
      "57\t1432\t0.0\t3\t462 643 230 97\n",
      "58\t13369\t0.0\t3\t9051 3069 841 408\n",
      "59\t15848\t0.0\t3\t10896 3313 1120 519\n",
      "60\t8142\t0.0\t3\t5387 1845 597 313\n",
      "61\t10330\t0.0\t3\t7102 2236 687 305\n",
      "62\t11757\t0.0\t3\t8049 2458 831 419\n",
      "63\t7034\t0.0\t3\t4753 1563 497 221\n",
      "64\t8898\t0.0\t3\t6349 1789 510 250\n",
      "65\t6742\t0.0\t3\t4792 1347 426 177\n",
      "66\t6924\t0.0\t3\t4910 1402 387 225\n",
      "67\t6186\t0.0\t3\t4277 1311 418 180\n",
      "68\t5639\t0.0\t3\t3803 1236 416 184\n",
      "69\t5017\t0.0\t3\t3469 1049 340 159\n",
      "70\t4552\t0.0\t3\t3147 973 299 133\n",
      "71\t3886\t0.0\t3\t2651 842 269 124\n",
      "72\t3669\t0.0\t3\t2477 792 264 136\n",
      "73\t3435\t0.0\t3\t2179 740 247 269\n",
      "74\t3297\t0.0\t3\t1951 769 342 235\n",
      "75\t4733\t0.0\t3\t2578 1474 503 178\n",
      "76\t14392\t0.0\t3\t10541 2765 768 318\n",
      "77\t11471\t0.0\t3\t8145 2165 763 398\n",
      "78\t2068\t0.0\t3\t1136 520 246 166\n",
      "79\t366\t0.0\t3\t113 94 90 69\n",
      "80\t110\t0.0\t3\t24 25 31 30\n",
      "81\t38\t0.0\t3\t6 9 9 14\n",
      "82\t33\t0.0\t3\t4 12 8 9\n",
      "83\t9\t0.0\t3\t0 4 2 3\n",
      "84\t12\t0.0\t3\t0 5 5 2\n",
      "85\t7\t0.0\t3\t0 2 3 2\n",
      "86\t4\t0.0\t3\t0 1 0 3\n",
      "87\t6\t0.0\t3\t2 1 2 1\n",
      "88\t5\t0.0\t3\t1 2 0 2\n",
      "89\t3\t0.0\t3\t0 2 1\n",
      "90\t1\t0.0\t3\t0 1\n",
      "91\t5\t0.0\t3\t1 2 0 2\n",
      "92\t1\t0.0\t3\t0 0 0 1\n",
      "95\t4\t0.0\t3\t1 0 1 2\n",
      "96\t3\t0.0\t3\t0 0 1 2\n",
      "97\t1\t0.0\t3\t0 0 1\n",
      "98\t3\t0.0\t3\t0 2 1\n",
      "100\t2\t0.0\t3\t0 0 0 2\n",
      "101\t2\t0.0\t3\t1 0 1\n",
      "102\t2\t0.0\t3\t1 0 0 1\n",
      "103\t1\t0.0\t3\t0 0 0 1\n",
      "104\t6\t0.0\t3\t0 0 3 3\n",
      "105\t1\t0.0\t3\t0 0 0 1\n",
      "106\t1\t0.0\t3\t0 1\n",
      "107\t3\t0.0\t3\t1 0 2\n",
      "108\t4\t0.0\t3\t0 0 3 1\n",
      "109\t6\t0.0\t3\t4 0 1 1\n",
      "110\t7\t0.0\t3\t1 2 2 2\n",
      "111\t1\t0.0\t3\t1\n",
      "112\t2\t0.0\t3\t0 1 0 1\n",
      "114\t1\t0.0\t3\t0 1\n",
      "115\t1\t0.0\t3\t0 0 0 1\n",
      "116\t1\t0.0\t3\t0 0 0 1\n",
      "117\t1\t0.0\t3\t0 0 1\n",
      "118\t3\t0.0\t3\t0 0 0 3\n",
      "121\t1\t0.0\t3\t0 0 1\n",
      "125\t1\t0.0\t3\t0 1\n",
      "126\t1\t0.0\t3\t0 0 0 1\n",
      "127\t2\t0.0\t3\t0 1 0 1\n",
      "128\t1\t0.0\t3\t0 0 1\n",
      "129\t1\t0.0\t3\t1\n",
      "131\t1\t0.0\t3\t0 0 1\n",
      "132\t1\t0.0\t3\t0 0 0 1\n",
      "134\t3\t0.0\t3\t1 0 0 2\n",
      "135\t1\t0.0\t3\t0 1\n",
      "136\t2\t0.0\t3\t2\n",
      "139\t1\t0.0\t3\t0 1\n",
      "140\t5\t0.0\t3\t1 1 2 1\n",
      "141\t5\t0.0\t3\t1 3 1\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_04_1.fastq.gz -p ./fastq_trimmed_sub/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-04_S45_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:31    17,961,443 reads @   1.8 µs/read;  34.22 M reads/minute\n",
      "Finished in 31.587 s (1.759 µs/read; 34.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         17,961,443\n",
      "  Read 1 with adapter:               5,185,669 (28.9%)\n",
      "  Read 2 with adapter:               5,109,925 (28.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             560,295 (3.1%)\n",
      "Pairs written (passing filters):    17,401,148 (96.9%)\n",
      "\n",
      "Total basepairs processed: 5,424,355,786 bp\n",
      "  Read 1: 2,712,177,893 bp\n",
      "  Read 2: 2,712,177,893 bp\n",
      "Quality-trimmed:             143,437,174 bp (2.6%)\n",
      "  Read 1:    52,462,393 bp\n",
      "  Read 2:    90,974,781 bp\n",
      "Total written (filtered):  4,587,232,496 bp (84.6%)\n",
      "  Read 1: 2,291,617,956 bp\n",
      "  Read 2: 2,295,614,540 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 5185669 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.1%\n",
      "  C: 29.6%\n",
      "  G: 36.2%\n",
      "  T: 18.9%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t360664\t280647.5\t0\t360664\n",
      "4\t156280\t70161.9\t0\t156280\n",
      "5\t285129\t17540.5\t0\t285129\n",
      "8\t89003\t274.1\t0\t89003\n",
      "9\t179205\t68.5\t0\t179205\n",
      "10\t1612\t17.1\t1\t0 1612\n",
      "11\t185188\t4.3\t1\t182459 2729\n",
      "12\t1048\t1.1\t1\t0 1048\n",
      "13\t92839\t0.3\t1\t90566 2273\n",
      "14\t92308\t0.1\t1\t89746 2562\n",
      "15\t91705\t0.0\t1\t89071 2634\n",
      "16\t93217\t0.0\t1\t89124 4093\n",
      "17\t180168\t0.0\t1\t176098 4070\n",
      "18\t1524\t0.0\t1\t0 1474 50\n",
      "19\t91342\t0.0\t1\t87732 3432 178\n",
      "20\t94383\t0.0\t2\t88716 5212 455\n",
      "21\t182396\t0.0\t2\t177185 4468 743\n",
      "22\t2103\t0.0\t2\t0 1943 160\n",
      "23\t93656\t0.0\t2\t89334 3831 491\n",
      "24\t94058\t0.0\t2\t89125 4349 584\n",
      "25\t91994\t0.0\t2\t87041 4395 558\n",
      "26\t90135\t0.0\t2\t85013 4522 600\n",
      "27\t88685\t0.0\t2\t83716 4331 638\n",
      "28\t90761\t0.0\t2\t83270 6952 535 4\n",
      "29\t171653\t0.0\t2\t165130 5590 927 6\n",
      "30\t2847\t0.0\t3\t0 2531 238 78\n",
      "31\t88968\t0.0\t3\t83245 4921 588 214\n",
      "32\t87933\t0.0\t3\t82049 5109 582 193\n",
      "33\t87538\t0.0\t3\t81071 5541 664 262\n",
      "34\t86514\t0.0\t3\t80552 5103 625 234\n",
      "35\t84284\t0.0\t3\t78579 4911 575 219\n",
      "36\t82922\t0.0\t3\t77444 4714 573 191\n",
      "37\t79759\t0.0\t3\t74577 4469 489 224\n",
      "38\t77780\t0.0\t3\t72474 4523 588 195\n",
      "39\t78674\t0.0\t3\t71926 6074 489 185\n",
      "40\t145882\t0.0\t3\t140244 4632 723 283\n",
      "41\t2877\t0.0\t3\t402 2153 242 80\n",
      "42\t73983\t0.0\t3\t69351 3960 483 189\n",
      "43\t72388\t0.0\t3\t67832 3902 464 190\n",
      "44\t73379\t0.0\t3\t68164 4276 745 194\n",
      "45\t77312\t0.0\t3\t68148 8173 765 226\n",
      "46\t232132\t0.0\t3\t220805 9777 1062 488\n",
      "47\t108171\t0.0\t3\t104013 3607 402 149\n",
      "48\t1903\t0.0\t3\t368 1433 68 34\n",
      "49\t44295\t0.0\t3\t42368 1575 272 80\n",
      "50\t9495\t0.0\t3\t7167 2064 202 62\n",
      "51\t62935\t0.0\t3\t60777 1869 203 86\n",
      "52\t3099\t0.0\t3\t2070 837 155 37\n",
      "53\t21272\t0.0\t3\t18792 2215 191 74\n",
      "54\t62011\t0.0\t3\t59610 2007 274 120\n",
      "55\t3557\t0.0\t3\t2171 1162 166 58\n",
      "56\t31248\t0.0\t3\t29324 1663 202 59\n",
      "57\t31657\t0.0\t3\t29555 1853 187 62\n",
      "58\t34671\t0.0\t3\t32475 1938 181 77\n",
      "59\t33046\t0.0\t3\t30905 1848 226 67\n",
      "60\t30579\t0.0\t3\t28383 1949 180 67\n",
      "61\t40079\t0.0\t3\t38136 1672 185 86\n",
      "62\t18017\t0.0\t3\t16682 1160 113 62\n",
      "63\t21064\t0.0\t3\t19566 1236 158 104\n",
      "64\t23513\t0.0\t3\t21527 1729 200 57\n",
      "65\t39517\t0.0\t3\t36904 2340 177 96\n",
      "66\t38668\t0.0\t3\t37147 1276 148 97\n",
      "67\t4520\t0.0\t3\t3402 791 128 199\n",
      "68\t16785\t0.0\t3\t14760 1248 292 485\n",
      "69\t23746\t0.0\t3\t20686 2014 530 516\n",
      "70\t40356\t0.0\t3\t36304 3234 686 132\n",
      "71\t52809\t0.0\t3\t48909 3594 233 73\n",
      "72\t42822\t0.0\t3\t40757 1827 172 66\n",
      "73\t6867\t0.0\t3\t6237 510 84 36\n",
      "74\t1384\t0.0\t3\t1140 171 43 30\n",
      "75\t577\t0.0\t3\t417 109 30 21\n",
      "76\t244\t0.0\t3\t167 45 22 10\n",
      "77\t125\t0.0\t3\t75 21 18 11\n",
      "78\t71\t0.0\t3\t41 18 7 5\n",
      "79\t43\t0.0\t3\t21 13 4 5\n",
      "80\t48\t0.0\t3\t27 10 7 4\n",
      "81\t27\t0.0\t3\t16 5 3 3\n",
      "82\t24\t0.0\t3\t7 4 2 11\n",
      "83\t15\t0.0\t3\t9 5 0 1\n",
      "84\t14\t0.0\t3\t11 2 0 1\n",
      "85\t7\t0.0\t3\t4 1 1 1\n",
      "86\t10\t0.0\t3\t5 2 3\n",
      "87\t8\t0.0\t3\t3 2 2 1\n",
      "88\t5\t0.0\t3\t1 1 1 2\n",
      "89\t7\t0.0\t3\t3 1 1 2\n",
      "90\t2\t0.0\t3\t2\n",
      "91\t7\t0.0\t3\t2 1 1 3\n",
      "92\t5\t0.0\t3\t3 0 1 1\n",
      "93\t2\t0.0\t3\t2\n",
      "94\t3\t0.0\t3\t1 0 1 1\n",
      "95\t5\t0.0\t3\t3 2\n",
      "96\t1\t0.0\t3\t0 0 0 1\n",
      "97\t3\t0.0\t3\t0 1 0 2\n",
      "98\t3\t0.0\t3\t1 1 1\n",
      "99\t3\t0.0\t3\t1 1 1\n",
      "100\t4\t0.0\t3\t2 1 1\n",
      "101\t2\t0.0\t3\t1 0 1\n",
      "102\t1\t0.0\t3\t1\n",
      "103\t4\t0.0\t3\t2 0 1 1\n",
      "104\t4\t0.0\t3\t2 1 0 1\n",
      "105\t1\t0.0\t3\t1\n",
      "107\t2\t0.0\t3\t0 1 1\n",
      "108\t2\t0.0\t3\t0 0 0 2\n",
      "110\t2\t0.0\t3\t0 1 0 1\n",
      "111\t2\t0.0\t3\t1 0 1\n",
      "112\t3\t0.0\t3\t1 0 1 1\n",
      "113\t1\t0.0\t3\t0 1\n",
      "114\t2\t0.0\t3\t0 1 1\n",
      "115\t1\t0.0\t3\t1\n",
      "116\t2\t0.0\t3\t0 2\n",
      "117\t1\t0.0\t3\t0 0 0 1\n",
      "118\t2\t0.0\t3\t1 0 0 1\n",
      "119\t2\t0.0\t3\t0 0 1 1\n",
      "120\t1\t0.0\t3\t0 1\n",
      "121\t1\t0.0\t3\t1\n",
      "122\t2\t0.0\t3\t0 1 0 1\n",
      "123\t3\t0.0\t3\t0 0 0 3\n",
      "124\t2\t0.0\t3\t1 0 1\n",
      "125\t1\t0.0\t3\t0 0 0 1\n",
      "126\t3\t0.0\t3\t2 1\n",
      "127\t2\t0.0\t3\t2\n",
      "128\t2\t0.0\t3\t1 1\n",
      "129\t3\t0.0\t3\t3\n",
      "130\t1\t0.0\t3\t1\n",
      "131\t2\t0.0\t3\t1 0 0 1\n",
      "132\t4\t0.0\t3\t3 1\n",
      "133\t3\t0.0\t3\t2 0 1\n",
      "134\t1\t0.0\t3\t0 1\n",
      "135\t1\t0.0\t3\t0 0 0 1\n",
      "136\t2\t0.0\t3\t2\n",
      "138\t4\t0.0\t3\t3 0 1\n",
      "139\t6\t0.0\t3\t2 1 2 1\n",
      "140\t4\t0.0\t3\t1 2 0 1\n",
      "141\t10\t0.0\t3\t0 5 2 3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 5109925 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.8%\n",
      "  C: 29.7%\n",
      "  G: 42.0%\n",
      "  T: 10.4%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t364113\t280647.5\t0\t364113\n",
      "4\t160731\t70161.9\t0\t160731\n",
      "5\t287770\t17540.5\t0\t287770\n",
      "8\t89086\t274.1\t0\t89086\n",
      "9\t179193\t68.5\t0\t179193\n",
      "10\t1469\t17.1\t1\t0 1469\n",
      "11\t184966\t4.3\t1\t182352 2614\n",
      "12\t1461\t1.1\t1\t0 1461\n",
      "13\t182819\t0.3\t1\t179987 2832\n",
      "14\t1016\t0.1\t1\t0 1016\n",
      "15\t92272\t0.0\t1\t89207 3065\n",
      "16\t180305\t0.0\t1\t177091 3214\n",
      "17\t2663\t0.0\t1\t0 2663\n",
      "18\t199436\t0.0\t1\t195101 4302 33\n",
      "19\t2075\t0.0\t1\t0 1813 262\n",
      "20\t117373\t0.0\t2\t108717 8158 498\n",
      "21\t328564\t0.0\t2\t294268 31604 2692\n",
      "22\t3964\t0.0\t2\t0 2978 986\n",
      "23\t519\t0.0\t2\t0 178 341\n",
      "24\t2363\t0.0\t2\t0 2159 204\n",
      "25\t83356\t0.0\t2\t69332 12209 1815\n",
      "26\t89575\t0.0\t2\t69422 16241 3912\n",
      "27\t131651\t0.0\t2\t105777 20091 5783\n",
      "28\t42436\t0.0\t2\t0 35902 6523 11\n",
      "29\t163491\t0.0\t2\t119479 34480 9504 28\n",
      "30\t10207\t0.0\t3\t0 6226 2816 1165\n",
      "31\t156455\t0.0\t3\t110674 33102 9618 3061\n",
      "32\t8835\t0.0\t3\t0 5214 2359 1262\n",
      "33\t85021\t0.0\t3\t57169 19447 5898 2507\n",
      "34\t83922\t0.0\t3\t55982 19270 6097 2573\n",
      "35\t82139\t0.0\t3\t54754 19190 5810 2385\n",
      "36\t79330\t0.0\t3\t51640 18619 6259 2812\n",
      "37\t77264\t0.0\t3\t52389 18021 4834 2020\n",
      "38\t99336\t0.0\t3\t69199 20504 6818 2815\n",
      "39\t49181\t0.0\t3\t29704 12741 4615 2121\n",
      "40\t75020\t0.0\t3\t49071 18398 5294 2257\n",
      "41\t104836\t0.0\t3\t74314 21007 6517 2998\n",
      "42\t40015\t0.0\t3\t21660 12301 4119 1935\n",
      "43\t128245\t0.0\t3\t90389 26742 8064 3050\n",
      "44\t7343\t0.0\t3\t2444 3142 1193 564\n",
      "45\t77956\t0.0\t3\t52080 18257 5419 2200\n",
      "46\t104258\t0.0\t3\t74015 21494 6395 2354\n",
      "47\t9193\t0.0\t3\t3830 3508 1216 639\n",
      "48\t58086\t0.0\t3\t40167 12807 3685 1427\n",
      "49\t53739\t0.0\t3\t36656 12292 3554 1237\n",
      "50\t51766\t0.0\t3\t35183 11541 3542 1500\n",
      "51\t53396\t0.0\t3\t35214 13362 3488 1332\n",
      "52\t110089\t0.0\t3\t82701 20377 5034 1977\n",
      "53\t28187\t0.0\t3\t15724 7359 3477 1627\n",
      "54\t4387\t0.0\t3\t419 3148 521 299\n",
      "55\t105443\t0.0\t3\t76925 20420 5731 2367\n",
      "56\t11116\t0.0\t3\t5733 3037 1540 806\n",
      "57\t1695\t0.0\t3\t174 1043 331 147\n",
      "58\t35254\t0.0\t3\t23871 8245 2258 880\n",
      "59\t52830\t0.0\t3\t38025 10492 3151 1162\n",
      "60\t7626\t0.0\t3\t3684 2492 984 466\n",
      "61\t27419\t0.0\t3\t19130 6085 1551 653\n",
      "62\t43093\t0.0\t3\t31448 8323 2393 929\n",
      "63\t6992\t0.0\t3\t3585 2210 815 382\n",
      "64\t25140\t0.0\t3\t18271 5064 1288 517\n",
      "65\t18696\t0.0\t3\t13474 3798 1018 406\n",
      "66\t19407\t0.0\t3\t13841 4059 1098 409\n",
      "67\t18136\t0.0\t3\t12735 3937 1045 419\n",
      "68\t16496\t0.0\t3\t11353 3714 1053 376\n",
      "69\t14546\t0.0\t3\t10198 3100 902 346\n",
      "70\t13848\t0.0\t3\t9860 2884 806 298\n",
      "71\t11759\t0.0\t3\t8191 2498 754 316\n",
      "72\t11483\t0.0\t3\t7805 2467 683 528\n",
      "73\t11161\t0.0\t3\t6827 2324 904 1106\n",
      "74\t12559\t0.0\t3\t6934 3660 1488 477\n",
      "75\t32968\t0.0\t3\t22563 7865 1907 633\n",
      "76\t67063\t0.0\t3\t50944 11842 3136 1141\n",
      "77\t11944\t0.0\t3\t6706 3130 1435 673\n",
      "78\t1494\t0.0\t3\t475 444 335 240\n",
      "79\t369\t0.0\t3\t87 123 85 74\n",
      "80\t109\t0.0\t3\t21 24 36 28\n",
      "81\t57\t0.0\t3\t12 15 19 11\n",
      "82\t29\t0.0\t3\t3 10 4 12\n",
      "83\t22\t0.0\t3\t5 6 7 4\n",
      "84\t14\t0.0\t3\t2 3 3 6\n",
      "85\t19\t0.0\t3\t1 2 8 8\n",
      "86\t14\t0.0\t3\t4 4 4 2\n",
      "87\t12\t0.0\t3\t1 4 4 3\n",
      "88\t14\t0.0\t3\t2 2 6 4\n",
      "89\t10\t0.0\t3\t3 3 2 2\n",
      "90\t8\t0.0\t3\t1 3 2 2\n",
      "91\t10\t0.0\t3\t1 1 3 5\n",
      "92\t8\t0.0\t3\t2 0 2 4\n",
      "93\t10\t0.0\t3\t3 5 2\n",
      "94\t6\t0.0\t3\t1 1 4\n",
      "95\t7\t0.0\t3\t1 2 2 2\n",
      "96\t5\t0.0\t3\t1 1 1 2\n",
      "97\t1\t0.0\t3\t0 0 0 1\n",
      "98\t3\t0.0\t3\t1 2\n",
      "99\t5\t0.0\t3\t1 2 2\n",
      "100\t5\t0.0\t3\t1 1 1 2\n",
      "101\t1\t0.0\t3\t0 0 0 1\n",
      "102\t3\t0.0\t3\t0 0 0 3\n",
      "103\t7\t0.0\t3\t1 0 2 4\n",
      "104\t4\t0.0\t3\t1 1 1 1\n",
      "105\t8\t0.0\t3\t2 0 3 3\n",
      "106\t7\t0.0\t3\t0 0 4 3\n",
      "107\t6\t0.0\t3\t1 0 4 1\n",
      "108\t4\t0.0\t3\t0 3 0 1\n",
      "109\t8\t0.0\t3\t1 4 2 1\n",
      "110\t6\t0.0\t3\t2 2 1 1\n",
      "111\t7\t0.0\t3\t2 1 1 3\n",
      "112\t2\t0.0\t3\t0 0 1 1\n",
      "113\t1\t0.0\t3\t0 1\n",
      "114\t2\t0.0\t3\t0 1 1\n",
      "115\t1\t0.0\t3\t0 0 1\n",
      "116\t2\t0.0\t3\t1 0 0 1\n",
      "117\t3\t0.0\t3\t1 1 0 1\n",
      "118\t2\t0.0\t3\t1 0 0 1\n",
      "119\t1\t0.0\t3\t1\n",
      "120\t2\t0.0\t3\t0 0 0 2\n",
      "122\t1\t0.0\t3\t0 0 0 1\n",
      "123\t1\t0.0\t3\t0 0 1\n",
      "124\t6\t0.0\t3\t1 1 2 2\n",
      "125\t3\t0.0\t3\t0 2 0 1\n",
      "126\t6\t0.0\t3\t2 2 1 1\n",
      "127\t1\t0.0\t3\t1\n",
      "129\t6\t0.0\t3\t2 2 1 1\n",
      "130\t1\t0.0\t3\t1\n",
      "131\t4\t0.0\t3\t0 2 2\n",
      "132\t5\t0.0\t3\t1 2 0 2\n",
      "133\t2\t0.0\t3\t0 2\n",
      "134\t1\t0.0\t3\t1\n",
      "135\t4\t0.0\t3\t1 2 1\n",
      "136\t2\t0.0\t3\t1 1\n",
      "137\t2\t0.0\t3\t1 0 1\n",
      "138\t1\t0.0\t3\t0 0 0 1\n",
      "139\t4\t0.0\t3\t2 1 0 1\n",
      "140\t6\t0.0\t3\t3 1 1 1\n",
      "141\t14\t0.0\t3\t1 3 3 7\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_01_1.fastq.gz -p ./fastq_trimmed_sub/Liv_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-01_S38_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:35    20,176,743 reads @   1.7 µs/read;  34.52 M reads/minute\n",
      "Finished in 35.183 s (1.744 µs/read; 34.41 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         20,176,743\n",
      "  Read 1 with adapter:               6,762,032 (33.5%)\n",
      "  Read 2 with adapter:               6,662,730 (33.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             440,108 (2.2%)\n",
      "Pairs written (passing filters):    19,736,635 (97.8%)\n",
      "\n",
      "Total basepairs processed: 6,093,376,386 bp\n",
      "  Read 1: 3,046,688,193 bp\n",
      "  Read 2: 3,046,688,193 bp\n",
      "Quality-trimmed:             122,115,088 bp (2.0%)\n",
      "  Read 1:    47,720,231 bp\n",
      "  Read 2:    74,394,857 bp\n",
      "Total written (filtered):  5,135,793,927 bp (84.3%)\n",
      "  Read 1: 2,564,896,921 bp\n",
      "  Read 2: 2,570,897,006 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 6762032 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.5%\n",
      "  C: 29.3%\n",
      "  G: 38.4%\n",
      "  T: 17.7%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t375750\t315261.6\t0\t375750\n",
      "4\t178448\t78815.4\t0\t178448\n",
      "5\t342693\t19703.9\t0\t342693\n",
      "8\t107520\t307.9\t0\t107520\n",
      "9\t219393\t77.0\t0\t219393\n",
      "10\t1793\t19.2\t1\t0 1793\n",
      "11\t227411\t4.8\t1\t224287 3124\n",
      "12\t973\t1.2\t1\t0 973\n",
      "13\t114015\t0.3\t1\t111636 2379\n",
      "14\t115828\t0.1\t1\t113062 2766\n",
      "15\t113757\t0.0\t1\t111030 2727\n",
      "16\t115350\t0.0\t1\t111318 4032\n",
      "17\t228276\t0.0\t1\t223857 4419\n",
      "18\t1534\t0.0\t1\t0 1502 32\n",
      "19\t114645\t0.0\t1\t111004 3483 158\n",
      "20\t119078\t0.0\t2\t113465 5099 514\n",
      "21\t233130\t0.0\t2\t227464 4768 898\n",
      "22\t2165\t0.0\t2\t0 1962 203\n",
      "23\t119787\t0.0\t2\t115269 3916 602\n",
      "24\t120116\t0.0\t2\t115069 4420 627\n",
      "25\t117227\t0.0\t2\t112113 4481 633\n",
      "26\t115811\t0.0\t2\t110337 4762 712\n",
      "27\t117108\t0.0\t2\t111839 4496 773\n",
      "28\t117593\t0.0\t2\t109659 7232 696 6\n",
      "29\t231344\t0.0\t2\t223860 6286 1182 16\n",
      "30\t3063\t0.0\t3\t0 2708 268 87\n",
      "31\t117192\t0.0\t3\t111072 5045 744 331\n",
      "32\t118052\t0.0\t3\t111753 5286 692 321\n",
      "33\t118100\t0.0\t3\t110807 5980 939 374\n",
      "34\t117957\t0.0\t3\t111671 5202 790 294\n",
      "35\t114019\t0.0\t3\t107838 5129 729 323\n",
      "36\t111340\t0.0\t3\t105214 5072 735 319\n",
      "37\t107145\t0.0\t3\t101426 4761 692 266\n",
      "38\t108920\t0.0\t3\t101301 6619 696 304\n",
      "39\t206368\t0.0\t3\t199328 5360 1125 555\n",
      "40\t2670\t0.0\t3\t151 2295 168 56\n",
      "41\t102694\t0.0\t3\t97633 4168 618 275\n",
      "42\t102570\t0.0\t3\t97138 4469 673 290\n",
      "43\t99423\t0.0\t3\t94270 4225 588 340\n",
      "44\t102800\t0.0\t3\t96302 4859 1332 307\n",
      "45\t120647\t0.0\t3\t99566 19433 1160 488\n",
      "46\t880141\t0.0\t3\t858743 17838 2449 1111\n",
      "47\t23851\t0.0\t3\t19722 3523 425 181\n",
      "48\t1061\t0.0\t3\t619 300 96 46\n",
      "49\t5927\t0.0\t3\t5176 512 160 79\n",
      "50\t1150\t0.0\t3\t736 263 105 46\n",
      "51\t1995\t0.0\t3\t1508 294 128 65\n",
      "52\t1288\t0.0\t3\t920 217 102 49\n",
      "53\t2716\t0.0\t3\t2113 412 116 75\n",
      "54\t6308\t0.0\t3\t5390 631 196 91\n",
      "55\t4152\t0.0\t3\t3367 549 150 86\n",
      "56\t6577\t0.0\t3\t5677 659 158 83\n",
      "57\t11421\t0.0\t3\t10123 1013 197 88\n",
      "58\t27054\t0.0\t3\t25184 1556 207 107\n",
      "59\t43350\t0.0\t3\t41172 1860 234 84\n",
      "60\t43205\t0.0\t3\t40822 2051 223 109\n",
      "61\t47788\t0.0\t3\t45636 1825 246 81\n",
      "62\t37842\t0.0\t3\t36068 1522 160 92\n",
      "63\t29954\t0.0\t3\t28204 1359 197 194\n",
      "64\t32029\t0.0\t3\t29739 1646 344 300\n",
      "65\t44146\t0.0\t3\t40132 3399 495 120\n",
      "66\t106336\t0.0\t3\t101564 4423 245 104\n",
      "67\t98747\t0.0\t3\t96036 2418 175 118\n",
      "68\t16572\t0.0\t3\t15270 989 182 131\n",
      "69\t10601\t0.0\t3\t9422 902 160 117\n",
      "70\t19620\t0.0\t3\t18186 1125 216 93\n",
      "71\t20340\t0.0\t3\t18950 1191 139 60\n",
      "72\t20474\t0.0\t3\t19356 911 142 65\n",
      "73\t8032\t0.0\t3\t7324 544 108 56\n",
      "74\t2775\t0.0\t3\t2333 305 97 40\n",
      "75\t1207\t0.0\t3\t955 166 64 22\n",
      "76\t618\t0.0\t3\t442 103 50 23\n",
      "77\t342\t0.0\t3\t217 78 22 25\n",
      "78\t193\t0.0\t3\t109 42 30 12\n",
      "79\t128\t0.0\t3\t71 27 19 11\n",
      "80\t80\t0.0\t3\t44 17 11 8\n",
      "81\t47\t0.0\t3\t25 9 10 3\n",
      "82\t42\t0.0\t3\t21 12 2 7\n",
      "83\t18\t0.0\t3\t7 6 2 3\n",
      "84\t20\t0.0\t3\t8 6 2 4\n",
      "85\t19\t0.0\t3\t8 7 1 3\n",
      "86\t11\t0.0\t3\t2 3 3 3\n",
      "87\t10\t0.0\t3\t2 1 3 4\n",
      "88\t10\t0.0\t3\t3 1 2 4\n",
      "89\t11\t0.0\t3\t3 2 2 4\n",
      "90\t10\t0.0\t3\t3 1 6\n",
      "91\t11\t0.0\t3\t4 4 3\n",
      "92\t4\t0.0\t3\t1 1 1 1\n",
      "93\t5\t0.0\t3\t2 2 0 1\n",
      "94\t1\t0.0\t3\t1\n",
      "95\t4\t0.0\t3\t0 1 1 2\n",
      "96\t2\t0.0\t3\t1 0 1\n",
      "97\t7\t0.0\t3\t0 4 1 2\n",
      "98\t3\t0.0\t3\t0 1 0 2\n",
      "99\t3\t0.0\t3\t1 1 1\n",
      "100\t3\t0.0\t3\t0 1 2\n",
      "102\t1\t0.0\t3\t0 1\n",
      "103\t1\t0.0\t3\t0 0 0 1\n",
      "104\t1\t0.0\t3\t0 1\n",
      "105\t2\t0.0\t3\t1 0 1\n",
      "106\t2\t0.0\t3\t0 0 2\n",
      "107\t1\t0.0\t3\t0 0 1\n",
      "108\t4\t0.0\t3\t2 0 2\n",
      "110\t2\t0.0\t3\t1 1\n",
      "111\t2\t0.0\t3\t1 1\n",
      "112\t1\t0.0\t3\t1\n",
      "114\t3\t0.0\t3\t1 1 0 1\n",
      "115\t1\t0.0\t3\t0 0 1\n",
      "117\t2\t0.0\t3\t0 1 0 1\n",
      "119\t4\t0.0\t3\t2 2\n",
      "120\t3\t0.0\t3\t2 0 0 1\n",
      "122\t1\t0.0\t3\t0 0 1\n",
      "125\t1\t0.0\t3\t1\n",
      "126\t1\t0.0\t3\t0 0 1\n",
      "130\t1\t0.0\t3\t0 0 1\n",
      "132\t2\t0.0\t3\t1 0 1\n",
      "135\t2\t0.0\t3\t1 1\n",
      "136\t3\t0.0\t3\t2 1\n",
      "137\t3\t0.0\t3\t0 1 2\n",
      "138\t1\t0.0\t3\t1\n",
      "139\t5\t0.0\t3\t2 2 1\n",
      "140\t8\t0.0\t3\t1 3 2 2\n",
      "141\t8\t0.0\t3\t1 3 2 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 6662730 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.0%\n",
      "  C: 28.3%\n",
      "  G: 44.8%\n",
      "  T: 9.8%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t406609\t315261.6\t0\t406609\n",
      "4\t181032\t78815.4\t0\t181032\n",
      "5\t346254\t19703.9\t0\t346254\n",
      "8\t107954\t307.9\t0\t107954\n",
      "9\t219451\t77.0\t0\t219451\n",
      "10\t1558\t19.2\t1\t0 1558\n",
      "11\t227233\t4.8\t1\t224050 3183\n",
      "12\t1349\t1.2\t1\t0 1349\n",
      "13\t227378\t0.3\t1\t223823 3555\n",
      "14\t922\t0.1\t1\t0 922\n",
      "15\t114301\t0.0\t1\t110964 3337\n",
      "16\t227208\t0.0\t1\t223141 4067\n",
      "17\t2351\t0.0\t1\t0 2351\n",
      "18\t262582\t0.0\t1\t257283 5273 26\n",
      "19\t1953\t0.0\t1\t0 1694 259\n",
      "20\t149385\t0.0\t2\t140814 7892 679\n",
      "21\t410197\t0.0\t2\t364681 41967 3549\n",
      "22\t4379\t0.0\t2\t0 3045 1334\n",
      "23\t517\t0.0\t2\t0 187 330\n",
      "24\t1993\t0.0\t2\t0 1828 165\n",
      "25\t105213\t0.0\t2\t86255 16303 2655\n",
      "26\t114112\t0.0\t2\t87270 21039 5803\n",
      "27\t177070\t0.0\t2\t138938 28823 9309\n",
      "28\t50896\t0.0\t2\t0 41649 9239 8\n",
      "29\t226788\t0.0\t2\t161368 49536 15844 40\n",
      "30\t10302\t0.0\t3\t0 5622 3196 1484\n",
      "31\t206752\t0.0\t3\t146297 44137 12399 3919\n",
      "32\t6455\t0.0\t3\t0 3726 1696 1033\n",
      "33\t113899\t0.0\t3\t76989 25178 8088 3644\n",
      "34\t113956\t0.0\t3\t75724 25469 8640 4123\n",
      "35\t110813\t0.0\t3\t73485 25339 8162 3827\n",
      "36\t105633\t0.0\t3\t68963 23952 8494 4224\n",
      "37\t101607\t0.0\t3\t70019 21904 6666 3018\n",
      "38\t103438\t0.0\t3\t67831 23337 8342 3928\n",
      "39\t102075\t0.0\t3\t66879 23366 8049 3781\n",
      "40\t101761\t0.0\t3\t66470 23799 7738 3754\n",
      "41\t145575\t0.0\t3\t101354 30084 9440 4697\n",
      "42\t49446\t0.0\t3\t29408 13592 4484 1962\n",
      "43\t135051\t0.0\t3\t93427 28222 9071 4331\n",
      "44\t55268\t0.0\t3\t35055 13382 4631 2200\n",
      "45\t96170\t0.0\t3\t64663 21936 6590 2981\n",
      "46\t163804\t0.0\t3\t113564 34206 11166 4868\n",
      "47\t7492\t0.0\t3\t2358 3078 1318 738\n",
      "48\t80575\t0.0\t3\t55971 17291 5151 2162\n",
      "49\t77954\t0.0\t3\t52575 17572 5532 2275\n",
      "50\t73884\t0.0\t3\t50088 16205 5240 2351\n",
      "51\t75233\t0.0\t3\t49975 17803 5193 2262\n",
      "52\t178307\t0.0\t3\t131197 34170 9239 3701\n",
      "53\t24254\t0.0\t3\t11907 6698 3586 2063\n",
      "54\t4508\t0.0\t3\t408 3199 552 349\n",
      "55\t166139\t0.0\t3\t118709 32624 10271 4535\n",
      "56\t9278\t0.0\t3\t3941 2585 1673 1079\n",
      "57\t1611\t0.0\t3\t178 977 283 173\n",
      "58\t52387\t0.0\t3\t35504 11632 3678 1573\n",
      "59\t86691\t0.0\t3\t61701 17447 5350 2193\n",
      "60\t6638\t0.0\t3\t2677 2331 1027 603\n",
      "61\t41333\t0.0\t3\t28571 9017 2581 1164\n",
      "62\t69170\t0.0\t3\t49920 13533 4018 1699\n",
      "63\t5800\t0.0\t3\t2499 1976 855 470\n",
      "64\t35639\t0.0\t3\t25513 7184 2049 893\n",
      "65\t30547\t0.0\t3\t22062 6065 1678 742\n",
      "66\t29615\t0.0\t3\t21123 5902 1762 828\n",
      "67\t27274\t0.0\t3\t19102 5769 1662 741\n",
      "68\t24816\t0.0\t3\t16864 5465 1724 763\n",
      "69\t22051\t0.0\t3\t15420 4518 1463 650\n",
      "70\t20429\t0.0\t3\t14465 4218 1223 523\n",
      "71\t18088\t0.0\t3\t12607 3805 1154 522\n",
      "72\t17682\t0.0\t3\t12014 3837 1214 617\n",
      "73\t16237\t0.0\t3\t10637 3429 1221 950\n",
      "74\t17533\t0.0\t3\t10721 4627 1572 613\n",
      "75\t50014\t0.0\t3\t35573 10479 2832 1130\n",
      "76\t87196\t0.0\t3\t64787 15796 4573 2040\n",
      "77\t11303\t0.0\t3\t5995 2812 1508 988\n",
      "78\t1394\t0.0\t3\t437 372 342 243\n",
      "79\t379\t0.0\t3\t93 94 113 79\n",
      "80\t144\t0.0\t3\t23 40 49 32\n",
      "81\t57\t0.0\t3\t11 7 23 16\n",
      "82\t47\t0.0\t3\t4 7 19 17\n",
      "83\t22\t0.0\t3\t0 6 7 9\n",
      "84\t20\t0.0\t3\t3 8 5 4\n",
      "85\t21\t0.0\t3\t5 5 6 5\n",
      "86\t13\t0.0\t3\t4 3 2 4\n",
      "87\t12\t0.0\t3\t3 4 4 1\n",
      "88\t13\t0.0\t3\t3 2 4 4\n",
      "89\t6\t0.0\t3\t1 2 2 1\n",
      "90\t14\t0.0\t3\t4 5 1 4\n",
      "91\t7\t0.0\t3\t1 0 2 4\n",
      "92\t10\t0.0\t3\t4 4 0 2\n",
      "93\t8\t0.0\t3\t1 1 3 3\n",
      "94\t6\t0.0\t3\t1 1 1 3\n",
      "95\t8\t0.0\t3\t3 3 2\n",
      "96\t3\t0.0\t3\t1 1 0 1\n",
      "97\t5\t0.0\t3\t4 0 1\n",
      "98\t5\t0.0\t3\t2 1 0 2\n",
      "99\t6\t0.0\t3\t2 2 1 1\n",
      "100\t8\t0.0\t3\t3 2 2 1\n",
      "101\t7\t0.0\t3\t0 3 1 3\n",
      "102\t3\t0.0\t3\t0 1 2\n",
      "103\t8\t0.0\t3\t1 4 0 3\n",
      "104\t4\t0.0\t3\t1 2 0 1\n",
      "105\t2\t0.0\t3\t1 1\n",
      "106\t5\t0.0\t3\t0 1 1 3\n",
      "107\t8\t0.0\t3\t1 0 5 2\n",
      "108\t18\t0.0\t3\t5 9 3 1\n",
      "109\t23\t0.0\t3\t9 11 0 3\n",
      "110\t18\t0.0\t3\t4 6 5 3\n",
      "111\t4\t0.0\t3\t0 2 0 2\n",
      "112\t3\t0.0\t3\t1 1 0 1\n",
      "114\t1\t0.0\t3\t0 1\n",
      "115\t1\t0.0\t3\t1\n",
      "117\t2\t0.0\t3\t0 1 1\n",
      "119\t2\t0.0\t3\t1 0 0 1\n",
      "120\t2\t0.0\t3\t0 0 1 1\n",
      "123\t2\t0.0\t3\t0 0 2\n",
      "124\t2\t0.0\t3\t1 0 0 1\n",
      "127\t1\t0.0\t3\t1\n",
      "130\t3\t0.0\t3\t2 1\n",
      "131\t2\t0.0\t3\t1 0 0 1\n",
      "132\t3\t0.0\t3\t0 0 2 1\n",
      "133\t2\t0.0\t3\t0 2\n",
      "134\t3\t0.0\t3\t1 2\n",
      "135\t1\t0.0\t3\t0 0 0 1\n",
      "136\t3\t0.0\t3\t1 0 2\n",
      "137\t2\t0.0\t3\t0 2\n",
      "138\t4\t0.0\t3\t2 2\n",
      "139\t4\t0.0\t3\t2 0 1 1\n",
      "140\t3\t0.0\t3\t1 0 1 1\n",
      "141\t8\t0.0\t3\t0 3 2 3\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_02_1.fastq.gz -p ./fastq_trimmed_sub/Liv_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-02_S39_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:29    14,755,790 reads @   2.0 µs/read;  29.83 M reads/minute\n",
      "Finished in 29.779 s (2.018 µs/read; 29.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         14,755,790\n",
      "  Read 1 with adapter:               4,835,691 (32.8%)\n",
      "  Read 2 with adapter:               4,662,862 (31.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:           1,123,164 (7.6%)\n",
      "Pairs written (passing filters):    13,632,626 (92.4%)\n",
      "\n",
      "Total basepairs processed: 4,456,248,580 bp\n",
      "  Read 1: 2,228,124,290 bp\n",
      "  Read 2: 2,228,124,290 bp\n",
      "Quality-trimmed:             284,474,525 bp (6.4%)\n",
      "  Read 1:   110,642,310 bp\n",
      "  Read 2:   173,832,215 bp\n",
      "Total written (filtered):  3,534,124,291 bp (79.3%)\n",
      "  Read 1: 1,764,647,515 bp\n",
      "  Read 2: 1,769,476,776 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 4835691 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.8%\n",
      "  C: 32.6%\n",
      "  G: 36.0%\n",
      "  T: 16.5%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t362399\t230559.2\t0\t362399\n",
      "4\t120905\t57639.8\t0\t120905\n",
      "5\t248824\t14410.0\t0\t248824\n",
      "8\t78238\t225.2\t0\t78238\n",
      "9\t159530\t56.3\t0\t159530\n",
      "10\t1062\t14.1\t1\t0 1062\n",
      "11\t166187\t3.5\t1\t164055 2132\n",
      "12\t761\t0.9\t1\t0 761\n",
      "13\t84173\t0.2\t1\t82389 1784\n",
      "14\t84046\t0.1\t1\t82170 1876\n",
      "15\t82318\t0.0\t1\t80284 2034\n",
      "16\t84234\t0.0\t1\t81246 2988\n",
      "17\t165591\t0.0\t1\t162341 3250\n",
      "18\t1087\t0.0\t1\t0 1063 24\n",
      "19\t83663\t0.0\t1\t81152 2394 117\n",
      "20\t86143\t0.0\t2\t82227 3596 320\n",
      "21\t169718\t0.0\t2\t165719 3383 616\n",
      "22\t1590\t0.0\t2\t0 1453 137\n",
      "23\t87383\t0.0\t2\t84068 2937 378\n",
      "24\t89393\t0.0\t2\t85746 3214 433\n",
      "25\t83842\t0.0\t2\t80322 3082 438\n",
      "26\t82579\t0.0\t2\t78743 3365 471\n",
      "27\t84273\t0.0\t2\t80403 3345 525\n",
      "28\t83458\t0.0\t2\t77988 5038 430 2\n",
      "29\t168664\t0.0\t2\t163348 4477 819 20\n",
      "30\t2224\t0.0\t3\t0 1973 189 62\n",
      "31\t85229\t0.0\t3\t80792 3645 578 214\n",
      "32\t85039\t0.0\t3\t80445 3853 523 218\n",
      "33\t83369\t0.0\t3\t78310 4138 684 237\n",
      "34\t86319\t0.0\t3\t80264 5282 511 262\n",
      "35\t155259\t0.0\t3\t149927 4161 834 337\n",
      "36\t4551\t0.0\t3\t937 3285 213 116\n",
      "37\t147084\t0.0\t3\t142150 3837 750 347\n",
      "38\t2092\t0.0\t3\t104 1796 136 56\n",
      "39\t74601\t0.0\t3\t70829 3100 481 191\n",
      "40\t74885\t0.0\t3\t69741 4431 477 236\n",
      "41\t138836\t0.0\t3\t134028 3704 776 328\n",
      "42\t2221\t0.0\t3\t301 1733 134 53\n",
      "43\t67672\t0.0\t3\t64083 2818 448 323\n",
      "44\t68297\t0.0\t3\t63790 3281 1026 200\n",
      "45\t80671\t0.0\t3\t66605 12997 738 331\n",
      "46\t579933\t0.0\t3\t565149 12289 1669 826\n",
      "47\t8284\t0.0\t3\t5632 2341 211 100\n",
      "48\t530\t0.0\t3\t325 128 59 18\n",
      "49\t797\t0.0\t3\t569 138 57 33\n",
      "50\t720\t0.0\t3\t492 147 50 31\n",
      "51\t2296\t0.0\t3\t1981 209 70 36\n",
      "52\t1104\t0.0\t3\t788 216 74 26\n",
      "53\t3059\t0.0\t3\t2555 350 110 44\n",
      "54\t6183\t0.0\t3\t5513 477 128 65\n",
      "55\t2297\t0.0\t3\t1813 352 92 40\n",
      "56\t6276\t0.0\t3\t5619 490 118 49\n",
      "57\t8613\t0.0\t3\t7734 696 126 57\n",
      "58\t20058\t0.0\t3\t18758 1096 145 59\n",
      "59\t29540\t0.0\t3\t28171 1155 145 69\n",
      "60\t25900\t0.0\t3\t24370 1312 145 73\n",
      "61\t32939\t0.0\t3\t31565 1167 144 63\n",
      "62\t19704\t0.0\t3\t18632 865 129 78\n",
      "63\t16460\t0.0\t3\t15243 817 151 249\n",
      "64\t20076\t0.0\t3\t18197 1261 414 204\n",
      "65\t38156\t0.0\t3\t34034 3664 361 97\n",
      "66\t91828\t0.0\t3\t88253 3295 189 91\n",
      "67\t40474\t0.0\t3\t38906 1324 143 101\n",
      "68\t6879\t0.0\t3\t5849 758 165 107\n",
      "69\t10250\t0.0\t3\t9163 815 165 107\n",
      "70\t13451\t0.0\t3\t12199 1019 178 55\n",
      "71\t13978\t0.0\t3\t12765 1070 113 30\n",
      "72\t11082\t0.0\t3\t10360 610 83 29\n",
      "73\t3366\t0.0\t3\t2936 332 64 34\n",
      "74\t1393\t0.0\t3\t1134 200 45 14\n",
      "75\t640\t0.0\t3\t482 110 36 12\n",
      "76\t322\t0.0\t3\t230 63 22 7\n",
      "77\t189\t0.0\t3\t117 41 23 8\n",
      "78\t125\t0.0\t3\t73 31 12 9\n",
      "79\t78\t0.0\t3\t41 18 16 3\n",
      "80\t48\t0.0\t3\t23 15 7 3\n",
      "81\t34\t0.0\t3\t16 9 6 3\n",
      "82\t29\t0.0\t3\t13 7 2 7\n",
      "83\t28\t0.0\t3\t16 5 5 2\n",
      "84\t12\t0.0\t3\t5 2 1 4\n",
      "85\t11\t0.0\t3\t4 2 3 2\n",
      "86\t9\t0.0\t3\t6 0 3\n",
      "87\t11\t0.0\t3\t6 2 1 2\n",
      "88\t5\t0.0\t3\t2 1 2\n",
      "89\t9\t0.0\t3\t5 1 2 1\n",
      "90\t3\t0.0\t3\t1 1 0 1\n",
      "91\t6\t0.0\t3\t2 2 1 1\n",
      "92\t4\t0.0\t3\t2 2\n",
      "93\t3\t0.0\t3\t0 2 0 1\n",
      "94\t8\t0.0\t3\t0 4 1 3\n",
      "95\t5\t0.0\t3\t3 0 1 1\n",
      "96\t1\t0.0\t3\t0 1\n",
      "97\t1\t0.0\t3\t0 1\n",
      "98\t7\t0.0\t3\t2 2 2 1\n",
      "99\t3\t0.0\t3\t1 2\n",
      "100\t4\t0.0\t3\t2 0 1 1\n",
      "101\t2\t0.0\t3\t0 1 1\n",
      "102\t1\t0.0\t3\t1\n",
      "103\t3\t0.0\t3\t1 1 0 1\n",
      "104\t2\t0.0\t3\t1 0 0 1\n",
      "105\t1\t0.0\t3\t0 0 0 1\n",
      "106\t1\t0.0\t3\t0 1\n",
      "107\t1\t0.0\t3\t1\n",
      "108\t2\t0.0\t3\t1 0 0 1\n",
      "110\t2\t0.0\t3\t1 1\n",
      "112\t3\t0.0\t3\t1 1 0 1\n",
      "113\t2\t0.0\t3\t2\n",
      "114\t4\t0.0\t3\t1 3\n",
      "116\t2\t0.0\t3\t1 0 1\n",
      "117\t2\t0.0\t3\t1 1\n",
      "118\t4\t0.0\t3\t3 0 1\n",
      "119\t2\t0.0\t3\t1 0 1\n",
      "120\t1\t0.0\t3\t1\n",
      "122\t1\t0.0\t3\t1\n",
      "124\t2\t0.0\t3\t1 1\n",
      "125\t1\t0.0\t3\t1\n",
      "129\t1\t0.0\t3\t1\n",
      "134\t1\t0.0\t3\t0 0 0 1\n",
      "136\t2\t0.0\t3\t1 1\n",
      "137\t3\t0.0\t3\t1 2\n",
      "138\t2\t0.0\t3\t0 2\n",
      "140\t6\t0.0\t3\t0 2 3 1\n",
      "141\t6\t0.0\t3\t0 4 1 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 4662862 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.5%\n",
      "  C: 27.7%\n",
      "  G: 46.4%\n",
      "  T: 10.3%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t280204\t230559.2\t0\t280204\n",
      "4\t121227\t57639.8\t0\t121227\n",
      "5\t247355\t14410.0\t0\t247355\n",
      "8\t78154\t225.2\t0\t78154\n",
      "9\t158831\t56.3\t0\t158831\n",
      "10\t1100\t14.1\t1\t0 1100\n",
      "11\t165698\t3.5\t1\t163355 2343\n",
      "12\t1055\t0.9\t1\t0 1055\n",
      "13\t165978\t0.2\t1\t163116 2862\n",
      "14\t774\t0.1\t1\t0 774\n",
      "15\t82546\t0.0\t1\t79913 2633\n",
      "16\t165950\t0.0\t1\t162635 3315\n",
      "17\t1800\t0.0\t1\t0 1800\n",
      "18\t174349\t0.0\t1\t170469 3857 23\n",
      "19\t1501\t0.0\t1\t0 1312 189\n",
      "20\t110677\t0.0\t2\t103797 6349 531\n",
      "21\t309270\t0.0\t2\t272874 33820 2576\n",
      "22\t4059\t0.0\t2\t0 3241 818\n",
      "23\t433\t0.0\t2\t0 179 254\n",
      "24\t1582\t0.0\t2\t0 1404 178\n",
      "25\t77778\t0.0\t2\t64346 11926 1506\n",
      "26\t82015\t0.0\t2\t64780 13910 3325\n",
      "27\t125300\t0.0\t2\t101630 18737 4933\n",
      "28\t39072\t0.0\t2\t0 32325 6735 12\n",
      "29\t171914\t0.0\t2\t122877 38498 10509 30\n",
      "30\t8999\t0.0\t3\t0 4599 3117 1283\n",
      "31\t143667\t0.0\t3\t103806 30310 7499 2052\n",
      "32\t4066\t0.0\t3\t0 2336 1075 655\n",
      "33\t80387\t0.0\t3\t53677 18584 5714 2412\n",
      "34\t82107\t0.0\t3\t54400 18847 6269 2591\n",
      "35\t78770\t0.0\t3\t52162 18461 5648 2499\n",
      "36\t74959\t0.0\t3\t48476 17663 5921 2899\n",
      "37\t74038\t0.0\t3\t49573 17238 4942 2285\n",
      "38\t138013\t0.0\t3\t92807 30528 10224 4454\n",
      "39\t5038\t0.0\t3\t926 2197 1180 735\n",
      "40\t73564\t0.0\t3\t47295 17920 5719 2630\n",
      "41\t128176\t0.0\t3\t88632 27783 8236 3525\n",
      "42\t5222\t0.0\t3\t933 2965 819 505\n",
      "43\t106220\t0.0\t3\t74296 22348 6733 2843\n",
      "44\t21301\t0.0\t3\t11889 6188 2175 1049\n",
      "45\t67250\t0.0\t3\t44973 15743 4496 2038\n",
      "46\t107647\t0.0\t3\t74874 23004 6950 2819\n",
      "47\t6683\t0.0\t3\t2494 2542 1068 579\n",
      "48\t53195\t0.0\t3\t36777 11821 3298 1299\n",
      "49\t51575\t0.0\t3\t34555 12049 3609 1362\n",
      "50\t47796\t0.0\t3\t32151 10822 3370 1453\n",
      "51\t48508\t0.0\t3\t31967 12007 3216 1318\n",
      "52\t103313\t0.0\t3\t76750 19829 4848 1886\n",
      "53\t27328\t0.0\t3\t15283 7280 3152 1613\n",
      "54\t3674\t0.0\t3\t486 2380 539 269\n",
      "55\t100443\t0.0\t3\t72621 19794 5701 2327\n",
      "56\t11262\t0.0\t3\t5747 3187 1517 811\n",
      "57\t1399\t0.0\t3\t187 764 271 177\n",
      "58\t32599\t0.0\t3\t21942 7585 2167 905\n",
      "59\t51073\t0.0\t3\t36415 10428 3090 1140\n",
      "60\t6776\t0.0\t3\t3257 2154 893 472\n",
      "61\t25210\t0.0\t3\t17368 5630 1586 626\n",
      "62\t39238\t0.0\t3\t28485 7786 2124 843\n",
      "63\t5870\t0.0\t3\t2973 1886 651 360\n",
      "64\t21943\t0.0\t3\t15794 4486 1232 431\n",
      "65\t19060\t0.0\t3\t13796 3878 981 405\n",
      "66\t18905\t0.0\t3\t13296 4050 1097 462\n",
      "67\t16642\t0.0\t3\t11649 3640 938 415\n",
      "68\t15149\t0.0\t3\t10266 3497 971 415\n",
      "69\t13599\t0.0\t3\t9436 2958 825 380\n",
      "70\t12591\t0.0\t3\t8883 2675 770 263\n",
      "71\t11119\t0.0\t3\t7639 2466 707 307\n",
      "72\t10812\t0.0\t3\t7427 2368 646 371\n",
      "73\t9946\t0.0\t3\t6378 2124 730 714\n",
      "74\t11320\t0.0\t3\t6376 3169 1284 491\n",
      "75\t30031\t0.0\t3\t19762 7751 1862 656\n",
      "76\t71682\t0.0\t3\t53962 13092 3297 1331\n",
      "77\t13358\t0.0\t3\t7579 3442 1538 799\n",
      "78\t1679\t0.0\t3\t568 493 372 246\n",
      "79\t449\t0.0\t3\t99 132 119 99\n",
      "80\t160\t0.0\t3\t34 43 46 37\n",
      "81\t63\t0.0\t3\t13 18 20 12\n",
      "82\t31\t0.0\t3\t1 6 15 9\n",
      "83\t27\t0.0\t3\t6 6 6 9\n",
      "84\t19\t0.0\t3\t5 5 4 5\n",
      "85\t14\t0.0\t3\t2 3 5 4\n",
      "86\t7\t0.0\t3\t1 4 1 1\n",
      "87\t11\t0.0\t3\t0 2 4 5\n",
      "88\t14\t0.0\t3\t4 7 3\n",
      "89\t7\t0.0\t3\t2 2 1 2\n",
      "90\t9\t0.0\t3\t0 3 4 2\n",
      "91\t7\t0.0\t3\t2 2 3\n",
      "92\t7\t0.0\t3\t1 2 3 1\n",
      "93\t3\t0.0\t3\t1 1 1\n",
      "94\t8\t0.0\t3\t1 3 1 3\n",
      "95\t4\t0.0\t3\t2 0 1 1\n",
      "96\t5\t0.0\t3\t1 0 3 1\n",
      "97\t2\t0.0\t3\t0 0 1 1\n",
      "98\t3\t0.0\t3\t0 0 3\n",
      "99\t3\t0.0\t3\t2 0 1\n",
      "100\t2\t0.0\t3\t1 0 0 1\n",
      "101\t4\t0.0\t3\t1 0 2 1\n",
      "102\t2\t0.0\t3\t1 0 0 1\n",
      "103\t3\t0.0\t3\t0 1 1 1\n",
      "104\t8\t0.0\t3\t1 1 4 2\n",
      "105\t7\t0.0\t3\t1 2 0 4\n",
      "106\t6\t0.0\t3\t1 2 1 2\n",
      "107\t4\t0.0\t3\t1 2 0 1\n",
      "108\t9\t0.0\t3\t4 3 1 1\n",
      "109\t21\t0.0\t3\t8 7 4 2\n",
      "110\t24\t0.0\t3\t12 6 3 3\n",
      "111\t6\t0.0\t3\t3 1 1 1\n",
      "112\t5\t0.0\t3\t0 1 1 3\n",
      "113\t3\t0.0\t3\t1 1 0 1\n",
      "114\t2\t0.0\t3\t1 0 1\n",
      "115\t2\t0.0\t3\t0 1 0 1\n",
      "116\t5\t0.0\t3\t2 3\n",
      "117\t7\t0.0\t3\t2 4 0 1\n",
      "118\t2\t0.0\t3\t0 1 0 1\n",
      "119\t1\t0.0\t3\t0 1\n",
      "120\t2\t0.0\t3\t0 2\n",
      "121\t7\t0.0\t3\t3 2 1 1\n",
      "122\t5\t0.0\t3\t0 0 3 2\n",
      "124\t4\t0.0\t3\t3 1\n",
      "125\t2\t0.0\t3\t1 0 0 1\n",
      "126\t2\t0.0\t3\t0 0 1 1\n",
      "127\t2\t0.0\t3\t0 1 1\n",
      "128\t3\t0.0\t3\t1 1 1\n",
      "129\t4\t0.0\t3\t2 1 0 1\n",
      "130\t1\t0.0\t3\t0 1\n",
      "131\t2\t0.0\t3\t0 1 1\n",
      "133\t3\t0.0\t3\t0 2 0 1\n",
      "135\t3\t0.0\t3\t1 0 2\n",
      "136\t1\t0.0\t3\t0 0 0 1\n",
      "137\t1\t0.0\t3\t0 1\n",
      "138\t3\t0.0\t3\t0 0 2 1\n",
      "139\t2\t0.0\t3\t1 0 1\n",
      "140\t5\t0.0\t3\t2 3\n",
      "141\t10\t0.0\t3\t0 1 6 3\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_03_1.fastq.gz -p ./fastq_trimmed_sub/Liv_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-03_S40_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:40    20,441,278 reads @   2.0 µs/read;  30.20 M reads/minute\n",
      "Finished in 40.714 s (1.992 µs/read; 30.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         20,441,278\n",
      "  Read 1 with adapter:               3,559,053 (17.4%)\n",
      "  Read 2 with adapter:               3,598,987 (17.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             254,832 (1.2%)\n",
      "Pairs written (passing filters):    20,186,446 (98.8%)\n",
      "\n",
      "Total basepairs processed: 6,173,265,956 bp\n",
      "  Read 1: 3,086,632,978 bp\n",
      "  Read 2: 3,086,632,978 bp\n",
      "Quality-trimmed:              54,253,570 bp (0.9%)\n",
      "  Read 1:    11,062,995 bp\n",
      "  Read 2:    43,190,575 bp\n",
      "Total written (filtered):  5,512,398,572 bp (89.3%)\n",
      "  Read 1: 2,755,452,107 bp\n",
      "  Read 2: 2,756,946,465 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 3559053 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.6%\n",
      "  C: 26.8%\n",
      "  G: 39.8%\n",
      "  T: 18.7%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t362102\t319395.0\t0\t362102\n",
      "4\t171769\t79848.7\t0\t171769\n",
      "5\t255943\t19962.2\t0\t255943\n",
      "8\t74968\t311.9\t0\t74968\n",
      "9\t149959\t78.0\t0\t149959\n",
      "10\t1560\t19.5\t1\t0 1560\n",
      "11\t149899\t4.9\t1\t147706 2193\n",
      "12\t873\t1.2\t1\t0 873\n",
      "13\t73662\t0.3\t1\t71864 1798\n",
      "14\t74321\t0.1\t1\t72376 1945\n",
      "15\t73677\t0.0\t1\t71634 2043\n",
      "16\t72920\t0.0\t1\t69923 2997\n",
      "17\t141178\t0.0\t1\t138326 2852\n",
      "18\t1237\t0.0\t1\t0 1202 35\n",
      "19\t69462\t0.0\t1\t66996 2341 125\n",
      "20\t70829\t0.0\t2\t67050 3462 317\n",
      "21\t134388\t0.0\t2\t130889 2942 557\n",
      "22\t1529\t0.0\t2\t0 1415 114\n",
      "23\t68071\t0.0\t2\t65078 2603 390\n",
      "24\t67472\t0.0\t2\t64161 2907 404\n",
      "25\t65408\t0.0\t2\t62215 2826 367\n",
      "26\t64358\t0.0\t2\t60953 2973 432\n",
      "27\t63223\t0.0\t2\t59919 2838 466\n",
      "28\t62953\t0.0\t2\t58071 4485 393 4\n",
      "29\t118630\t0.0\t2\t114505 3475 641 9\n",
      "30\t1935\t0.0\t3\t0 1704 172 59\n",
      "31\t58675\t0.0\t3\t55247 2866 402 160\n",
      "32\t57460\t0.0\t3\t53869 3028 396 167\n",
      "33\t56190\t0.0\t3\t52233 3271 504 182\n",
      "34\t53976\t0.0\t3\t50603 2825 372 176\n",
      "35\t52674\t0.0\t3\t49452 2710 362 150\n",
      "36\t51364\t0.0\t3\t48118 2757 347 142\n",
      "37\t49138\t0.0\t3\t46050 2589 361 138\n",
      "38\t47366\t0.0\t3\t44465 2452 319 130\n",
      "39\t46492\t0.0\t3\t43658 2369 328 137\n",
      "40\t43927\t0.0\t3\t41220 2280 310 117\n",
      "41\t42230\t0.0\t3\t39707 2103 301 119\n",
      "42\t41707\t0.0\t3\t39193 2084 307 123\n",
      "43\t39450\t0.0\t3\t37073 1953 265 159\n",
      "44\t39649\t0.0\t3\t36802 2176 512 159\n",
      "45\t43893\t0.0\t3\t36455 6758 516 164\n",
      "46\t223277\t0.0\t3\t214795 7291 801 390\n",
      "47\t86624\t0.0\t3\t83697 2590 239 98\n",
      "48\t735\t0.0\t3\t343 330 38 24\n",
      "49\t1915\t0.0\t3\t1647 177 55 36\n",
      "50\t610\t0.0\t3\t443 105 42 20\n",
      "51\t836\t0.0\t3\t665 114 43 14\n",
      "52\t457\t0.0\t3\t318 100 25 14\n",
      "53\t958\t0.0\t3\t753 145 40 20\n",
      "54\t2207\t0.0\t3\t1932 203 51 21\n",
      "55\t987\t0.0\t3\t757 152 51 27\n",
      "56\t1598\t0.0\t3\t1328 195 56 19\n",
      "57\t2672\t0.0\t3\t2290 297 63 22\n",
      "58\t6974\t0.0\t3\t6439 451 61 23\n",
      "59\t12023\t0.0\t3\t11353 566 80 24\n",
      "60\t10817\t0.0\t3\t9996 734 55 32\n",
      "61\t15747\t0.0\t3\t15124 519 75 29\n",
      "62\t6778\t0.0\t3\t6350 351 48 29\n",
      "63\t4659\t0.0\t3\t4109 357 72 121\n",
      "64\t6792\t0.0\t3\t5972 597 168 55\n",
      "65\t15265\t0.0\t3\t14146 991 101 27\n",
      "66\t21208\t0.0\t3\t20463 645 55 45\n",
      "67\t3908\t0.0\t3\t3604 216 48 40\n",
      "68\t1171\t0.0\t3\t925 159 50 37\n",
      "69\t2620\t0.0\t3\t2324 196 57 43\n",
      "70\t3348\t0.0\t3\t3053 240 38 17\n",
      "71\t3785\t0.0\t3\t3474 251 39 21\n",
      "72\t2800\t0.0\t3\t2595 154 31 20\n",
      "73\t848\t0.0\t3\t707 101 27 13\n",
      "74\t355\t0.0\t3\t261 63 18 13\n",
      "75\t181\t0.0\t3\t125 35 14 7\n",
      "76\t93\t0.0\t3\t50 25 12 6\n",
      "77\t68\t0.0\t3\t31 16 15 6\n",
      "78\t32\t0.0\t3\t19 7 4 2\n",
      "79\t23\t0.0\t3\t16 2 1 4\n",
      "80\t28\t0.0\t3\t10 10 5 3\n",
      "81\t11\t0.0\t3\t4 2 3 2\n",
      "82\t11\t0.0\t3\t4 2 3 2\n",
      "83\t9\t0.0\t3\t1 2 3 3\n",
      "84\t8\t0.0\t3\t1 3 4\n",
      "85\t9\t0.0\t3\t2 4 2 1\n",
      "86\t10\t0.0\t3\t1 3 4 2\n",
      "87\t2\t0.0\t3\t2\n",
      "88\t1\t0.0\t3\t0 0 1\n",
      "89\t7\t0.0\t3\t1 1 4 1\n",
      "90\t3\t0.0\t3\t0 1 1 1\n",
      "91\t5\t0.0\t3\t2 2 1\n",
      "92\t3\t0.0\t3\t0 1 1 1\n",
      "93\t6\t0.0\t3\t2 1 1 2\n",
      "94\t7\t0.0\t3\t2 0 2 3\n",
      "95\t2\t0.0\t3\t0 0 0 2\n",
      "96\t2\t0.0\t3\t1 0 0 1\n",
      "97\t4\t0.0\t3\t0 3 1\n",
      "98\t1\t0.0\t3\t0 1\n",
      "101\t1\t0.0\t3\t0 1\n",
      "102\t4\t0.0\t3\t0 2 1 1\n",
      "104\t2\t0.0\t3\t0 0 0 2\n",
      "107\t1\t0.0\t3\t1\n",
      "108\t1\t0.0\t3\t0 1\n",
      "109\t2\t0.0\t3\t0 1 1\n",
      "110\t1\t0.0\t3\t0 1\n",
      "112\t1\t0.0\t3\t1\n",
      "113\t1\t0.0\t3\t0 0 1\n",
      "114\t1\t0.0\t3\t1\n",
      "115\t2\t0.0\t3\t0 0 0 2\n",
      "119\t2\t0.0\t3\t0 0 0 2\n",
      "121\t1\t0.0\t3\t0 0 0 1\n",
      "122\t1\t0.0\t3\t0 0 0 1\n",
      "123\t2\t0.0\t3\t0 0 1 1\n",
      "126\t1\t0.0\t3\t0 0 1\n",
      "128\t1\t0.0\t3\t0 1\n",
      "129\t1\t0.0\t3\t0 1\n",
      "132\t1\t0.0\t3\t0 0 1\n",
      "134\t2\t0.0\t3\t1 0 1\n",
      "135\t1\t0.0\t3\t0 1\n",
      "136\t1\t0.0\t3\t1\n",
      "137\t1\t0.0\t3\t1\n",
      "140\t3\t0.0\t3\t1 0 1 1\n",
      "141\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 3598987 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.1%\n",
      "  C: 27.9%\n",
      "  G: 44.0%\n",
      "  T: 9.8%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t438355\t319395.0\t0\t438355\n",
      "4\t172378\t79848.7\t0\t172378\n",
      "5\t261044\t19962.2\t0\t261044\n",
      "8\t75833\t311.9\t0\t75833\n",
      "9\t150120\t78.0\t0\t150120\n",
      "10\t1154\t19.5\t1\t0 1154\n",
      "11\t150213\t4.9\t1\t148149 2064\n",
      "12\t997\t1.2\t1\t0 997\n",
      "13\t146349\t0.3\t1\t144176 2173\n",
      "14\t683\t0.1\t1\t0 683\n",
      "15\t74103\t0.0\t1\t71941 2162\n",
      "16\t142019\t0.0\t1\t139559 2460\n",
      "17\t1567\t0.0\t1\t0 1567\n",
      "18\t146539\t0.0\t1\t143731 2784 24\n",
      "19\t1261\t0.0\t1\t0 1099 162\n",
      "20\t86088\t0.0\t2\t80681 5002 405\n",
      "21\t249150\t0.0\t2\t218061 29072 2017\n",
      "22\t3526\t0.0\t2\t0 2786 740\n",
      "23\t294\t0.0\t2\t0 107 187\n",
      "24\t1199\t0.0\t2\t0 1086 113\n",
      "25\t58308\t0.0\t2\t48519 8767 1022\n",
      "26\t64021\t0.0\t2\t49117 12067 2837\n",
      "27\t99211\t0.0\t2\t78494 16398 4319\n",
      "28\t24796\t0.0\t2\t0 20270 4522 4\n",
      "29\t118801\t0.0\t2\t86453 25229 7092 27\n",
      "30\t5635\t0.0\t3\t0 3222 1713 700\n",
      "31\t100751\t0.0\t3\t73127 20843 5267 1514\n",
      "32\t3344\t0.0\t3\t0 1832 939 573\n",
      "33\t54719\t0.0\t3\t37424 12063 3678 1554\n",
      "34\t52481\t0.0\t3\t35616 11571 3746 1548\n",
      "35\t52231\t0.0\t3\t34853 12276 3512 1590\n",
      "36\t84677\t0.0\t3\t60719 16630 5036 2292\n",
      "37\t10754\t0.0\t3\t4867 3652 1431 804\n",
      "38\t46072\t0.0\t3\t30843 10366 3463 1400\n",
      "39\t53270\t0.0\t3\t33900 13141 4222 2007\n",
      "40\t112371\t0.0\t3\t79735 23010 6766 2860\n",
      "41\t1784\t0.0\t3\t453 537 461 333\n",
      "42\t1672\t0.0\t3\t71 1353 170 78\n",
      "43\t67737\t0.0\t3\t48391 13796 3991 1559\n",
      "44\t6711\t0.0\t3\t3161 2244 872 434\n",
      "45\t37120\t0.0\t3\t25349 8390 2389 992\n",
      "46\t59978\t0.0\t3\t42757 12159 3737 1325\n",
      "47\t3099\t0.0\t3\t1154 1229 453 263\n",
      "48\t29392\t0.0\t3\t20882 6200 1651 659\n",
      "49\t27660\t0.0\t3\t19076 6197 1738 649\n",
      "50\t25999\t0.0\t3\t18140 5440 1671 748\n",
      "51\t25041\t0.0\t3\t17028 5878 1519 616\n",
      "52\t56935\t0.0\t3\t42518 10811 2604 1002\n",
      "53\t6073\t0.0\t3\t2860 1699 1006 508\n",
      "54\t1420\t0.0\t3\t115 1042 180 83\n",
      "55\t48157\t0.0\t3\t35452 9059 2617 1029\n",
      "56\t2089\t0.0\t3\t809 611 398 271\n",
      "57\t480\t0.0\t3\t34 312 91 43\n",
      "58\t14708\t0.0\t3\t10246 3209 895 358\n",
      "59\t23345\t0.0\t3\t17030 4521 1263 531\n",
      "60\t1597\t0.0\t3\t622 588 240 147\n",
      "61\t10911\t0.0\t3\t7815 2239 596 261\n",
      "62\t17524\t0.0\t3\t13038 3274 859 353\n",
      "63\t1529\t0.0\t3\t677 514 229 109\n",
      "64\t8544\t0.0\t3\t6279 1615 478 172\n",
      "65\t7041\t0.0\t3\t5189 1362 354 136\n",
      "66\t6822\t0.0\t3\t4942 1321 378 181\n",
      "67\t6069\t0.0\t3\t4367 1244 317 141\n",
      "68\t5527\t0.0\t3\t3904 1136 344 143\n",
      "69\t4903\t0.0\t3\t3429 1046 296 132\n",
      "70\t4385\t0.0\t3\t3172 863 256 94\n",
      "71\t3733\t0.0\t3\t2716 725 193 99\n",
      "72\t3593\t0.0\t3\t2463 765 205 160\n",
      "73\t3365\t0.0\t3\t2129 638 260 338\n",
      "74\t3348\t0.0\t3\t1983 822 409 134\n",
      "75\t8438\t0.0\t3\t5928 1854 456 200\n",
      "76\t15258\t0.0\t3\t11720 2567 695 276\n",
      "77\t2123\t0.0\t3\t1162 532 265 164\n",
      "78\t298\t0.0\t3\t99 93 59 47\n",
      "79\t80\t0.0\t3\t9 21 24 26\n",
      "80\t44\t0.0\t3\t6 10 13 15\n",
      "81\t24\t0.0\t3\t0 7 8 9\n",
      "82\t11\t0.0\t3\t1 3 3 4\n",
      "83\t10\t0.0\t3\t1 2 5 2\n",
      "84\t3\t0.0\t3\t1 0 1 1\n",
      "85\t7\t0.0\t3\t1 1 2 3\n",
      "86\t7\t0.0\t3\t1 0 2 4\n",
      "87\t2\t0.0\t3\t0 0 2\n",
      "88\t5\t0.0\t3\t0 0 3 2\n",
      "89\t3\t0.0\t3\t0 0 0 3\n",
      "90\t4\t0.0\t3\t0 0 2 2\n",
      "91\t2\t0.0\t3\t0 1 0 1\n",
      "92\t1\t0.0\t3\t0 1\n",
      "93\t4\t0.0\t3\t0 1 1 2\n",
      "94\t2\t0.0\t3\t0 0 0 2\n",
      "95\t1\t0.0\t3\t0 1\n",
      "96\t1\t0.0\t3\t1\n",
      "97\t4\t0.0\t3\t0 1 1 2\n",
      "98\t2\t0.0\t3\t0 0 1 1\n",
      "99\t3\t0.0\t3\t0 0 2 1\n",
      "100\t2\t0.0\t3\t0 0 0 2\n",
      "101\t2\t0.0\t3\t0 0 2\n",
      "104\t1\t0.0\t3\t0 0 1\n",
      "105\t1\t0.0\t3\t0 0 0 1\n",
      "106\t2\t0.0\t3\t0 0 1 1\n",
      "107\t2\t0.0\t3\t0 0 1 1\n",
      "109\t2\t0.0\t3\t0 0 0 2\n",
      "110\t1\t0.0\t3\t0 0 0 1\n",
      "111\t4\t0.0\t3\t0 0 1 3\n",
      "112\t3\t0.0\t3\t1 1 0 1\n",
      "116\t1\t0.0\t3\t0 0 0 1\n",
      "117\t1\t0.0\t3\t0 1\n",
      "118\t1\t0.0\t3\t1\n",
      "119\t2\t0.0\t3\t0 0 0 2\n",
      "120\t1\t0.0\t3\t0 0 0 1\n",
      "122\t1\t0.0\t3\t0 1\n",
      "128\t2\t0.0\t3\t1 0 0 1\n",
      "130\t1\t0.0\t3\t0 1\n",
      "131\t1\t0.0\t3\t0 1\n",
      "132\t1\t0.0\t3\t0 0 1\n",
      "133\t2\t0.0\t3\t1 1\n",
      "135\t1\t0.0\t3\t0 0 0 1\n",
      "136\t1\t0.0\t3\t0 0 0 1\n",
      "138\t1\t0.0\t3\t0 1\n",
      "140\t4\t0.0\t3\t1 0 1 2\n",
      "141\t4\t0.0\t3\t0 1 3\n",
      "This is cutadapt 4.9 with Python 3.12.3\n",
      "Command line parameters: -j 32 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_04_1.fastq.gz -p ./fastq_trimmed_sub/Liv_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-04_S41_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Processing paired-end reads on 32 cores ...\n",
      "Done           00:00:28    13,461,820 reads @   2.1 µs/read;  28.13 M reads/minute\n",
      "Finished in 28.804 s (2.140 µs/read; 28.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         13,461,820\n",
      "  Read 1 with adapter:               3,183,970 (23.7%)\n",
      "  Read 2 with adapter:               3,141,808 (23.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:             345,138 (2.6%)\n",
      "Pairs written (passing filters):    13,116,682 (97.4%)\n",
      "\n",
      "Total basepairs processed: 4,065,469,640 bp\n",
      "  Read 1: 2,032,734,820 bp\n",
      "  Read 2: 2,032,734,820 bp\n",
      "Quality-trimmed:              86,508,083 bp (2.1%)\n",
      "  Read 1:    30,677,465 bp\n",
      "  Read 2:    55,830,618 bp\n",
      "Total written (filtered):  3,520,748,879 bp (86.6%)\n",
      "  Read 1: 1,759,082,986 bp\n",
      "  Read 2: 1,761,665,893 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCA; Type: regular 3'; Length: 33; Trimmed: 3183970 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.2%\n",
      "  C: 29.4%\n",
      "  G: 36.6%\n",
      "  T: 18.7%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t270368\t210340.9\t0\t270368\n",
      "4\t117125\t52585.2\t0\t117125\n",
      "5\t199446\t13146.3\t0\t199446\n",
      "8\t61022\t205.4\t0\t61022\n",
      "9\t122393\t51.4\t0\t122393\n",
      "10\t1144\t12.8\t1\t0 1144\n",
      "11\t124980\t3.2\t1\t123314 1666\n",
      "12\t687\t0.8\t1\t0 687\n",
      "13\t62528\t0.2\t1\t61172 1356\n",
      "14\t61866\t0.1\t1\t60384 1482\n",
      "15\t61102\t0.0\t1\t59604 1498\n",
      "16\t61300\t0.0\t1\t58960 2340\n",
      "17\t118599\t0.0\t1\t116185 2414\n",
      "18\t961\t0.0\t1\t0 940 21\n",
      "19\t59705\t0.0\t1\t57678 1930 97\n",
      "20\t60972\t0.0\t2\t57890 2809 273\n",
      "21\t118190\t0.0\t2\t115226 2463 501\n",
      "22\t1273\t0.0\t2\t0 1177 96\n",
      "23\t60157\t0.0\t2\t57659 2192 306\n",
      "24\t59868\t0.0\t2\t57085 2455 328\n",
      "25\t57556\t0.0\t2\t54993 2231 332\n",
      "26\t56575\t0.0\t2\t53744 2498 333\n",
      "27\t55900\t0.0\t2\t53183 2345 372\n",
      "28\t55989\t0.0\t2\t52200 3504 281 4\n",
      "29\t108448\t0.0\t2\t104859 3017 566 6\n",
      "30\t1571\t0.0\t3\t0 1397 122 52\n",
      "31\t53998\t0.0\t3\t50945 2577 351 125\n",
      "32\t53949\t0.0\t3\t50892 2629 313 115\n",
      "33\t53329\t0.0\t3\t49970 2842 373 144\n",
      "34\t51915\t0.0\t3\t48937 2503 318 157\n",
      "35\t51111\t0.0\t3\t48199 2445 331 136\n",
      "36\t48563\t0.0\t3\t45722 2386 345 110\n",
      "37\t46782\t0.0\t3\t44208 2177 286 111\n",
      "38\t45779\t0.0\t3\t43112 2275 289 103\n",
      "39\t45166\t0.0\t3\t42652 2103 278 133\n",
      "40\t43812\t0.0\t3\t41278 2158 258 118\n",
      "41\t42599\t0.0\t3\t40228 2006 268 97\n",
      "42\t41526\t0.0\t3\t39329 1847 257 93\n",
      "43\t40754\t0.0\t3\t38523 1856 251 124\n",
      "44\t40412\t0.0\t3\t37816 2056 426 114\n",
      "45\t43910\t0.0\t3\t37161 6196 392 161\n",
      "46\t236955\t0.0\t3\t230385 5573 672 325\n",
      "47\t21273\t0.0\t3\t19807 1230 182 54\n",
      "48\t722\t0.0\t3\t184 477 44 17\n",
      "49\t18629\t0.0\t3\t17939 553 98 39\n",
      "50\t1428\t0.0\t3\t1024 324 55 25\n",
      "51\t6623\t0.0\t3\t6187 319 77 40\n",
      "52\t1151\t0.0\t3\t870 202 55 24\n",
      "53\t4247\t0.0\t3\t3732 425 59 31\n",
      "54\t11910\t0.0\t3\t11284 485 105 36\n",
      "55\t1667\t0.0\t3\t1233 342 67 25\n",
      "56\t8199\t0.0\t3\t7660 434 73 32\n",
      "57\t10133\t0.0\t3\t9417 609 70 37\n",
      "58\t15592\t0.0\t3\t14705 771 72 44\n",
      "59\t16117\t0.0\t3\t15289 702 92 34\n",
      "60\t14768\t0.0\t3\t13905 740 91 32\n",
      "61\t17833\t0.0\t3\t17043 687 78 25\n",
      "62\t10339\t0.0\t3\t9806 442 61 30\n",
      "63\t9079\t0.0\t3\t8499 451 57 72\n",
      "64\t10584\t0.0\t3\t9735 694 116 39\n",
      "65\t17923\t0.0\t3\t16512 1285 92 34\n",
      "66\t34217\t0.0\t3\t33284 828 62 43\n",
      "67\t5489\t0.0\t3\t5058 323 59 49\n",
      "68\t3410\t0.0\t3\t2931 328 71 80\n",
      "69\t8288\t0.0\t3\t7581 553 91 63\n",
      "70\t11190\t0.0\t3\t10360 724 81 25\n",
      "71\t12562\t0.0\t3\t11843 641 59 19\n",
      "72\t7751\t0.0\t3\t7318 356 49 28\n",
      "73\t1537\t0.0\t3\t1362 139 28 8\n",
      "74\t484\t0.0\t3\t360 83 29 12\n",
      "75\t178\t0.0\t3\t119 35 17 7\n",
      "76\t89\t0.0\t3\t55 18 7 9\n",
      "77\t56\t0.0\t3\t31 8 14 3\n",
      "78\t35\t0.0\t3\t16 12 4 3\n",
      "79\t28\t0.0\t3\t7 9 4 8\n",
      "80\t16\t0.0\t3\t9 1 3 3\n",
      "81\t16\t0.0\t3\t8 2 2 4\n",
      "82\t11\t0.0\t3\t5 2 3 1\n",
      "83\t7\t0.0\t3\t4 1 0 2\n",
      "84\t5\t0.0\t3\t1 3 0 1\n",
      "85\t4\t0.0\t3\t1 0 2 1\n",
      "86\t5\t0.0\t3\t2 1 0 2\n",
      "87\t2\t0.0\t3\t2\n",
      "88\t3\t0.0\t3\t1 1 1\n",
      "89\t6\t0.0\t3\t2 1 0 3\n",
      "90\t2\t0.0\t3\t0 1 1\n",
      "91\t2\t0.0\t3\t2\n",
      "92\t3\t0.0\t3\t0 0 3\n",
      "93\t1\t0.0\t3\t0 0 1\n",
      "94\t2\t0.0\t3\t0 1 0 1\n",
      "95\t2\t0.0\t3\t1 0 1\n",
      "96\t2\t0.0\t3\t1 0 0 1\n",
      "97\t1\t0.0\t3\t0 0 0 1\n",
      "98\t1\t0.0\t3\t0 0 1\n",
      "101\t3\t0.0\t3\t1 1 0 1\n",
      "102\t3\t0.0\t3\t1 0 0 2\n",
      "103\t4\t0.0\t3\t1 0 0 3\n",
      "104\t2\t0.0\t3\t1 0 0 1\n",
      "105\t3\t0.0\t3\t1 2\n",
      "106\t7\t0.0\t3\t1 2 2 2\n",
      "107\t3\t0.0\t3\t0 2 0 1\n",
      "108\t1\t0.0\t3\t0 0 1\n",
      "110\t1\t0.0\t3\t0 1\n",
      "111\t1\t0.0\t3\t0 0 1\n",
      "112\t1\t0.0\t3\t0 0 1\n",
      "113\t3\t0.0\t3\t1 0 1 1\n",
      "114\t2\t0.0\t3\t0 0 1 1\n",
      "115\t1\t0.0\t3\t0 1\n",
      "116\t1\t0.0\t3\t0 0 1\n",
      "118\t2\t0.0\t3\t0 0 1 1\n",
      "119\t1\t0.0\t3\t0 0 1\n",
      "120\t1\t0.0\t3\t0 0 0 1\n",
      "121\t1\t0.0\t3\t0 1\n",
      "122\t5\t0.0\t3\t2 1 0 2\n",
      "127\t1\t0.0\t3\t0 1\n",
      "128\t2\t0.0\t3\t0 1 0 1\n",
      "129\t1\t0.0\t3\t0 1\n",
      "131\t1\t0.0\t3\t0 0 1\n",
      "133\t2\t0.0\t3\t0 2\n",
      "135\t2\t0.0\t3\t1 1\n",
      "136\t1\t0.0\t3\t0 1\n",
      "137\t2\t0.0\t3\t1 0 1\n",
      "139\t1\t0.0\t3\t0 1\n",
      "140\t2\t0.0\t3\t1 1\n",
      "141\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: regular 3'; Length: 33; Trimmed: 3141808 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.2%\n",
      "  C: 29.1%\n",
      "  G: 42.0%\n",
      "  T: 10.6%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t282867\t210340.9\t0\t282867\n",
      "4\t120519\t52585.2\t0\t120519\n",
      "5\t201646\t13146.3\t0\t201646\n",
      "8\t61223\t205.4\t0\t61223\n",
      "9\t122129\t51.4\t0\t122129\n",
      "10\t1047\t12.8\t1\t0 1047\n",
      "11\t124953\t3.2\t1\t123170 1783\n",
      "12\t860\t0.8\t1\t0 860\n",
      "13\t123078\t0.2\t1\t121217 1861\n",
      "14\t611\t0.1\t1\t0 611\n",
      "15\t61341\t0.0\t1\t59429 1912\n",
      "16\t119102\t0.0\t1\t117023 2079\n",
      "17\t1451\t0.0\t1\t0 1451\n",
      "18\t137706\t0.0\t1\t134772 2923 11\n",
      "19\t1092\t0.0\t1\t0 959 133\n",
      "20\t73255\t0.0\t2\t68621 4282 352\n",
      "21\t208053\t0.0\t2\t184217 21936 1900\n",
      "22\t2313\t0.0\t2\t0 1632 681\n",
      "23\t293\t0.0\t2\t0 122 171\n",
      "24\t1191\t0.0\t2\t0 1085 106\n",
      "25\t51403\t0.0\t2\t41988 8057 1358\n",
      "26\t55820\t0.0\t2\t42195 10672 2953\n",
      "27\t86215\t0.0\t2\t67062 14400 4753\n",
      "28\t22542\t0.0\t2\t0 18320 4218 4\n",
      "29\t110476\t0.0\t2\t78559 23924 7974 19\n",
      "30\t5349\t0.0\t3\t0 2990 1585 774\n",
      "31\t90471\t0.0\t3\t65568 18385 4984 1534\n",
      "32\t2376\t0.0\t3\t0 1395 617 364\n",
      "33\t51954\t0.0\t3\t34465 12094 3696 1699\n",
      "34\t82225\t0.0\t3\t55914 16900 6336 3075\n",
      "35\t21806\t0.0\t3\t11926 5877 2591 1412\n",
      "36\t41143\t0.0\t3\t27699 9486 2764 1194\n",
      "37\t77171\t0.0\t3\t54109 15356 5231 2475\n",
      "38\t10399\t0.0\t3\t4986 3228 1379 806\n",
      "39\t45048\t0.0\t3\t28941 10236 3868 2003\n",
      "40\t40426\t0.0\t3\t26857 9407 2865 1297\n",
      "41\t57193\t0.0\t3\t39921 11424 3860 1988\n",
      "42\t22457\t0.0\t3\t13458 6084 1987 928\n",
      "43\t57297\t0.0\t3\t39771 11554 4011 1961\n",
      "44\t19000\t0.0\t3\t11552 4894 1662 892\n",
      "45\t38118\t0.0\t3\t25621 8425 2663 1409\n",
      "46\t62715\t0.0\t3\t43623 12809 4366 1917\n",
      "47\t3133\t0.0\t3\t1066 1229 510 328\n",
      "48\t30626\t0.0\t3\t21194 6406 2118 908\n",
      "49\t29260\t0.0\t3\t19752 6493 2065 950\n",
      "50\t27321\t0.0\t3\t18426 5975 1945 975\n",
      "51\t27366\t0.0\t3\t18068 6489 1913 896\n",
      "52\t61105\t0.0\t3\t45150 11182 3315 1458\n",
      "53\t9325\t0.0\t3\t4678 2466 1378 803\n",
      "54\t1846\t0.0\t3\t190 1266 240 150\n",
      "55\t54778\t0.0\t3\t39457 10403 3354 1564\n",
      "56\t3613\t0.0\t3\t1574 1061 624 354\n",
      "57\t643\t0.0\t3\t59 397 122 65\n",
      "58\t17253\t0.0\t3\t11659 3785 1216 593\n",
      "59\t27107\t0.0\t3\t19348 5255 1753 751\n",
      "60\t2378\t0.0\t3\t969 818 378 213\n",
      "61\t13349\t0.0\t3\t9231 2850 835 433\n",
      "62\t20736\t0.0\t3\t15008 3935 1217 576\n",
      "63\t2066\t0.0\t3\t961 659 273 173\n",
      "64\t10722\t0.0\t3\t7703 2068 662 289\n",
      "65\t9247\t0.0\t3\t6680 1758 594 215\n",
      "66\t8389\t0.0\t3\t5933 1667 509 280\n",
      "67\t7717\t0.0\t3\t5401 1607 478 231\n",
      "68\t7069\t0.0\t3\t4772 1528 526 243\n",
      "69\t6092\t0.0\t3\t4245 1204 451 192\n",
      "70\t5809\t0.0\t3\t4109 1166 355 179\n",
      "71\t4870\t0.0\t3\t3366 988 356 160\n",
      "72\t4794\t0.0\t3\t3184 1057 353 200\n",
      "73\t4361\t0.0\t3\t2868 865 358 270\n",
      "74\t4641\t0.0\t3\t2731 1240 484 186\n",
      "75\t12074\t0.0\t3\t8315 2642 772 345\n",
      "76\t23477\t0.0\t3\t17680 4019 1211 567\n",
      "77\t3501\t0.0\t3\t1970 834 422 275\n",
      "78\t474\t0.0\t3\t126 133 120 95\n",
      "79\t133\t0.0\t3\t30 36 31 36\n",
      "80\t52\t0.0\t3\t7 13 19 13\n",
      "81\t28\t0.0\t3\t2 9 6 11\n",
      "82\t8\t0.0\t3\t0 3 2 3\n",
      "83\t8\t0.0\t3\t0 1 4 3\n",
      "84\t5\t0.0\t3\t0 0 0 5\n",
      "85\t6\t0.0\t3\t1 3 0 2\n",
      "86\t13\t0.0\t3\t1 3 1 8\n",
      "87\t5\t0.0\t3\t0 2 1 2\n",
      "88\t3\t0.0\t3\t1 0 0 2\n",
      "89\t2\t0.0\t3\t1 0 0 1\n",
      "90\t4\t0.0\t3\t0 0 1 3\n",
      "91\t2\t0.0\t3\t0 0 2\n",
      "92\t1\t0.0\t3\t0 1\n",
      "93\t6\t0.0\t3\t0 1 3 2\n",
      "94\t6\t0.0\t3\t1 2 0 3\n",
      "96\t1\t0.0\t3\t0 0 0 1\n",
      "97\t1\t0.0\t3\t0 0 1\n",
      "98\t2\t0.0\t3\t2\n",
      "99\t1\t0.0\t3\t0 0 0 1\n",
      "100\t1\t0.0\t3\t0 0 1\n",
      "102\t3\t0.0\t3\t0 1 2\n",
      "104\t3\t0.0\t3\t0 1 2\n",
      "105\t1\t0.0\t3\t0 0 1\n",
      "106\t2\t0.0\t3\t0 1 0 1\n",
      "107\t3\t0.0\t3\t0 0 2 1\n",
      "108\t1\t0.0\t3\t1\n",
      "109\t6\t0.0\t3\t4 2\n",
      "110\t5\t0.0\t3\t3 1 1\n",
      "111\t2\t0.0\t3\t0 0 0 2\n",
      "112\t3\t0.0\t3\t0 1 0 2\n",
      "113\t2\t0.0\t3\t1 0 0 1\n",
      "114\t1\t0.0\t3\t0 0 0 1\n",
      "119\t1\t0.0\t3\t0 0 1\n",
      "123\t2\t0.0\t3\t0 0 1 1\n",
      "127\t1\t0.0\t3\t0 0 1\n",
      "129\t1\t0.0\t3\t0 0 0 1\n",
      "131\t1\t0.0\t3\t0 1\n",
      "135\t1\t0.0\t3\t0 1\n",
      "137\t1\t0.0\t3\t1\n",
      "140\t2\t0.0\t3\t0 2\n",
      "141\t2\t0.0\t3\t0 0 1 1\n"
     ]
    }
   ],
   "source": [
    "!mkdir fastq_trimmed_sub\n",
    "    #creates the directory that will hold the trimmed fastq files \n",
    "\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_01_1.fastq.gz -p ./fastq_trimmed_sub/Int_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-01_S42_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-01_S42_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_02_1.fastq.gz -p ./fastq_trimmed_sub/Int_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-02_S43_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-02_S43_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_03_1.fastq.gz -p ./fastq_trimmed_sub/Int_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-03_S44_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-03_S44_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Int_04_1.fastq.gz -p ./fastq_trimmed_sub/Int_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-04_S45_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Int-04_S45_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_01_1.fastq.gz -p ./fastq_trimmed_sub/Liv_01_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-01_S38_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-01_S38_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_02_1.fastq.gz -p ./fastq_trimmed_sub/Liv_02_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-02_S39_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-02_S39_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_03_1.fastq.gz -p ./fastq_trimmed_sub/Liv_03_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-03_S40_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-03_S40_L005_R2_001.fastq.gz\n",
    "!cutadapt -j 0 -m 20 --nextseq-trim=10 -u 10 -U 10 -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o ./fastq_trimmed_sub/Liv_04_1.fastq.gz -p ./fastq_trimmed_sub/Liv_04_2.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-04_S41_L005_R1_001.fastq.gz /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/Liv-04_S41_L005_R2_001.fastq.gz"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fasta Quality Control Check\n",
    "-   Performs quality analysis on the Worm Study genomes and the trimmed worm Study genome\n",
    "-   Generates a \n",
    "-   Confirms if the trimming improved the quality of the reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int_01_1.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int_01_2.fastq.gz\n",
      "Started analysis of Int_02_1.fastq.gz\n",
      "Started analysis of Int_02_2.fastq.gz\n",
      "Started analysis of Int_03_1.fastq.gz\n",
      "Started analysis of Int_03_2.fastq.gz\n",
      "Started analysis of Int_04_1.fastq.gz\n",
      "Started analysis of Int_04_2.fastq.gz\n",
      "Started analysis of Liv_01_1.fastq.gz\n",
      "Started analysis of Liv_01_2.fastq.gz\n",
      "Approx 5% complete for Int_01_1.fastq.gz\n",
      "Started analysis of Liv_02_1.fastq.gz\n",
      "Started analysis of Liv_02_2.fastq.gz\n",
      "Approx 5% complete for Int_01_2.fastq.gz\n",
      "Started analysis of Liv_03_1.fastq.gz\n",
      "Started analysis of Liv_03_2.fastq.gz\n",
      "Started analysis of Liv_04_1.fastq.gz\n",
      "Started analysis of Liv_04_2.fastq.gz\n",
      "Approx 5% complete for Int_03_1.fastq.gz\n",
      "Approx 5% complete for Int_02_1.fastq.gz\n",
      "Approx 5% complete for Int_02_2.fastq.gz\n",
      "Approx 5% complete for Int_03_2.fastq.gz\n",
      "Approx 5% complete for Int_04_1.fastq.gz\n",
      "Approx 10% complete for Int_01_1.fastq.gz\n",
      "Approx 5% complete for Liv_02_1.fastq.gz\n",
      "Approx 10% complete for Int_01_2.fastq.gz\n",
      "Approx 5% complete for Int_04_2.fastq.gz\n",
      "Approx 5% complete for Liv_02_2.fastq.gz\n",
      "Approx 5% complete for Liv_01_1.fastq.gz\n",
      "Approx 5% complete for Liv_01_2.fastq.gz\n",
      "Approx 5% complete for Liv_04_1.fastq.gz\n",
      "Approx 5% complete for Liv_04_2.fastq.gz\n",
      "Approx 10% complete for Int_03_1.fastq.gz\n",
      "Approx 10% complete for Int_02_1.fastq.gz\n",
      "Approx 10% complete for Int_03_2.fastq.gz\n",
      "Approx 5% complete for Liv_03_1.fastq.gz\n",
      "Approx 10% complete for Int_02_2.fastq.gz\n",
      "Approx 15% complete for Int_01_1.fastq.gz\n",
      "Approx 5% complete for Liv_03_2.fastq.gz\n",
      "Approx 15% complete for Int_01_2.fastq.gz\n",
      "Approx 10% complete for Liv_02_1.fastq.gz\n",
      "Approx 10% complete for Int_04_1.fastq.gz\n",
      "Approx 10% complete for Liv_02_2.fastq.gz\n",
      "Approx 10% complete for Int_04_2.fastq.gz\n",
      "Approx 10% complete for Liv_04_2.fastq.gz\n",
      "Approx 10% complete for Liv_04_1.fastq.gz\n",
      "Approx 15% complete for Int_03_1.fastq.gz\n",
      "Approx 10% complete for Liv_01_1.fastq.gz\n",
      "Approx 10% complete for Liv_01_2.fastq.gz\n",
      "Approx 20% complete for Int_01_1.fastq.gz\n",
      "Approx 20% complete for Int_01_2.fastq.gz\n",
      "Approx 15% complete for Int_03_2.fastq.gz\n",
      "Approx 15% complete for Liv_02_1.fastq.gz\n",
      "Approx 15% complete for Int_02_1.fastq.gz\n",
      "Approx 15% complete for Int_02_2.fastq.gz\n",
      "Approx 15% complete for Liv_02_2.fastq.gz\n",
      "Approx 15% complete for Liv_04_2.fastq.gz\n",
      "Approx 10% complete for Liv_03_1.fastq.gz\n",
      "Approx 15% complete for Liv_04_1.fastq.gz\n",
      "Approx 15% complete for Int_04_1.fastq.gz\n",
      "Approx 10% complete for Liv_03_2.fastq.gz\n",
      "Approx 20% complete for Int_03_1.fastq.gz\n",
      "Approx 15% complete for Int_04_2.fastq.gz\n",
      "Approx 25% complete for Int_01_1.fastq.gz\n",
      "Approx 25% complete for Int_01_2.fastq.gz\n",
      "Approx 20% complete for Liv_02_1.fastq.gz\n",
      "Approx 20% complete for Int_03_2.fastq.gz\n",
      "Approx 15% complete for Liv_01_1.fastq.gz\n",
      "Approx 20% complete for Int_02_1.fastq.gz\n",
      "Approx 15% complete for Liv_01_2.fastq.gz\n",
      "Approx 20% complete for Liv_02_2.fastq.gz\n",
      "Approx 20% complete for Int_02_2.fastq.gz\n",
      "Approx 20% complete for Liv_04_2.fastq.gz\n",
      "Approx 20% complete for Liv_04_1.fastq.gz\n",
      "Approx 25% complete for Int_03_1.fastq.gz\n",
      "Approx 30% complete for Int_01_2.fastq.gz\n",
      "Approx 30% complete for Int_01_1.fastq.gz\n",
      "Approx 20% complete for Int_04_1.fastq.gz\n",
      "Approx 15% complete for Liv_03_1.fastq.gz\n",
      "Approx 25% complete for Liv_02_1.fastq.gz\n",
      "Approx 20% complete for Int_04_2.fastq.gz\n",
      "Approx 25% complete for Int_03_2.fastq.gz\n",
      "Approx 15% complete for Liv_03_2.fastq.gz\n",
      "Approx 25% complete for Liv_02_2.fastq.gz\n",
      "Approx 25% complete for Liv_04_2.fastq.gz\n",
      "Approx 25% complete for Int_02_1.fastq.gz\n",
      "Approx 35% complete for Int_01_2.fastq.gz\n",
      "Approx 35% complete for Int_01_1.fastq.gz\n",
      "Approx 25% complete for Liv_04_1.fastq.gz\n",
      "Approx 30% complete for Int_03_1.fastq.gz\n",
      "Approx 25% complete for Int_02_2.fastq.gz\n",
      "Approx 20% complete for Liv_01_2.fastq.gz\n",
      "Approx 20% complete for Liv_01_1.fastq.gz\n",
      "Approx 30% complete for Liv_02_1.fastq.gz\n",
      "Approx 25% complete for Int_04_1.fastq.gz\n",
      "Approx 25% complete for Int_04_2.fastq.gz\n",
      "Approx 30% complete for Int_03_2.fastq.gz\n",
      "Approx 20% complete for Liv_03_1.fastq.gz\n",
      "Approx 30% complete for Liv_02_2.fastq.gz\n",
      "Approx 30% complete for Liv_04_2.fastq.gz\n",
      "Approx 40% complete for Int_01_2.fastq.gz\n",
      "Approx 40% complete for Int_01_1.fastq.gz\n",
      "Approx 30% complete for Liv_04_1.fastq.gz\n",
      "Approx 35% complete for Int_03_1.fastq.gz\n",
      "Approx 30% complete for Int_02_1.fastq.gz\n",
      "Approx 30% complete for Int_02_2.fastq.gz\n",
      "Approx 35% complete for Liv_02_1.fastq.gz\n",
      "Approx 20% complete for Liv_03_2.fastq.gz\n",
      "Approx 25% complete for Liv_01_2.fastq.gz\n",
      "Approx 25% complete for Liv_01_1.fastq.gz\n",
      "Approx 35% complete for Int_03_2.fastq.gz\n",
      "Approx 45% complete for Int_01_2.fastq.gz\n",
      "Approx 45% complete for Int_01_1.fastq.gz\n",
      "Approx 35% complete for Liv_02_2.fastq.gz\n",
      "Approx 35% complete for Liv_04_2.fastq.gz\n",
      "Approx 30% complete for Int_04_1.fastq.gz\n",
      "Approx 30% complete for Int_04_2.fastq.gz\n",
      "Approx 40% complete for Int_03_1.fastq.gz\n",
      "Approx 35% complete for Liv_04_1.fastq.gz\n",
      "Approx 40% complete for Liv_02_1.fastq.gz\n",
      "Approx 25% complete for Liv_03_1.fastq.gz\n",
      "Approx 35% complete for Int_02_1.fastq.gz\n",
      "Approx 35% complete for Int_02_2.fastq.gz\n",
      "Approx 50% complete for Int_01_2.fastq.gz\n",
      "Approx 50% complete for Int_01_1.fastq.gz\n",
      "Approx 40% complete for Int_03_2.fastq.gz\n",
      "Approx 40% complete for Liv_02_2.fastq.gz\n",
      "Approx 40% complete for Liv_04_2.fastq.gz\n",
      "Approx 30% complete for Liv_01_2.fastq.gz\n",
      "Approx 25% complete for Liv_03_2.fastq.gz\n",
      "Approx 30% complete for Liv_01_1.fastq.gz\n",
      "Approx 45% complete for Int_03_1.fastq.gz\n",
      "Approx 40% complete for Liv_04_1.fastq.gz\n",
      "Approx 35% complete for Int_04_1.fastq.gz\n",
      "Approx 35% complete for Int_04_2.fastq.gz\n",
      "Approx 45% complete for Liv_02_1.fastq.gz\n",
      "Approx 55% complete for Int_01_2.fastq.gz\n",
      "Approx 40% complete for Int_02_1.fastq.gz\n",
      "Approx 55% complete for Int_01_1.fastq.gz\n",
      "Approx 30% complete for Liv_03_1.fastq.gz\n",
      "Approx 40% complete for Int_02_2.fastq.gz\n",
      "Approx 45% complete for Liv_02_2.fastq.gz\n",
      "Approx 45% complete for Liv_04_2.fastq.gz\n",
      "Approx 45% complete for Int_03_2.fastq.gz\n",
      "Approx 50% complete for Int_03_1.fastq.gz\n",
      "Approx 50% complete for Liv_02_1.fastq.gz\n",
      "Approx 45% complete for Liv_04_1.fastq.gz\n",
      "Approx 35% complete for Liv_01_2.fastq.gz\n",
      "Approx 60% complete for Int_01_2.fastq.gz\n",
      "Approx 35% complete for Liv_01_1.fastq.gz\n",
      "Approx 40% complete for Int_04_1.fastq.gz\n",
      "Approx 40% complete for Int_04_2.fastq.gz\n",
      "Approx 60% complete for Int_01_1.fastq.gz\n",
      "Approx 30% complete for Liv_03_2.fastq.gz\n",
      "Approx 45% complete for Int_02_1.fastq.gz\n",
      "Approx 50% complete for Liv_04_2.fastq.gz\n",
      "Approx 50% complete for Liv_02_2.fastq.gz\n",
      "Approx 45% complete for Int_02_2.fastq.gz\n",
      "Approx 50% complete for Int_03_2.fastq.gz\n",
      "Approx 55% complete for Int_03_1.fastq.gz\n",
      "Approx 55% complete for Liv_02_1.fastq.gz\n",
      "Approx 35% complete for Liv_03_1.fastq.gz\n",
      "Approx 65% complete for Int_01_2.fastq.gz\n",
      "Approx 50% complete for Liv_04_1.fastq.gz\n",
      "Approx 65% complete for Int_01_1.fastq.gz\n",
      "Approx 40% complete for Liv_01_2.fastq.gz\n",
      "Approx 45% complete for Int_04_2.fastq.gz\n",
      "Approx 45% complete for Int_04_1.fastq.gz\n",
      "Approx 40% complete for Liv_01_1.fastq.gz\n",
      "Approx 55% complete for Liv_04_2.fastq.gz\n",
      "Approx 55% complete for Liv_02_2.fastq.gz\n",
      "Approx 50% complete for Int_02_1.fastq.gz\n",
      "Approx 60% complete for Liv_02_1.fastq.gz\n",
      "Approx 60% complete for Int_03_1.fastq.gz\n",
      "Approx 70% complete for Int_01_2.fastq.gz\n",
      "Approx 55% complete for Int_03_2.fastq.gz\n",
      "Approx 35% complete for Liv_03_2.fastq.gz\n",
      "Approx 50% complete for Int_02_2.fastq.gz\n",
      "Approx 55% complete for Liv_04_1.fastq.gz\n",
      "Approx 70% complete for Int_01_1.fastq.gz\n",
      "Approx 40% complete for Liv_03_1.fastq.gz\n",
      "Approx 60% complete for Liv_04_2.fastq.gz\n",
      "Approx 60% complete for Liv_02_2.fastq.gz\n",
      "Approx 75% complete for Int_01_2.fastq.gz\n",
      "Approx 65% complete for Liv_02_1.fastq.gz\n",
      "Approx 65% complete for Int_03_1.fastq.gz\n",
      "Approx 50% complete for Int_04_2.fastq.gz\n",
      "Approx 50% complete for Int_04_1.fastq.gz\n",
      "Approx 45% complete for Liv_01_2.fastq.gz\n",
      "Approx 55% complete for Int_02_1.fastq.gz\n",
      "Approx 75% complete for Int_01_1.fastq.gz\n",
      "Approx 60% complete for Int_03_2.fastq.gz\n",
      "Approx 60% complete for Liv_04_1.fastq.gz\n",
      "Approx 45% complete for Liv_01_1.fastq.gz\n",
      "Approx 55% complete for Int_02_2.fastq.gz\n",
      "Approx 65% complete for Liv_02_2.fastq.gz\n",
      "Approx 80% complete for Int_01_2.fastq.gz\n",
      "Approx 65% complete for Liv_04_2.fastq.gz\n",
      "Approx 40% complete for Liv_03_2.fastq.gz\n",
      "Approx 70% complete for Liv_02_1.fastq.gz\n",
      "Approx 70% complete for Int_03_1.fastq.gz\n",
      "Approx 80% complete for Int_01_1.fastq.gz\n",
      "Approx 45% complete for Liv_03_1.fastq.gz\n",
      "Approx 65% complete for Int_03_2.fastq.gz\n",
      "Approx 65% complete for Liv_04_1.fastq.gz\n",
      "Approx 55% complete for Int_04_2.fastq.gz\n",
      "Approx 60% complete for Int_02_1.fastq.gz\n",
      "Approx 55% complete for Int_04_1.fastq.gz\n",
      "Approx 50% complete for Liv_01_2.fastq.gz\n",
      "Approx 60% complete for Int_02_2.fastq.gz\n",
      "Approx 85% complete for Int_01_2.fastq.gz\n",
      "Approx 50% complete for Liv_01_1.fastq.gz\n",
      "Approx 70% complete for Liv_02_2.fastq.gz\n",
      "Approx 70% complete for Liv_04_2.fastq.gz\n",
      "Approx 75% complete for Liv_02_1.fastq.gz\n",
      "Approx 85% complete for Int_01_1.fastq.gz\n",
      "Approx 75% complete for Int_03_1.fastq.gz\n",
      "Approx 70% complete for Int_03_2.fastq.gz\n",
      "Approx 70% complete for Liv_04_1.fastq.gz\n",
      "Approx 65% complete for Int_02_1.fastq.gz\n",
      "Approx 45% complete for Liv_03_2.fastq.gz\n",
      "Approx 60% complete for Int_04_2.fastq.gz\n",
      "Approx 90% complete for Int_01_2.fastq.gz\n",
      "Approx 60% complete for Int_04_1.fastq.gz\n",
      "Approx 75% complete for Liv_02_2.fastq.gz\n",
      "Approx 75% complete for Liv_04_2.fastq.gz\n",
      "Approx 80% complete for Liv_02_1.fastq.gz\n",
      "Approx 65% complete for Int_02_2.fastq.gz\n",
      "Approx 90% complete for Int_01_1.fastq.gz\n",
      "Approx 50% complete for Liv_03_1.fastq.gz\n",
      "Approx 55% complete for Liv_01_2.fastq.gz\n",
      "Approx 80% complete for Int_03_1.fastq.gz\n",
      "Approx 55% complete for Liv_01_1.fastq.gz\n",
      "Approx 75% complete for Int_03_2.fastq.gz\n",
      "Approx 75% complete for Liv_04_1.fastq.gz\n",
      "Approx 95% complete for Int_01_2.fastq.gz\n",
      "Approx 80% complete for Liv_02_2.fastq.gz\n",
      "Approx 70% complete for Int_02_1.fastq.gz\n",
      "Approx 85% complete for Liv_02_1.fastq.gz\n",
      "Approx 80% complete for Liv_04_2.fastq.gz\n",
      "Approx 95% complete for Int_01_1.fastq.gz\n",
      "Approx 65% complete for Int_04_2.fastq.gz\n",
      "Approx 70% complete for Int_02_2.fastq.gz\n",
      "Approx 85% complete for Int_03_1.fastq.gz\n",
      "Approx 65% complete for Int_04_1.fastq.gz\n",
      "Approx 50% complete for Liv_03_2.fastq.gz\n",
      "Analysis complete for Int_01_2.fastq.gz\n",
      "Approx 60% complete for Liv_01_2.fastq.gz\n",
      "Approx 80% complete for Int_03_2.fastq.gz\n",
      "Approx 80% complete for Liv_04_1.fastq.gz\n",
      "Approx 60% complete for Liv_01_1.fastq.gz\n",
      "Approx 55% complete for Liv_03_1.fastq.gz\n",
      "Approx 85% complete for Liv_02_2.fastq.gz\n",
      "Analysis complete for Int_01_1.fastq.gz\n",
      "Approx 90% complete for Liv_02_1.fastq.gz\n",
      "Approx 85% complete for Liv_04_2.fastq.gz\n",
      "Approx 75% complete for Int_02_1.fastq.gz\n",
      "Approx 90% complete for Int_03_1.fastq.gz\n",
      "Approx 70% complete for Int_04_2.fastq.gz\n",
      "Approx 75% complete for Int_02_2.fastq.gz\n",
      "Approx 70% complete for Int_04_1.fastq.gz\n",
      "Approx 85% complete for Liv_04_1.fastq.gz\n",
      "Approx 85% complete for Int_03_2.fastq.gz\n",
      "Approx 90% complete for Liv_02_2.fastq.gz\n",
      "Approx 95% complete for Liv_02_1.fastq.gz\n",
      "Approx 55% complete for Liv_03_2.fastq.gz\n",
      "Approx 65% complete for Liv_01_2.fastq.gz\n",
      "Approx 90% complete for Liv_04_2.fastq.gz\n",
      "Approx 80% complete for Int_02_1.fastq.gz\n",
      "Approx 95% complete for Int_03_1.fastq.gz\n",
      "Approx 65% complete for Liv_01_1.fastq.gz\n",
      "Approx 60% complete for Liv_03_1.fastq.gz\n",
      "Approx 80% complete for Int_02_2.fastq.gz\n",
      "Approx 75% complete for Int_04_2.fastq.gz\n",
      "Approx 90% complete for Int_03_2.fastq.gz\n",
      "Approx 90% complete for Liv_04_1.fastq.gz\n",
      "Analysis complete for Liv_02_1.fastq.gz\n",
      "Approx 95% complete for Liv_02_2.fastq.gz\n",
      "Approx 95% complete for Liv_04_2.fastq.gz\n",
      "Approx 75% complete for Int_04_1.fastq.gz\n",
      "Analysis complete for Int_03_1.fastq.gz\n",
      "Approx 85% complete for Int_02_1.fastq.gz\n",
      "Approx 70% complete for Liv_01_2.fastq.gz\n",
      "Approx 60% complete for Liv_03_2.fastq.gz\n",
      "Approx 95% complete for Int_03_2.fastq.gz\n",
      "Approx 70% complete for Liv_01_1.fastq.gz\n",
      "Analysis complete for Liv_02_2.fastq.gz\n",
      "Approx 95% complete for Liv_04_1.fastq.gz\n",
      "Approx 85% complete for Int_02_2.fastq.gz\n",
      "Analysis complete for Liv_04_2.fastq.gz\n",
      "Approx 80% complete for Int_04_2.fastq.gz\n",
      "Approx 65% complete for Liv_03_1.fastq.gz\n",
      "Approx 80% complete for Int_04_1.fastq.gz\n",
      "Approx 90% complete for Int_02_1.fastq.gz\n",
      "Analysis complete for Int_03_2.fastq.gz\n",
      "Analysis complete for Liv_04_1.fastq.gz\n",
      "Approx 75% complete for Liv_01_2.fastq.gz\n",
      "Approx 90% complete for Int_02_2.fastq.gz\n",
      "Approx 65% complete for Liv_03_2.fastq.gz\n",
      "Approx 75% complete for Liv_01_1.fastq.gz\n",
      "Approx 85% complete for Int_04_2.fastq.gz\n",
      "Approx 85% complete for Int_04_1.fastq.gz\n",
      "Approx 70% complete for Liv_03_1.fastq.gz\n",
      "Approx 95% complete for Int_02_1.fastq.gz\n",
      "Approx 95% complete for Int_02_2.fastq.gz\n",
      "Approx 80% complete for Liv_01_2.fastq.gz\n",
      "Approx 80% complete for Liv_01_1.fastq.gz\n",
      "Approx 70% complete for Liv_03_2.fastq.gz\n",
      "Analysis complete for Int_02_1.fastq.gz\n",
      "Approx 90% complete for Int_04_2.fastq.gz\n",
      "Approx 90% complete for Int_04_1.fastq.gz\n",
      "Approx 75% complete for Liv_03_1.fastq.gz\n",
      "Analysis complete for Int_02_2.fastq.gz\n",
      "Approx 85% complete for Liv_01_2.fastq.gz\n",
      "Approx 95% complete for Int_04_2.fastq.gz\n",
      "Approx 95% complete for Int_04_1.fastq.gz\n",
      "Approx 85% complete for Liv_01_1.fastq.gz\n",
      "Approx 75% complete for Liv_03_2.fastq.gz\n",
      "Approx 80% complete for Liv_03_1.fastq.gz\n",
      "Approx 90% complete for Liv_01_2.fastq.gz\n",
      "Analysis complete for Int_04_1.fastq.gz\n",
      "Analysis complete for Int_04_2.fastq.gz\n",
      "Approx 90% complete for Liv_01_1.fastq.gz\n",
      "Approx 80% complete for Liv_03_2.fastq.gz\n",
      "Approx 85% complete for Liv_03_1.fastq.gz\n",
      "Approx 95% complete for Liv_01_2.fastq.gz\n",
      "Approx 95% complete for Liv_01_1.fastq.gz\n",
      "Approx 85% complete for Liv_03_2.fastq.gz\n",
      "Approx 90% complete for Liv_03_1.fastq.gz\n",
      "Analysis complete for Liv_01_2.fastq.gz\n",
      "Analysis complete for Liv_01_1.fastq.gz\n",
      "Approx 95% complete for Liv_03_1.fastq.gz\n",
      "Approx 90% complete for Liv_03_2.fastq.gz\n",
      "Analysis complete for Liv_03_1.fastq.gz\n",
      "Approx 95% complete for Liv_03_2.fastq.gz\n",
      "Analysis complete for Liv_03_2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!mkdir fastq_trimmed_qc_sub\n",
    "!fastqc -t 16 fastq_trimmed_sub/*.fastq.gz -o fastq_trimmed_qc_sub "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n",
      "application/gzip\n",
      "Started analysis of Int-01_S42_L005_R1_001.fastq.gz\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "Approx 5% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-01_S42_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-02_S43_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-03_S44_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-03_S44_L005_R2_001.fastq.gz\n",
      "Started analysis of Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Analysis complete for Int-04_S45_L005_R1_001.fastq.gz\n",
      "Started analysis of Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Analysis complete for Int-04_S45_L005_R2_001.fastq.gz\n",
      "Started analysis of Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Analysis complete for Liv-01_S38_L005_R1_001.fastq.gz\n",
      "Started analysis of Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Analysis complete for Liv-01_S38_L005_R2_001.fastq.gz\n",
      "Started analysis of Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Analysis complete for Liv-02_S39_L005_R1_001.fastq.gz\n",
      "Started analysis of Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Analysis complete for Liv-02_S39_L005_R2_001.fastq.gz\n",
      "Started analysis of Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Analysis complete for Liv-03_S40_L005_R1_001.fastq.gz\n",
      "Started analysis of Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Analysis complete for Liv-03_S40_L005_R2_001.fastq.gz\n",
      "Started analysis of Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 5% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 10% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 15% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 20% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 25% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 30% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 35% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 40% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 45% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 50% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 55% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 60% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 65% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 70% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 75% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 80% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 85% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 90% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Approx 95% complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Analysis complete for Liv-04_S41_L005_R1_001.fastq.gz\n",
      "Started analysis of Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 5% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 10% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 15% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 20% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 25% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 30% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 35% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 40% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 45% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 50% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 55% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 60% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 65% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 70% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 75% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 80% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 85% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 90% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Approx 95% complete for Liv-04_S41_L005_R2_001.fastq.gz\n",
      "Analysis complete for Liv-04_S41_L005_R2_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!mkdir fastq_qc_sub\n",
    "!fastqc /data/classes/2024/fall/biol343/course_files/20240923_LH00283_0144_B22NMVKLT3/subsample/*fastq.gz -o fastq_qc_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[91m///\u001b[0m \u001b]8;id=152343;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Prepending directory to sample names\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/fastq_qc_sub\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/fastq_trimmed_qc_sub\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m64/64\u001b[0m  _trimmed_qc_sub/Liv_04_2_fastqc.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 32 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report_all_ipynb_sub.html   (overwritten)\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_report_all_ipynb_sub_data   (overwritten)\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc -f -d fastq_qc_sub/ fastq_trimmed_qc_sub/ -n multiqc_report_all_ipynb_sub.html"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## STAR Alignment\n",
    "    In this step the program runs STAR Alignment to map the Worm reads to the reference genome by finding where in the genome each read best matches to a gene\n",
    "    STAR Alignment is applied to the reads twice to ensure quality results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir genome/star\n",
    "!mkdir alignment_sub_2\n",
    "!mkdir alignment_sub_2/star"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate the Suffix Array\n",
    "The suffix Array is created to increase the speed and accurary of the alignment of the reads by allowing STAR to quickly index and locate potential alignment positions in the reference genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/data/users/mccallke0364/.conda/envs/final/bin/STAR-avx2 --runThreadN 16 --runMode genomeGenerate --genomeDir genome/star --genomeFastaFiles genome/genome.fa --sjdbGTFfile genome/annotations.gtf --sjdbOverhang 139 --genomeSAindexNbases 13\n",
      "\tSTAR version: 2.7.10b   compiled: 2023-05-25T06:56:23+0000 :/opt/conda/conda-bld/star_1684997536154/work/source\n",
      "Dec 15 17:25:42 ..... started STAR run\n",
      "Dec 15 17:25:42 ... starting to generate Genome files\n",
      "Dec 15 17:25:48 ..... processing annotations GTF\n",
      "Dec 15 17:25:50 ... starting to sort Suffix Array. This may take a long time...\n",
      "Dec 15 17:25:52 ... sorting Suffix Array chunks and saving them to disk...\n",
      "Dec 15 17:26:50 ... loading chunks from disk, packing SA...\n",
      "Dec 15 17:27:00 ... finished generating suffix array\n",
      "Dec 15 17:27:00 ... generating Suffix Array index\n",
      "Dec 15 17:27:37 ... completed Suffix Array index\n",
      "Dec 15 17:27:37 ..... inserting junctions into the genome indices\n",
      "Dec 15 17:28:13 ... writing Genome to disk ...\n",
      "Dec 15 17:28:14 ... writing Suffix Array to disk ...\n",
      "Dec 15 17:28:18 ... writing SAindex to disk\n",
      "Dec 15 17:28:19 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "!STAR \\\n",
    "--runThreadN 16  \\\n",
    "--runMode genomeGenerate \\\n",
    "--genomeDir genome/star \\\n",
    "--genomeFastaFiles genome/genome.fa \\\n",
    "--sjdbGTFfile genome/annotations.gtf \\\n",
    "--sjdbOverhang 139\\\n",
    "--genomeSAindexNbases 13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/data/users/mccallke0364/.conda/envs/final/bin/STAR-avx2 --runThreadN 16 --runMode alignReads --genomeDir genome/star --readFilesManifest manifest.tsv --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outSAMunmapped Within --outFileNamePrefix alignment_sub_2/star/\n",
      "\tSTAR version: 2.7.10b   compiled: 2023-05-25T06:56:23+0000 :/opt/conda/conda-bld/star_1684997536154/work/source\n",
      "Dec 18 19:32:45 ..... started STAR run\n",
      "Dec 18 19:32:45 ..... loading genome\n",
      "Dec 18 19:32:48 ..... started mapping\n",
      "Dec 18 23:58:55 ..... finished mapping\n",
      "Dec 18 23:58:58 ..... started sorting BAM\n",
      "Dec 19 00:05:27 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "!STAR \\\n",
    "--runThreadN 16\\\n",
    "--runMode alignReads \\\n",
    "--genomeDir genome/star \\\n",
    "--readFilesManifest manifest.tsv \\\n",
    "--readFilesCommand zcat \\\n",
    "--outSAMtype BAM SortedByCoordinate \\\n",
    "--outSAMunmapped Within \\\n",
    "--outFileNamePrefix alignment_sub_2/star/\n",
    "\n",
    "!cp alignment_sub_2/star/Log.final.out alignment_sub_2/star/firstlog.final.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/data/users/mccallke0364/.conda/envs/final/bin/STAR-avx2 --runThreadN 16 --runMode alignReads --genomeDir genome/star --readFilesManifest manifest.tsv --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outSAMunmapped Within --outSAMattributes NH HI AS nM RG --outFileNamePrefix alignment_sub_2/star/ --sjdbFileChrStartEnd alignment_sub_2/star/SJ.out.tab\n",
      "\tSTAR version: 2.7.10b   compiled: 2023-05-25T06:56:23+0000 :/opt/conda/conda-bld/star_1684997536154/work/source\n",
      "Dec 19 00:05:33 ..... started STAR run\n",
      "Dec 19 00:05:36 ..... loading genome\n",
      "Dec 19 00:05:45 ..... inserting junctions into the genome indices\n",
      "Dec 19 00:06:08 ..... started mapping\n",
      "Dec 19 04:46:45 ..... finished mapping\n",
      "Dec 19 04:46:48 ..... started sorting BAM\n",
      "Dec 19 04:53:33 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "!STAR\\\n",
    "--runThreadN 16 \\\n",
    "--runMode alignReads \\\n",
    "--genomeDir genome/star \\\n",
    "--readFilesManifest manifest.tsv \\\n",
    "--readFilesCommand zcat \\\n",
    "--outSAMtype BAM SortedByCoordinate \\\n",
    "--outSAMunmapped Within \\\n",
    "--outSAMattributes NH HI AS nM RG \\\n",
    "--outFileNamePrefix alignment_sub_2/star/ \\\n",
    "--sjdbFileChrStartEnd alignment_sub_2/star/SJ.out.tab\n",
    "\n",
    "!cp alignment_sub_2/star/Log.final.out alignment_sub_2/star/second.final.out"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Summary Stats\n",
    "    produces statistics about the alignment "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "252673365 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "3664337 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "15397899 + 0 mapped (6.09% : N/A)\n",
      "249009028 + 0 paired in sequencing\n",
      "124504514 + 0 read1\n",
      "124504514 + 0 read2\n",
      "11733514 + 0 properly paired (4.71% : N/A)\n",
      "11733514 + 0 with itself and mate mapped\n",
      "48 + 0 singletons (0.00% : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "!samtools flagstat alignment_sub_2/star/Aligned.sortedByCoord.out.bam"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Qualimap\n",
    "    This step allows us to perform quality control on the read mapping to see if the alignment is up to par"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Java memory size is set to 1200M\n",
      "Launching application...\n",
      "\n",
      "QualiMap v.2.3\n",
      "Built on 2023-05-19 16:57\n",
      "\n",
      "Selected tool: bamqc\n",
      "Available memory (Mb): 35\n",
      "Max memory (Mb): 1258\n",
      "Starting bam qc....\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 409\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 32\n",
      "Initializing regions from genome/annotations.gtf.....\n",
      "Found 241537 regions\n",
      "Filling region references... \n",
      "Processed 50 out of 409 windows...\n",
      "Processed 100 out of 409 windows...\n",
      "Processed 150 out of 409 windows...\n",
      "Processed 200 out of 409 windows...\n",
      "Processed 250 out of 409 windows...\n",
      "Processed 300 out of 409 windows...\n",
      "Processed 350 out of 409 windows...\n",
      "Processed 400 out of 409 windows...\n",
      "Total processed windows:409\n",
      "Number of reads: 249009028\n",
      "Number of valid reads: 11733562\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 9718104\n",
      "Num mapped first of pair: 4863165\n",
      "Num mapped second of pair: 4854939\n",
      "Num singletons: 30\n",
      "Time taken to analyze reads: 392\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 12342115281\n",
      "referenceSize: 391422299\n",
      "numberOfSequencedBases: 1284302800\n",
      "numberOfAs: 392663928\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 394\n",
      "end of bam qc\n",
      "Computing report...\n",
      "Writing HTML report...\n",
      "HTML report created successfully\n",
      "\n",
      "Finished\n",
      "Java memory size is set to 1200M\n",
      "Launching application...\n",
      "\n",
      "QualiMap v.2.3\n",
      "Built on 2023-05-19 16:57\n",
      "\n",
      "Selected tool: rnaseq\n",
      "Initializing regions from genome/annotations.gtf...\n",
      "\n",
      "Initialized 100000 regions...\n",
      "Initialized 200000 regions...\n",
      "\n",
      "Initialized 241537 regions it total\n",
      "\n",
      "Starting constructing transcripts for RNA-seq stats...\n",
      "Finished constructing transcripts\n",
      "\n",
      "Starting BAM file analysis\n",
      "\n",
      "Analyzed 500000 reads...\n",
      "Analyzed 1500000 reads...\n",
      "Analyzed 2000000 reads...\n",
      "Analyzed 3000000 reads...\n",
      "Analyzed 4000000 reads...\n",
      "Analyzed 4500000 reads...\n",
      "Analyzed 5000000 reads...\n",
      "Analyzed 5500000 reads...\n",
      "Analyzed 6500000 reads...\n",
      "Analyzed 7000000 reads...\n",
      "Analyzed 7500000 reads...\n",
      "Analyzed 8000000 reads...\n",
      "Analyzed 8500000 reads...\n",
      "Analyzed 9000000 reads...\n",
      "Analyzed 9500000 reads...\n",
      "Analyzed 10000000 reads...\n",
      "Analyzed 11500000 reads...\n",
      "Analyzed 14000000 reads...\n",
      "Analyzed 14500000 reads...\n",
      "Analyzed 15000000 reads...\n",
      "\n",
      "Processed 15397899 reads in total\n",
      "\n",
      "BAM file analysis finished\n",
      "Creating plots\n",
      "Writing HTML report...\n",
      "HTML report created successfully\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!qualimap bamqc -nt 32 -outdir qualimap_2/star/bam -bam alignment_sub_2/star/Aligned.sortedByCoord.out.bam --feature-file genome/annotations.gtf \n",
    "!qualimap rnaseq -outdir qualimap_2/star/rnaseq -bam alignment_sub_2/star/Aligned.sortedByCoord.out.bam -gtf genome/annotations.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[91m///\u001b[0m \u001b]8;id=471038;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Prepending directory to sample names\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/fastq_qc_sub\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/fastq_trimmed_qc_sub\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/alignment_sub_2\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m373/373\u001b[0m  0malimap/star/rnaseq/css/agogo.css\u001b[0m\n",
      "\u001b[?25h\u001b[34m|          qualimap\u001b[0m | Found 2 BamQC reports\n",
      "\u001b[34m|          qualimap\u001b[0m | Found 2 RNASeq reports\n",
      "\u001b[34m|              star\u001b[0m | Found 3 reports\n",
      "\u001b[34m|            fastqc\u001b[0m | Found 32 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report.html   (overwritten)\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_data   (overwritten)\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc --force -d fastq_qc_sub/ fastq_trimmed_qc_sub/ alignment_sub_2/ -n post_align.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir counting_sub_2\n",
    "!mkdir counting_sub_2/logs_sub\n",
    "!mkdir counting_sub_2/dedup_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2024-12-18 19:28:28\tAddOrReplaceReadGroups\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** \n",
      "https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    AddOrReplaceReadGroups -I alignment_sub/star/Aligned.sortedByCoord.out.bam -O alignment_sub/star/Aligned_with_correct_RG.bam -RGID Int-01 -RGLB lib1 -RGPL illumina -RGPU unit1 -RGSM Int-01\n",
      "**********\n",
      "\n",
      "\n",
      "19:28:28.068 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/users/mccallke0364/.conda/envs/final/share/picard-3.2.0-0/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Wed Dec 18 19:28:28 CST 2024] AddOrReplaceReadGroups INPUT=alignment_sub/star/Aligned.sortedByCoord.out.bam OUTPUT=alignment_sub/star/Aligned_with_correct_RG.bam RGID=Int-01 RGLB=lib1 RGPL=illumina RGPU=unit1 RGSM=Int-01    VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Wed Dec 18 19:28:28 CST 2024] Executing as mccallke0364@cn20 on Linux 4.18.0-193.el8.x86_64 amd64; OpenJDK 64-Bit Server VM 22.0.1-internal-adhoc.conda.src; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: 3.2.0-1-g3948afb6b\n",
      "INFO\t2024-12-18 19:28:28\tAddOrReplaceReadGroups\tCreated read-group ID=Int-01 PL=illumina LB=lib1 SM=Int-01\n",
      "\n",
      "INFO\t2024-12-18 19:28:34\tAddOrReplaceReadGroups\tProcessed     1,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_1:34,163,443\n",
      "INFO\t2024-12-18 19:28:40\tAddOrReplaceReadGroups\tProcessed     2,000,000 records.  Elapsed time: 00:00:12s.  Time for last 1,000,000:    5s.  Last read position: SM_V10_1:61,151,760\n",
      "INFO\t2024-12-18 19:28:45\tAddOrReplaceReadGroups\tProcessed     3,000,000 records.  Elapsed time: 00:00:17s.  Time for last 1,000,000:    5s.  Last read position: SM_V10_1:66,932,065\n",
      "INFO\t2024-12-18 19:28:51\tAddOrReplaceReadGroups\tProcessed     4,000,000 records.  Elapsed time: 00:00:23s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_Z:18,415,555\n",
      "INFO\t2024-12-18 19:28:57\tAddOrReplaceReadGroups\tProcessed     5,000,000 records.  Elapsed time: 00:00:29s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_Z:45,726,428\n",
      "INFO\t2024-12-18 19:29:03\tAddOrReplaceReadGroups\tProcessed     6,000,000 records.  Elapsed time: 00:00:35s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_Z:83,285,024\n",
      "INFO\t2024-12-18 19:29:09\tAddOrReplaceReadGroups\tProcessed     7,000,000 records.  Elapsed time: 00:00:41s.  Time for last 1,000,000:    5s.  Last read position: SM_V10_3:25,752,309\n",
      "INFO\t2024-12-18 19:29:15\tAddOrReplaceReadGroups\tProcessed     8,000,000 records.  Elapsed time: 00:00:47s.  Time for last 1,000,000:    5s.  Last read position: SM_V10_4:7,204,455\n",
      "INFO\t2024-12-18 19:29:21\tAddOrReplaceReadGroups\tProcessed     9,000,000 records.  Elapsed time: 00:00:53s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_4:45,588,107\n",
      "INFO\t2024-12-18 19:29:27\tAddOrReplaceReadGroups\tProcessed    10,000,000 records.  Elapsed time: 00:00:59s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_2:36,688,696\n",
      "INFO\t2024-12-18 19:29:32\tAddOrReplaceReadGroups\tProcessed    11,000,000 records.  Elapsed time: 00:01:04s.  Time for last 1,000,000:    4s.  Last read position: SM_V10_2:40,296,333\n",
      "INFO\t2024-12-18 19:29:38\tAddOrReplaceReadGroups\tProcessed    12,000,000 records.  Elapsed time: 00:01:10s.  Time for last 1,000,000:    5s.  Last read position: SM_V10_6:23,118,571\n",
      "INFO\t2024-12-18 19:29:44\tAddOrReplaceReadGroups\tProcessed    13,000,000 records.  Elapsed time: 00:01:16s.  Time for last 1,000,000:    6s.  Last read position: SM_V10_7:12,547,306\n",
      "INFO\t2024-12-18 19:29:51\tAddOrReplaceReadGroups\tProcessed    14,000,000 records.  Elapsed time: 00:01:22s.  Time for last 1,000,000:    6s.  Last read position: */*\n",
      "INFO\t2024-12-18 19:29:59\tAddOrReplaceReadGroups\tProcessed    15,000,000 records.  Elapsed time: 00:01:31s.  Time for last 1,000,000:    8s.  Last read position: */*\n",
      "^C\n",
      "@HD\tVN:1.4\tSO:coordinate\n",
      "@SQ\tSN:SM_V10_1\tLN:87984048\n",
      "@SQ\tSN:SM_V10_Z\tLN:86692428\n",
      "@SQ\tSN:SM_V10_3\tLN:49793710\n",
      "@SQ\tSN:SM_V10_4\tLN:46471573\n",
      "@SQ\tSN:SM_V10_2\tLN:45716244\n",
      "@SQ\tSN:SM_V10_6\tLN:24696076\n",
      "@SQ\tSN:SM_V10_5\tLN:24128858\n",
      "@SQ\tSN:SM_V10_7\tLN:19942475\n",
      "@SQ\tSN:SM_V10_WSR\tLN:5969970\n",
      "@SQ\tSN:SM_V10_MITO\tLN:26917\n",
      "@PG\tID:STAR\tPN:STAR\tVN:2.7.10b\tCL:/data/users/mccallke0364/.conda/envs/final/bin/STAR-avx2   --runMode alignReads      --runThreadN 16   --genomeDir genome/star   --readFilesCommand zcat      --readFilesManifest manifest.tsv      --outFileNamePrefix alignment_sub/star/   --outSAMtype BAM   SortedByCoordinate      --outSAMunmapped Within   \n",
      "@RG\tID:Int_01\tSM:Int_01\n",
      "@RG\tID:Int_02\tSM:Int_02\n",
      "@RG\tID:Int_03\tSM:Int_03\n",
      "@RG\tID:Int_04\tSM:Int_04\n",
      "@RG\tID:Liv_01\tSM:Liv_01\n",
      "@RG\tID:Liv_02\tSM:Liv_02\n",
      "@RG\tID:Liv_03\tSM:Liv_03\n",
      "@RG\tID:Liv_04\tSM:Liv_04\n",
      "@CO\tuser command line: /data/users/mccallke0364/.conda/envs/final/bin/STAR-avx2 --runThreadN 16 --runMode alignReads --genomeDir genome/star --readFilesManifest manifest.tsv --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outSAMunmapped Within --outFileNamePrefix alignment_sub/star/\n"
     ]
    }
   ],
   "source": [
    "# !samtools view -H alignment_sub/star/Aligned.sortedByCoord.out.bam"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mark Duplicates\n",
    "    Allows efficient plotting by marking reads that would affect alignment anaysis and over-represent data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "09:27:42.544 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/users/mccallke0364/.conda/envs/final/share/picard-3.2.0-0/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Thu Dec 19 09:27:42 CST 2024] MarkDuplicates --INPUT alignment_sub_2/star/Aligned.sortedByCoord.out.bam --OUTPUT counting_sub_2/dedup_sub/star.bam --METRICS_FILE counting_sub_2/logs_sub/star_duplicates --VALIDATION_STRINGENCY SILENT --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_DUP_STRATEGY FLOW_QUALITY_SUM_STRATEGY --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --UNPAIRED_START_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --REMOVE_DUPLICATES false --ASSUME_SORTED false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Thu Dec 19 09:27:42 CST 2024] Executing as mccallke0364@cn20 on Linux 4.18.0-193.el8.x86_64 amd64; OpenJDK 64-Bit Server VM 22.0.1-internal-adhoc.conda.src; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: Version:3.2.0-1-g3948afb6b\n",
      "INFO\t2024-12-19 09:27:42\tMarkDuplicates\tStart of doWork freeMemory: 222675184; totalMemory: 234881024; maxMemory: 107374182400\n",
      "INFO\t2024-12-19 09:27:42\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2024-12-19 09:27:42\tMarkDuplicates\tWill retain up to 389036892 data points before spilling to disk.\n",
      "INFO\t2024-12-19 09:27:46\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:02s.  Time for last 1,000,000:    2s.  Last read position: SM_V10_1:34,149,235\n",
      "INFO\t2024-12-19 09:27:46\tMarkDuplicates\tTracking 879 as yet unmatched pairs. 879 records in RAM.\n",
      "INFO\t2024-12-19 09:27:47\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_1:58,243,431\n",
      "INFO\t2024-12-19 09:27:47\tMarkDuplicates\tTracking 205 as yet unmatched pairs. 205 records in RAM.\n",
      "INFO\t2024-12-19 09:27:49\tMarkDuplicates\tRead     3,000,000 records.  Elapsed time: 00:00:05s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_1:65,792,109\n",
      "INFO\t2024-12-19 09:27:49\tMarkDuplicates\tTracking 442 as yet unmatched pairs. 442 records in RAM.\n",
      "INFO\t2024-12-19 09:27:51\tMarkDuplicates\tRead     4,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_Z:16,115,995\n",
      "INFO\t2024-12-19 09:27:51\tMarkDuplicates\tTracking 201 as yet unmatched pairs. 201 records in RAM.\n",
      "INFO\t2024-12-19 09:27:53\tMarkDuplicates\tRead     5,000,000 records.  Elapsed time: 00:00:09s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_Z:43,825,406\n",
      "INFO\t2024-12-19 09:27:53\tMarkDuplicates\tTracking 834 as yet unmatched pairs. 834 records in RAM.\n",
      "INFO\t2024-12-19 09:27:55\tMarkDuplicates\tRead     6,000,000 records.  Elapsed time: 00:00:11s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_Z:82,979,802\n",
      "INFO\t2024-12-19 09:27:55\tMarkDuplicates\tTracking 206 as yet unmatched pairs. 206 records in RAM.\n",
      "INFO\t2024-12-19 09:27:56\tMarkDuplicates\tRead     7,000,000 records.  Elapsed time: 00:00:13s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_3:25,531,109\n",
      "INFO\t2024-12-19 09:27:56\tMarkDuplicates\tTracking 100 as yet unmatched pairs. 100 records in RAM.\n",
      "INFO\t2024-12-19 09:27:58\tMarkDuplicates\tRead     8,000,000 records.  Elapsed time: 00:00:14s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_4:5,612,627\n",
      "INFO\t2024-12-19 09:27:58\tMarkDuplicates\tTracking 73 as yet unmatched pairs. 73 records in RAM.\n",
      "INFO\t2024-12-19 09:28:00\tMarkDuplicates\tRead     9,000,000 records.  Elapsed time: 00:00:16s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_4:44,867,134\n",
      "INFO\t2024-12-19 09:28:00\tMarkDuplicates\tTracking 139 as yet unmatched pairs. 139 records in RAM.\n",
      "INFO\t2024-12-19 09:28:02\tMarkDuplicates\tRead    10,000,000 records.  Elapsed time: 00:00:18s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_2:35,870,700\n",
      "INFO\t2024-12-19 09:28:02\tMarkDuplicates\tTracking 217 as yet unmatched pairs. 217 records in RAM.\n",
      "INFO\t2024-12-19 09:28:03\tMarkDuplicates\tRead    11,000,000 records.  Elapsed time: 00:00:20s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_2:39,478,608\n",
      "INFO\t2024-12-19 09:28:03\tMarkDuplicates\tTracking 837 as yet unmatched pairs. 837 records in RAM.\n",
      "INFO\t2024-12-19 09:28:05\tMarkDuplicates\tRead    12,000,000 records.  Elapsed time: 00:00:21s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_2:39,755,229\n",
      "INFO\t2024-12-19 09:28:05\tMarkDuplicates\tTracking 74491 as yet unmatched pairs. 74491 records in RAM.\n",
      "INFO\t2024-12-19 09:28:07\tMarkDuplicates\tRead    13,000,000 records.  Elapsed time: 00:00:23s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_6:7,768,270\n",
      "INFO\t2024-12-19 09:28:07\tMarkDuplicates\tTracking 96 as yet unmatched pairs. 96 records in RAM.\n",
      "INFO\t2024-12-19 09:28:08\tMarkDuplicates\tRead    14,000,000 records.  Elapsed time: 00:00:25s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_5:12,732,503\n",
      "INFO\t2024-12-19 09:28:08\tMarkDuplicates\tTracking 58 as yet unmatched pairs. 58 records in RAM.\n",
      "INFO\t2024-12-19 09:28:10\tMarkDuplicates\tRead    15,000,000 records.  Elapsed time: 00:00:26s.  Time for last 1,000,000:    1s.  Last read position: SM_V10_MITO:2,892\n",
      "INFO\t2024-12-19 09:28:10\tMarkDuplicates\tTracking 12659 as yet unmatched pairs. 12659 records in RAM.\n",
      "INFO\t2024-12-19 09:28:11\tMarkDuplicates\tRead 15397899 records. 0 pairs never matched.\n",
      "INFO\t2024-12-19 09:28:12\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 9906522008; totalMemory: 17448304640; maxMemory: 107374182400\n",
      "INFO\t2024-12-19 09:28:12\tMarkDuplicates\tWill retain up to 2147483642 duplicate indices before spilling to disk.\n",
      "INFO\t2024-12-19 09:28:15\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 4225\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 3:1,379\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         2,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 4:15,424\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         3,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 6:-27,246\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         4,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 4:14,555\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 1179\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 5:-31,518\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 2245\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 0:10,459\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         2,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 7:-29,567\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 1799\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 4:24,115\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 1513\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 3:8,323\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 1047\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 0:5,420\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tLarge duplicate set. size = 1018\n",
      "INFO\t2024-12-19 09:28:17\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 0:31,734\n",
      "INFO\t2024-12-19 09:28:18\tOpticalDuplicateFinder\tLarge duplicate set. size = 1053\n",
      "INFO\t2024-12-19 09:28:18\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 0:-27,505\n",
      "INFO\t2024-12-19 09:28:18\tOpticalDuplicateFinder\tLarge duplicate set. size = 1074\n",
      "INFO\t2024-12-19 09:28:18\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 4:-27,330\n",
      "INFO\t2024-12-19 09:28:18\tOpticalDuplicateFinder\tLarge duplicate set. size = 1701\n",
      "INFO\t2024-12-19 09:28:18\tOpticalDuplicateFinder\tcompared         1,000 ReadEnds to others.  Elapsed time: 00:00:00s.  Time for last 1,000:    0s.  Last read position: 7:-27,884\n",
      "INFO\t2024-12-19 09:28:18\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2024-12-19 09:28:19\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2024-12-19 09:28:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 25137169640; totalMemory: 42345693184; maxMemory: 107374182400\n",
      "INFO\t2024-12-19 09:28:19\tMarkDuplicates\tMarking 4015259 records as duplicates.\n",
      "INFO\t2024-12-19 09:28:19\tMarkDuplicates\tFound 218783 optical duplicate clusters.\n",
      "INFO\t2024-12-19 09:28:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
      "INFO\t2024-12-19 09:29:22\tMarkDuplicates\tWritten    10,000,000 records.  Elapsed time: 00:01:03s.  Time for last 10,000,000:   63s.  Last read position: SM_V10_2:35,870,700\n",
      "INFO\t2024-12-19 09:30:34\tMarkDuplicates\tWritten    20,000,000 records.  Elapsed time: 00:02:15s.  Time for last 10,000,000:   71s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:32:03\tMarkDuplicates\tWritten    30,000,000 records.  Elapsed time: 00:03:44s.  Time for last 10,000,000:   89s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:33:34\tMarkDuplicates\tWritten    40,000,000 records.  Elapsed time: 00:05:15s.  Time for last 10,000,000:   91s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:35:08\tMarkDuplicates\tWritten    50,000,000 records.  Elapsed time: 00:06:49s.  Time for last 10,000,000:   93s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:36:44\tMarkDuplicates\tWritten    60,000,000 records.  Elapsed time: 00:08:25s.  Time for last 10,000,000:   96s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:38:20\tMarkDuplicates\tWritten    70,000,000 records.  Elapsed time: 00:10:01s.  Time for last 10,000,000:   95s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:39:54\tMarkDuplicates\tWritten    80,000,000 records.  Elapsed time: 00:11:35s.  Time for last 10,000,000:   94s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:41:31\tMarkDuplicates\tWritten    90,000,000 records.  Elapsed time: 00:13:12s.  Time for last 10,000,000:   97s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:43:05\tMarkDuplicates\tWritten   100,000,000 records.  Elapsed time: 00:14:45s.  Time for last 10,000,000:   93s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:44:38\tMarkDuplicates\tWritten   110,000,000 records.  Elapsed time: 00:16:18s.  Time for last 10,000,000:   92s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:46:13\tMarkDuplicates\tWritten   120,000,000 records.  Elapsed time: 00:17:53s.  Time for last 10,000,000:   95s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:47:46\tMarkDuplicates\tWritten   130,000,000 records.  Elapsed time: 00:19:27s.  Time for last 10,000,000:   93s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:49:18\tMarkDuplicates\tWritten   140,000,000 records.  Elapsed time: 00:20:59s.  Time for last 10,000,000:   91s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:50:51\tMarkDuplicates\tWritten   150,000,000 records.  Elapsed time: 00:22:32s.  Time for last 10,000,000:   93s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:52:26\tMarkDuplicates\tWritten   160,000,000 records.  Elapsed time: 00:24:06s.  Time for last 10,000,000:   94s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:53:57\tMarkDuplicates\tWritten   170,000,000 records.  Elapsed time: 00:25:38s.  Time for last 10,000,000:   91s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:55:27\tMarkDuplicates\tWritten   180,000,000 records.  Elapsed time: 00:27:08s.  Time for last 10,000,000:   90s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:57:00\tMarkDuplicates\tWritten   190,000,000 records.  Elapsed time: 00:28:40s.  Time for last 10,000,000:   92s.  Last read position: */*\n",
      "INFO\t2024-12-19 09:58:34\tMarkDuplicates\tWritten   200,000,000 records.  Elapsed time: 00:30:14s.  Time for last 10,000,000:   93s.  Last read position: */*\n",
      "INFO\t2024-12-19 10:00:07\tMarkDuplicates\tWritten   210,000,000 records.  Elapsed time: 00:31:48s.  Time for last 10,000,000:   93s.  Last read position: */*\n",
      "INFO\t2024-12-19 10:01:46\tMarkDuplicates\tWritten   220,000,000 records.  Elapsed time: 00:33:26s.  Time for last 10,000,000:   98s.  Last read position: */*\n",
      "INFO\t2024-12-19 10:03:22\tMarkDuplicates\tWritten   230,000,000 records.  Elapsed time: 00:35:03s.  Time for last 10,000,000:   96s.  Last read position: */*\n",
      "INFO\t2024-12-19 10:04:55\tMarkDuplicates\tWritten   240,000,000 records.  Elapsed time: 00:36:36s.  Time for last 10,000,000:   92s.  Last read position: */*\n",
      "INFO\t2024-12-19 10:06:31\tMarkDuplicates\tWritten   250,000,000 records.  Elapsed time: 00:38:12s.  Time for last 10,000,000:   95s.  Last read position: */*\n",
      "INFO\t2024-12-19 10:06:57\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
      "INFO\t2024-12-19 10:06:57\tMarkDuplicates\tDuplicate Index cleanup.\n",
      "INFO\t2024-12-19 10:06:57\tMarkDuplicates\tGetting Memory Stats.\n",
      "INFO\t2024-12-19 10:06:57\tMarkDuplicates\tBefore output close freeMemory: 2554697544; totalMemory: 2583691264; maxMemory: 107374182400\n",
      "INFO\t2024-12-19 10:06:57\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
      "INFO\t2024-12-19 10:06:57\tMarkDuplicates\tAfter output close freeMemory: 642094920; totalMemory: 671088640; maxMemory: 107374182400\n",
      "[Thu Dec 19 10:06:57 CST 2024] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 39.25 minutes.\n",
      "Runtime.totalMemory()=671088640\n"
     ]
    }
   ],
   "source": [
    "!picard MarkDuplicates -I alignment_sub_2/star/Aligned.sortedByCoord.out.bam -M counting_sub_2/logs_sub/star_duplicates -O counting_sub_2/dedup_sub/star.bam -Xmx100g --VALIDATION_STRINGENCY SILENT"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Count Features\n",
    "    Counts the number of reads that map to each feature (genes, exons, promoters) featureCounts is known for it's efficiency and accuracy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.6\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m1 BAM file  \u001b[0m \u001b[0m                                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                           \u001b[36mstar.bam\u001b[0m \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mstar_counts_2.tsv\u001b[0m \u001b[0m                               ||\n",
      "||                 Summary : \u001b[36mstar_counts_2.tsv.summary\u001b[0m \u001b[0m                       ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||              Annotation : \u001b[36mannotations.gtf (GTF)\u001b[0m \u001b[0m                           ||\n",
      "||      Dir for temp files : \u001b[36m./\u001b[0m \u001b[0m                                              ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m32\u001b[0m \u001b[0m                                              ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||      Multimapping reads : \u001b[36mcounted (fractional)\u001b[0m \u001b[0m                            ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||        Duplicated Reads : \u001b[36mignored\u001b[0m \u001b[0m                                         ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file annotations.gtf \u001b[0m... \u001b[0m                                  ||\n",
      "||    Features : \u001b[36m88812\u001b[0m \u001b[0m                                                       ||\n",
      "||    Meta-features : \u001b[36m9920\u001b[0m \u001b[0m                                                   ||\n",
      "||    Chromosomes/contigs : \u001b[36m10\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file star.bam... \u001b[0m                                              ||\n",
      "||    Strand specific : \u001b[36mstranded\u001b[0m \u001b[0m                                             ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| WARNING: \u001b[36mmore than 10 'M' segments were found in CIGAR string.\u001b[0m \u001b[0m            ||\n",
      "|| The exceeding sections are not useed. Please consider to increase maxMOp. \u001b[0m ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total read groups : \u001b[36m8\u001b[0m \u001b[0m                                                  ||\n",
      "||    Total alignments : \u001b[36m252673365\u001b[0m \u001b[0m                                           ||\n",
      "||    Successfully assigned alignments : \u001b[36m135014 (0.1%)\u001b[0m \u001b[0m                       ||\n",
      "||    Running time : \u001b[36m0.15 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"star_counts_2.tsv.summa \u001b[0m ||\n",
      "|| ry\" \u001b[0m                                                                       ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!featureCounts -T 32 \\\n",
    "    counting_sub_2/dedup_sub/star.bam \\\n",
    "    -T 32 \\\n",
    "    -p \\\n",
    "    --byReadGroup \\\n",
    "    -s 1 \\\n",
    "    --ignoreDup \\\n",
    "    -M \\\n",
    "    --fraction \\\n",
    "    -a genome/annotations.gtf \\\n",
    "    -o star_counts_2.tsv \\\n",
    "    --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[91m///\u001b[0m \u001b]8;id=74092;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.17\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Prepending directory to sample names\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/fastq_qc_sub\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/fastq_trimmed_qc_sub\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/alignment_sub_2\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/counting_sub_2\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m75/75\u001b[0m  qc_sub/Liv_03_1_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m|            picard\u001b[0m | Found 1 MarkDuplicates reports\n",
      "\u001b[34m|              star\u001b[0m | Found 1 reports\n",
      "\u001b[34m|            fastqc\u001b[0m | Found 32 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : alignment_qc.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : alignment_qc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc --force -d fastq_qc_sub/ fastq_trimmed_qc_sub/ alignment_sub_2/ counting_sub_2/ -n alignment_qc.html"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plotting results\n",
    "Creates a heat map, volcano plot and a PCA graph\n",
    "-   The heat map shows the distance of features being mapped in each sample\n",
    "-   <img src=\"/data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/plots/dists_plot.png\">\n",
    "-   The volcano plot shows the expression of each feature\n",
    "-   <img src= \"/data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/plots/volcano_plot_2.png\">\n",
    "-   PCA graph shows how the different cell types are effected by the most expressed genes\n",
    "<img src=\"/data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/plots/pca_desq.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘plots’: File exists\n",
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.5\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.1     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.4     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.2     \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "\u001b[?25h\u001b[?25hLoading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    second, second<-\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:lubridate’:\n",
      "\n",
      "    %within%\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: SummarizedExperiment\n",
      "Loading required package: MatrixGenerics\n",
      "Loading required package: matrixStats\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\u001b[?25h\u001b[1mRows: \u001b[22m\u001b[34m9920\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m14\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (5): Geneid, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (9): Length, counting_sub_2/dedup_sub/star.bam:Int-01, counting_sub_2/de...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[1m\u001b[22mSaving 7 x 7 in image\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h[1] TRUE\n",
      "\u001b[?25hWarning message:\n",
      "In DESeqDataSet(se, design = design, ignoreRank) :\n",
      "  some variables in design formula are characters, converting to factors\n",
      "\u001b[?25hestimating size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[1m\u001b[22mSaving 7 x 7 in image\n",
      "\u001b[?25h\u001b[?25h\u001b[1m\u001b[22mSaving 7 x 7 in image\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[1m\u001b[33mError\u001b[39m in `pivot_longer()`:\u001b[22m\n",
      "\u001b[33m!\u001b[39m Names must be unique.\n",
      "\u001b[31m✖\u001b[39m These names are duplicated:\n",
      "  * \"comp\" at locations 1 and 3.\n",
      "  * \"dist\" at locations 2 and 4.\n",
      "\u001b[34mℹ\u001b[39m Use argument `names_repair` to specify repair strategy.\n",
      "Backtrace:\n",
      "\u001b[90m     \u001b[39m▆\n",
      "\u001b[90m  1. \u001b[39m├─\u001b[1mtidyr\u001b[22m::pivot_longer(...)\n",
      "\u001b[90m  2. \u001b[39m└─\u001b[1m\u001b[94mtidyr:::pivot_longer.data.frame(\u001b[39m\u001b[22mdists_df, cols = \"sample\", names_to = \"comp\", values_to = \"dist\"\u001b[1m\u001b[94m)\u001b[39m\u001b[22m\n",
      "\u001b[90m  3. \u001b[39m  └─tidyr::pivot_longer_spec(...)\n",
      "\u001b[90m  4. \u001b[39m    ├─tidyr:::wrap_error_names(...)\n",
      "\u001b[90m  5. \u001b[39m    │ └─\u001b[1mbase\u001b[22m::withCallingHandlers(...)\n",
      "\u001b[90m  6. \u001b[39m    └─\u001b[1mvctrs\u001b[22m::vec_cbind(...)\n",
      "\u001b[90m  7. \u001b[39m      └─vctrs (local) `<fn>`()\n",
      "\u001b[90m  8. \u001b[39m        └─vctrs:::validate_unique(names = names, arg = arg, call = call)\n",
      "\u001b[90m  9. \u001b[39m          └─vctrs:::stop_names_must_be_unique(names, arg, call = call)\n",
      "\u001b[90m 10. \u001b[39m            └─vctrs:::stop_names(...)\n",
      "\u001b[90m 11. \u001b[39m              └─vctrs:::stop_vctrs(...)\n",
      "\u001b[90m 12. \u001b[39m                └─\u001b[1mrlang\u001b[22m::abort(message, class = c(class, \"vctrs_error\"), ..., call = call)\n",
      "Execution halted\n",
      "\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!mkdir plots\n",
    "!Rscript differential_expressions.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"/data/users/mccallke0364/Final_Project/Sm_Mira_IvT/subfiles/plots/dist_plot.png\">"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "final_project_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
